<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2025-07-31">

<title>Plotting the phylogenetic tree and associated taxonmic information â€“ duckweedSynCom</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="../../site_libs/dt-ext-fixedcolumns-1.13.6/css/fixedColumns.dataTables.min.css" rel="stylesheet">
<script src="../../site_libs/dt-ext-fixedcolumns-1.13.6/js/dataTables.fixedColumns.min.js"></script>
<link href="../../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/bac_taxonomy_sanger_16S/01_read_format_sanger.html">1. Taxonomic characterization of plant-associated heterotrophic bacteria</a></li><li class="breadcrumb-item"><a href="../../R/bac_taxonomy_sanger_16S/03_tree_summary_table.html">iii) Taxonomy summary</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">duckweedSynCom</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. Taxonomic characterization of plant-associated heterotrophic bacteria</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/bac_taxonomy_sanger_16S/01_read_format_sanger.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) 16S rRNA gene Sanger sequencing: Process ab1files to fasta</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/bac_taxonomy_sanger_16S/02_seq_cluster_phylo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Clustering and phylogeny of 16S gene</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/bac_taxonomy_sanger_16S/03_tree_summary_table.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">iii) Taxonomy summary</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read-and-format" id="toc-read-and-format" class="nav-link" data-scroll-target="#read-and-format"><span class="header-section-number">2</span> Read and format</a></li>
  <li><a href="#summary-table" id="toc-summary-table" class="nav-link" data-scroll-target="#summary-table"><span class="header-section-number">3</span> Summary table</a></li>
  <li><a href="#phylogenetic-tree-plotting" id="toc-phylogenetic-tree-plotting" class="nav-link" data-scroll-target="#phylogenetic-tree-plotting"><span class="header-section-number">4</span> Phylogenetic tree plotting</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/bac_taxonomy_sanger_16S/01_read_format_sanger.html">1. Taxonomic characterization of plant-associated heterotrophic bacteria</a></li><li class="breadcrumb-item"><a href="../../R/bac_taxonomy_sanger_16S/03_tree_summary_table.html">iii) Taxonomy summary</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Plotting the phylogenetic tree and associated taxonmic information</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 31, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    Display information from the previous steps in a nice format
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Loads required libraries</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">options</span>(<span class="at">repos =</span> <span class="fu">c</span>(<span class="at">CRAN =</span> <span class="st">"https://packagemanager.posit.co/cran/__linux__/jammy/latest"</span>))</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(here)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(DT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-and-format" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read and format</h1>
<p>Read phylogenetic tree and associated data table from the previous step</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>syncom16s_seqs_mltree_nwk <span class="ot">&lt;-</span> treeio<span class="sc">::</span><span class="fu">read.newick</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.nwk"</span>),</span>
<span id="cb2-2"><a href="#cb2-2"></a>                            <span class="at">node.label=</span><span class="st">"support"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a>final_table <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strain_table_info.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="summary-table" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Summary table</h1>
<p>Below is a compiled table with taxonomic info, isolation source, and mapping of new strainID to the original <code>google_drive_id.</code> Taxa cluster refers to the 99% sequence identity cluster that each strain is a member of. Table is sorted by <code>taxa_cluster.</code> Columns <code>domain</code> through <code>genus</code> are the taxonomic output from the IDTAXA algorithm using the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a> 16S rRNA reference training set. The percentage following the assigned taxa classification represents the classification confidence/probability.</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-77f153797a935c2df6e1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-77f153797a935c2df6e1">{"x":{"filter":"none","vertical":false,"extensions":["FixedColumns"],"data":[["HA25","HA37","HA53","HA63","TL20","TL26","TL35","TL10","TL19","TL33","TL25","TL24","TL39","TL32","TL46","TL62","TL45","TL53","TL12","SJ08","SJ20","SJ18","HA64","TL01","TL05","TL07","TL16","TL52","TL50","TL30","TL42","TL44","TL34","TL29","TL37","HA12","SJ16","HA02","HA30","SJ13","HA39","SJ10","SJ12","SJ07","SJ23","HA81","TL38","TL06","HA13","HA04","HA23","HA03","TL09","TL23","TL47","TL54","TL61","HA21","TL03","TL04","TL51","TL11","TL41","TL22","TL18","TL63","TL65","TL31","TL14","HA82","HA46","HA32","HA68","HA35","HA65","SJ05","TL55","SJ03","HA51","HA08","HA28","HA45","HA56","HA67","HA33","HA57","HA80","HA16","TL64","SJ14","HA40","HA50","HA48","HA17","HA01","HA11","HA47","HA06","HA72","HA10","HA14","HA61","HA59","HA07","HA41","TL58","SJ02","HA29","HA34","HA42","HA69","HA44","HA18","HA52","HA27","TL02","SJ17","HA36"],["HA_C6","HA_E2","HA_F10","HA_G12","TL_B8","TL_C8","TL_D11","TL_A6","TL_B7","TL_D8","TL_C7","TL_C6","TL_E7","TL_D6","TL_F6","TL_H8","TL_F5","TL_G6","TL_A8","SJ_D5","SJ_D09","SJ_H9","HA_H1","TL_A1","TL_D1","TL_G1","TL_B2","TL_G5","TL_G2","TL_D3","TL_E10","TL_F4","TL_D9","TL_D2","TL_E4","HA_B2","SJ_G5","HA_A2","HA_C11","SJ_F6","HA_E5","SJ_E5","SJ_F5","SJ_D2","SJ_F10","HA_G05","TL_E5","TL_E1","HA_B3","HA_A4","HA_C4","HA_A3","TL_A5","TL_C5","TL_F7","TL_G8","TL_H7","HA_C1","TL_B1","TL_C1","TL_G4","TL_A7","TL_E9","TL_C3","TL_B6","TL_H9","TL_H11","TL_D5","TL_A11","HA_H10","HA_F3","HA_D7","HA_H8","HA_D12","HA_H2","SJ_C5","TL_G9","SJ_A10","HA_F8","HA_A8","HA_C9","HA_F2","HA_G2","HA_H6","HA_D8","HA_G4","HA_G02","HA_B6","TL_H10","SJ_F9","HA_E6","HA_F7","HA_F5","HA_B7","HA_A1","HA_B1","HA_F4","HA_A6","HA_A05","HA_A12","HA_B4","HA_G9","HA_G6","HA_A7","HA_E8","TL_H4","SJ_A9","HA_C10","HA_D10","HA_E9","HA_H9","HA_F1","HA_B8","HA_F9","HA_C8","TL_A2","SJ_H2","HA_E1"],[1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,3,4,4,5,5,5,5,5,5,5,5,5,6,6,6,7,7,8,8,8,8,8,8,8,8,8,9,10,11,11,11,11,12,13,13,13,13,14,15,15,15,15,15,15,15,15,15,15,15,16,17,17,17,17,17,18,19,20,21,21,21,21,21,21,21,21,21,22,23,24,25,25,25,25,26,26,26,26,26,27,27,27,28,29,29,30,31,32,32,32,33,34,35,36,37,38,39,40],["Exiguobacterium HA25","Exiguobacterium HA37","Exiguobacterium HA53","Exiguobacterium HA63","Lactococcus TL20","Lactococcus TL26","Lactococcus TL35","Lactococcus TL10","Lactococcus TL19","Lactococcus TL33","Lactococcus TL25","Lactococcus TL24","Lactococcus TL39","Lactococcus TL32","Lactococcus TL46","Lactococcus TL62","Lactococcus TL45","Lactococcus TL53","Lactococcus TL12","Serratia SJ08","Serratia SJ20","Aeromonas SJ18","Aeromonas HA64","Pseudomonas TL01","Pseudomonas TL05","Pseudomonas TL07","Pseudomonas TL16","Pseudomonas TL52","Pseudomonas TL50","Pseudomonas TL30","Pseudomonas TL42","Pseudomonas TL44","Pseudomonas TL34","Pseudomonas TL29","Pseudomonas TL37","Pseudomonas HA12","Pseudomonas SJ16","Pseudomonas HA02","Pseudomonas HA30","Pseudomonas SJ13","Pseudomonas HA39","Pseudomonas SJ10","Pseudomonas SJ12","Pseudomonas SJ07","Pseudomonas SJ23","Pseudomonas HA81","Pantoea TL38","Pseudomonas TL06","Aeromonas HA13","Aeromonas HA04","Aeromonas HA23","Aeromonas HA03","Lelliottia TL09","Pseudomonas TL23","Pseudomonas TL47","Pseudomonas TL54","Pseudomonas TL61","Aeromonas HA21","Pseudomonas TL03","Pseudomonas TL04","Pseudomonas TL51","Pseudomonas TL11","Pseudomonas TL41","Pseudomonas TL22","Pseudomonas TL18","Pseudomonas TL63","Pseudomonas TL65","Pseudomonas TL31","Pseudomonas TL14","Acinetobacter HA82","Aquitalea HA46","Aquitalea HA32","Aquitalea HA68","Aquitalea HA35","Aquitalea HA65","Pseudomonas SJ05","Pseudomonas TL55","Pelomonas SJ03","Chryseobacterium HA51","Chryseobacterium HA08","Chryseobacterium HA28","Chryseobacterium HA45","Chryseobacterium HA56","Chryseobacterium HA67","Chryseobacterium HA33","Chryseobacterium HA57","Chryseobacterium HA80","Comamonas HA16","Acinetobacter TL64","Roseateles SJ14","Aeromonas HA40","Aeromonas HA50","Aeromonas HA48","Aeromonas HA17","Chromobacterium HA01","Chromobacterium HA11","Chromobacterium HA47","Chromobacterium HA06","Chromobacterium HA72","Yersinia HA10","Yersinia HA14","Yersinia HA61","Pseudomonas HA59","Aeromonas HA07","Aeromonas HA41","Chryseobacterium TL58","Sphingomonas SJ02","Deinococcus HA29","Deinococcus HA34","Deinococcus HA42","Undibacterium HA69","Erwinia HA44","Herbaspirillum HA18","Acidovorax HA52","Herbaspirillum HA27","Rouxiella TL02","Pseudomonas SJ17","Pseudomonas HA36"],["C06","E02","F10","G12","B08","C08","D11","A06","B07","D08","C07","C06","E07","D06","F06","H08","F05","G06","A08","D05","D09","H09","H01","A01","D01","G01","B02","G05","G02","D03","E10","F04","D09","D02","E04","B02","G05","A02","C11","F06","E05","E05","F05","D02","F10","G05","E05","E01","B03","A04","C04","A03","A05","C05","F07","G08","H07","C01","B01","C01","G04","A07","E09","C03","B06","H09","H11","D05","A11","H10","F03","D07","H08","D12","H02","C05","G09","A10","F08","A08","C09","F02","G02","H06","D08","G04","G02","B06","H10","F09","E06","F07","F05","B07","A01","B01","F04","A06","A05","A12","B04","G09","G06","A07","E08","H04","A09","C10","D10","E09","H09","F01","B08","F09","C08","A02","H02","E01"],["Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","SavojÃ¤rvi","SavojÃ¤rvi","SavojÃ¤rvi","Haunisten Allas","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Haunisten Allas","SavojÃ¤rvi","Haunisten Allas","Haunisten Allas","SavojÃ¤rvi","Haunisten Allas","SavojÃ¤rvi","SavojÃ¤rvi","SavojÃ¤rvi","SavojÃ¤rvi","Haunisten Allas","Turun Linna","Turun Linna","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Haunisten Allas","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Turun Linna","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","SavojÃ¤rvi","Turun Linna","SavojÃ¤rvi","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Turun Linna","SavojÃ¤rvi","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Turun Linna","SavojÃ¤rvi","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Haunisten Allas","Turun Linna","SavojÃ¤rvi","Haunisten Allas"],["Exiguobacterium artemiae","Exiguobacterium artemiae","Exiguobacterium artemiae","Exiguobacterium artemiae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Lactococcus lactis subsp. hordniae","Serratia fonticola DSM 22080","Serratia fonticola DSM 22080","Aeromonas salmonicida","Aeromonas salmonicida","Pseudomonas germanica","Pseudomonas crudilactis","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas crudilactis","Pseudomonas germanica","Pseudomonas crudilactis","Pseudomonas crudilactis","Pseudomonas germanica","Pseudomonas migulae","Pseudomonas kielensis","Pseudomonas migulae","Pseudomonas protegens","Pseudomonas protegens","Pseudomonas glycinae","Pseudomonas glycinae","Pseudomonas glycinae","Pseudomonas glycinae","Pseudomonas glycinae","Pseudomonas glycinae","Pseudomonas koreensis","Pseudomonas glycinae","Pseudomonas glycinae","Pantoea ananatis","Pseudomonas fungipugnans","Aeromonas jandaei","Aeromonas jandaei","Aeromonas jandaei","Aeromonas jandaei","Lelliottia amnigena","Pseudomonas peli","Pseudomonas peli","Pseudomonas peli","Pseudomonas peli","Aeromonas allosaccharophila","Pseudomonas mohnii","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Pseudomonas germanica","Acinetobacter modestus","Aquitalea aquatilis","Aquitalea magnusonii","Aquitalea magnusonii","Aquitalea magnusonii","Aquitalea aquatilis","Pseudomonas graminis","Pseudomonas pratensis","Pelomonas candidula","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Chryseobacterium gambrini","Delftia acidovorans","Acinetobacter johnsonii","Roseateles depolymerans","Aeromonas finlandensis","Aeromonas finlandensis","Aeromonas tecta","Aeromonas veronii bv. veronii","Chromobacterium piscinae","Chromobacterium vaccinii","Chromobacterium vaccinii","Chromobacterium vaccinii","Chromobacterium vaccinii","Yersinia massiliensis","Yersinia intermedia","Yersinia rochesterensis","Pseudomonas grimontii","Aeromonas hydrophila","Aeromonas hydrophila","Chryseobacterium yeoncheonense","Sphingomonas azotifigens","Deinococcus aquaticus","Deinococcus aquaticus","Deinococcus aquaticus","Undibacterium aquatile","Erwinia rhapontici","Herbaspirillum aquaticum","Acidovorax radicis N35","Herbaspirillum autotrophicum","Enterobacter ludwigii","Pseudomonas kribbensis","Pseudomonas fildesensis"],[340145,340145,340145,340145,203404,203404,203404,203404,203404,203404,1358,203404,203404,203404,203404,203404,203404,203404,203404,649197,649197,645,645,2815720,2697028,2815720,2815720,2697028,2815720,2697028,2697028,2815720,78543,2762577,78543,380021,380021,1785145,1785145,1785145,1785145,1785145,1785145,198620,1785145,1785145,553,3024217,650,650,650,650,61646,592361,592361,592361,592361,656,395600,2815720,2815720,2815720,2815720,2815720,2815720,2815720,2815720,2815720,2815720,1776740,1537400,332411,332411,332411,1537400,158627,2654755,3299025,373672,373672,373672,373672,373672,373672,373672,373672,373672,80866,40214,76731,1543375,1543375,324617,197701,686831,1108595,1108595,1108595,1108595,419257,631,1604335,129847,644,644,1216946,330920,328692,328692,328692,1537398,55212,568783,999386,180195,299767,1628086,1674920],["Root (97.8%)","Root (98.1%)","Root (97.8%)","Root (99.2%)","Root (99.9%)","Root (99.2%)","Root (100%)","Root (98.8%)","Root (98.3%)","Root (99.8%)","Root (97.7%)","Root (98.4%)","Root (99.2%)","Root (98.8%)","Root (99.4%)","Root (99.3%)","Root (99%)","Root (99.4%)","Root (97%)","Root (90%)","Root (89.5%)","Root (98.3%)","Root (99.2%)","Root (95.4%)","Root (93.6%)","Root (96.3%)","Root (96.3%)","Root (93.1%)","Root (97.1%)","Root (93.8%)","Root (93.3%)","Root (96.6%)","Root (97.8%)","Root (98.9%)","Root (97.8%)","Root (98%)","Root (97.1%)","Root (97.9%)","Root (95.4%)","Root (97.6%)","Root (97.2%)","Root (97%)","Root (98.1%)","Root (98.4%)","Root (99.2%)","Root (98.7%)","Root (95.9%)","Root (94.4%)","Root (98.1%)","Root (98.1%)","Root (98.6%)","Root (98.6%)","Root (98.5%)","Root (96.6%)","Root (95.2%)","Root (95.4%)","Root (97.5%)","Root (96.2%)","Root (96.5%)","Root (96.8%)","Root (96.5%)","Root (95.6%)","Root (97.9%)","Root (97.3%)","Root (96.9%)","Root (96%)","Root (97%)","Root (97.2%)","Root (97.8%)","Root (91.8%)","Root (98.4%)","Root (98.9%)","Root (98.7%)","Root (98.5%)","Root (100%)","Root (88.1%)","Root (84.8%)","Root (95.8%)","Root (99.6%)","Root (99.5%)","Root (97.9%)","Root (96%)","Root (86.1%)","Root (99.5%)","Root (97.6%)","Root (99.4%)","Root (96.9%)","Root (99.4%)","Root (99.7%)","Root (98.9%)","Root (96.3%)","Root (95%)","Root (97.3%)","Root (95.1%)","Root (98.6%)","Root (98.6%)","Root (98.3%)","Root (97.6%)","Root (99.1%)","Root (31.2%)","Root (97.7%)","Root (98%)","Root (96.5%)","Root (97%)","Root (100%)","Root (92.5%)","Root (98.1%)","Root (96.7%)","Root (96.9%)","Root (97.4%)","Root (96.6%)","Root (92%)","Root (99.1%)","Root (95.9%)","Root (88.7%)","Root (96.8%)","Root (99.4%)","Root (100%)"],["Bacteria (97.8%)","Bacteria (98.1%)","Bacteria (97.8%)","Bacteria (99.2%)","Bacteria (99.9%)","Bacteria (99.2%)","Bacteria (100%)","Bacteria (98.8%)","Bacteria (98.3%)","Bacteria (99.8%)","Bacteria (97.7%)","Bacteria (98.4%)","Bacteria (99.2%)","Bacteria (98.8%)","Bacteria (99.4%)","Bacteria (99.3%)","Bacteria (99%)","Bacteria (99.4%)","Bacteria (97%)","Bacteria (90%)","Bacteria (89.5%)","Bacteria (98.3%)","Bacteria (99.2%)","Bacteria (95.4%)","Bacteria (93.6%)","Bacteria (96.3%)","Bacteria (96.3%)","Bacteria (93.1%)","Bacteria (97.1%)","Bacteria (93.8%)","Bacteria (93.3%)","Bacteria (96.6%)","Bacteria (97.8%)","Bacteria (98.9%)","Bacteria (97.8%)","Bacteria (98%)","Bacteria (97.1%)","Bacteria (97.9%)","Bacteria (95.4%)","Bacteria (97.6%)","Bacteria (97.2%)","Bacteria (97%)","Bacteria (98.1%)","Bacteria (98.4%)","Bacteria (99.2%)","Bacteria (98.7%)","Bacteria (95.9%)","Bacteria (94.4%)","Bacteria (98.1%)","Bacteria (98.1%)","Bacteria (98.6%)","Bacteria (98.6%)","Bacteria (98.5%)","Bacteria (96.6%)","Bacteria (95.2%)","Bacteria (95.4%)","Bacteria (97.5%)","Bacteria (96.2%)","Bacteria (96.5%)","Bacteria (96.8%)","Bacteria (96.5%)","Bacteria (95.6%)","Bacteria (97.9%)","Bacteria (97.3%)","Bacteria (96.9%)","Bacteria (96%)","Bacteria (97%)","Bacteria (97.2%)","Bacteria (97.8%)","Bacteria (91.8%)","Bacteria (98.4%)","Bacteria (98.9%)","Bacteria (98.7%)","Bacteria (98.5%)","Bacteria (100%)","Bacteria (88.1%)","Bacteria (84.8%)","Bacteria (95.8%)","Bacteria (99.6%)","Bacteria (99.5%)","Bacteria (97.9%)","Bacteria (96%)","Bacteria (86.1%)","Bacteria (99.5%)","Bacteria (97.6%)","Bacteria (99.4%)","Bacteria (96.9%)","Bacteria (99.4%)","Bacteria (99.7%)","Bacteria (98.9%)","Bacteria (96.3%)","Bacteria (95%)","Bacteria (97.3%)","Bacteria (95.1%)","Bacteria (98.6%)","Bacteria (98.6%)","Bacteria (98.3%)","Bacteria (97.6%)","Bacteria (99.1%)","unclassified_Root (31.2%)","Bacteria (97.7%)","Bacteria (98%)","Bacteria (96.5%)","Bacteria (97%)","Bacteria (100%)","Bacteria (92.5%)","Bacteria (98.1%)","Bacteria (96.7%)","Bacteria (96.9%)","Bacteria (97.4%)","Bacteria (96.6%)","Bacteria (92%)","Bacteria (99.1%)","Bacteria (95.9%)","Bacteria (88.7%)","Bacteria (96.8%)","Bacteria (99.4%)","Bacteria (100%)"],["Bacillota (97.8%)","Bacillota (98.1%)","Bacillota (97.8%)","Bacillota (99.2%)","Bacillota (99.9%)","Bacillota (99.2%)","Bacillota (100%)","Bacillota (98.8%)","Bacillota (98.3%)","Bacillota (99.8%)","Bacillota (97.7%)","Bacillota (98.4%)","Bacillota (99.2%)","Bacillota (98.8%)","Bacillota (99.4%)","Bacillota (99.3%)","Bacillota (99%)","Bacillota (99.4%)","Bacillota (97%)","Pseudomonadota (90%)","Pseudomonadota (89.5%)","Pseudomonadota (98.3%)","Pseudomonadota (99.2%)","Pseudomonadota (95.4%)","Pseudomonadota (93.6%)","Pseudomonadota (96.3%)","Pseudomonadota (96.3%)","Pseudomonadota (93.1%)","Pseudomonadota (97.1%)","Pseudomonadota (93.8%)","Pseudomonadota (93.3%)","Pseudomonadota (96.6%)","Pseudomonadota (97.8%)","Pseudomonadota (98.9%)","Pseudomonadota (97.8%)","Pseudomonadota (98%)","Pseudomonadota (97.1%)","Pseudomonadota (97.9%)","Pseudomonadota (95.4%)","Pseudomonadota (97.6%)","Pseudomonadota (97.2%)","Pseudomonadota (97%)","Pseudomonadota (98.1%)","Pseudomonadota (98.4%)","Pseudomonadota (99.2%)","Pseudomonadota (98.7%)","Pseudomonadota (95.9%)","Pseudomonadota (94.4%)","Pseudomonadota (98.1%)","Pseudomonadota (98.1%)","Pseudomonadota (98.6%)","Pseudomonadota (98.6%)","Pseudomonadota (98.5%)","Pseudomonadota (96.6%)","Pseudomonadota (95.2%)","Pseudomonadota (95.4%)","Pseudomonadota (97.5%)","Pseudomonadota (96.2%)","Pseudomonadota (93.9%)","Pseudomonadota (96.8%)","Pseudomonadota (96.5%)","Pseudomonadota (95.6%)","Pseudomonadota (97.9%)","Pseudomonadota (97.3%)","Pseudomonadota (96.9%)","Pseudomonadota (96%)","Pseudomonadota (97%)","Pseudomonadota (97.2%)","Pseudomonadota (97.8%)","Pseudomonadota (91.8%)","Pseudomonadota (98.4%)","Pseudomonadota (98.9%)","Pseudomonadota (98.7%)","Pseudomonadota (98.5%)","Pseudomonadota (100%)","Pseudomonadota (88.1%)","Pseudomonadota (84.8%)","Pseudomonadota (95.8%)","Bacteroidota (99.6%)","Bacteroidota (99.5%)","Bacteroidota (97.9%)","Bacteroidota (96%)","Bacteroidota (86.1%)","Bacteroidota (99.5%)","Bacteroidota (97.6%)","Bacteroidota (99.4%)","Bacteroidota (96.9%)","Pseudomonadota (99.4%)","Pseudomonadota (99.7%)","Pseudomonadota (98.9%)","Pseudomonadota (96.3%)","Pseudomonadota (95%)","Pseudomonadota (97.3%)","Pseudomonadota (95.1%)","Pseudomonadota (98.6%)","Pseudomonadota (98.6%)","Pseudomonadota (98.3%)","Pseudomonadota (97.6%)","Pseudomonadota (99.1%)",null,"Pseudomonadota (97.7%)","Pseudomonadota (98%)","Pseudomonadota (96.5%)","Pseudomonadota (97%)","Pseudomonadota (100%)","Bacteroidota (92.5%)","Pseudomonadota (98.1%)","Deinococcota (96.7%)","Deinococcota (96.9%)","Deinococcota (97.4%)","Pseudomonadota (96.6%)","Pseudomonadota (92%)","Pseudomonadota (99.1%)","Pseudomonadota (95.9%)","Pseudomonadota (88.7%)","Pseudomonadota (96.8%)","Pseudomonadota (98.7%)","Pseudomonadota (88.5%)"],["Bacilli (97.8%)","Bacilli (98.1%)","Bacilli (97.8%)","Bacilli (99.2%)","Bacilli (88.1%)","Bacilli (83.4%)","Bacilli (79.8%)","Bacilli (88.3%)","Bacilli (74.3%)","Bacilli (86.3%)","Bacilli (70.7%)","Bacilli (82%)","Bacilli (88.7%)","Bacilli (80.9%)","Bacilli (88.3%)","Bacilli (84.5%)","Bacilli (85.7%)","Bacilli (83.2%)","Bacilli (77.6%)","Gammaproteobacteria (90%)","Gammaproteobacteria (89.5%)","Gammaproteobacteria (98.3%)","Gammaproteobacteria (99.2%)","Gammaproteobacteria (95.4%)","Gammaproteobacteria (93.6%)","Gammaproteobacteria (96.3%)","Gammaproteobacteria (96.3%)","Gammaproteobacteria (93.1%)","Gammaproteobacteria (97.1%)","Gammaproteobacteria (93.8%)","Gammaproteobacteria (93.3%)","Gammaproteobacteria (96.6%)","Gammaproteobacteria (97.8%)","Gammaproteobacteria (98.9%)","Gammaproteobacteria (97.8%)","Gammaproteobacteria (98%)","Gammaproteobacteria (97.1%)","Gammaproteobacteria (97.9%)","Gammaproteobacteria (95.4%)","Gammaproteobacteria (97.6%)","Gammaproteobacteria (97.2%)","Gammaproteobacteria (97%)","Gammaproteobacteria (98.1%)","Gammaproteobacteria (98.4%)","Gammaproteobacteria (99.2%)","Gammaproteobacteria (98.7%)","Gammaproteobacteria (95.9%)","Gammaproteobacteria (94.4%)","Gammaproteobacteria (98.1%)","Gammaproteobacteria (98.1%)","Gammaproteobacteria (98.6%)","Gammaproteobacteria (98.6%)","Gammaproteobacteria (98.5%)","Gammaproteobacteria (96.6%)","Gammaproteobacteria (95.2%)","Gammaproteobacteria (95.4%)","Gammaproteobacteria (97.5%)","Gammaproteobacteria (96.2%)","Gammaproteobacteria (93.9%)","Gammaproteobacteria (96.8%)","Gammaproteobacteria (96.5%)","Gammaproteobacteria (95.6%)","Gammaproteobacteria (97.9%)","Gammaproteobacteria (97.3%)","Gammaproteobacteria (96.9%)","Gammaproteobacteria (96%)","Gammaproteobacteria (97%)","Gammaproteobacteria (97.2%)","Gammaproteobacteria (97.8%)","Gammaproteobacteria (91.8%)","Gammaproteobacteria (98.4%)","Gammaproteobacteria (98.9%)","Gammaproteobacteria (98.7%)","Gammaproteobacteria (98.5%)","Gammaproteobacteria (100%)","Gammaproteobacteria (88.1%)","Gammaproteobacteria (84.8%)","Gammaproteobacteria (95.8%)","Bacteroidia (99.6%)","Bacteroidia (99.5%)","Bacteroidia (97.9%)","Bacteroidia (96%)","Bacteroidia (86.1%)","Bacteroidia (99.5%)","Bacteroidia (97.6%)","Bacteroidia (99.4%)","Bacteroidia (96.9%)","Gammaproteobacteria (99.4%)","Gammaproteobacteria (99.7%)","Gammaproteobacteria (98.9%)","Gammaproteobacteria (96.3%)","Gammaproteobacteria (95%)","Gammaproteobacteria (97.3%)","Gammaproteobacteria (95.1%)","Gammaproteobacteria (98.6%)","Gammaproteobacteria (98.6%)","Gammaproteobacteria (98.3%)","Gammaproteobacteria (97.6%)","Gammaproteobacteria (99.1%)",null,"Gammaproteobacteria (97.7%)","Gammaproteobacteria (98%)","Gammaproteobacteria (96.5%)","Gammaproteobacteria (97%)","Gammaproteobacteria (100%)","Bacteroidia (92.5%)","Alphaproteobacteria (98.1%)","Deinococci (96.7%)","Deinococci (96.9%)","Deinococci (97.4%)","Gammaproteobacteria (96.6%)","Gammaproteobacteria (92%)","Gammaproteobacteria (99.1%)","Gammaproteobacteria (95.9%)","Gammaproteobacteria (88.7%)","Gammaproteobacteria (96.8%)","Gammaproteobacteria (98.7%)","Gammaproteobacteria (88.5%)"],["Exiguobacteriales (90.6%)","Exiguobacteriales (93.4%)","Exiguobacteriales (92.5%)","Exiguobacteriales (97.3%)","Lactobacillales (88.1%)","Lactobacillales (83.4%)","Lactobacillales (79.8%)","Lactobacillales (88.3%)","Lactobacillales (74.3%)","Lactobacillales (86.3%)","Lactobacillales (70.7%)","Lactobacillales (82%)","Lactobacillales (88.7%)","Lactobacillales (80.9%)","Lactobacillales (88.3%)","Lactobacillales (84.5%)","Lactobacillales (85.7%)","Lactobacillales (83.2%)","Lactobacillales (77.6%)","Enterobacterales (90%)","Enterobacterales (89.5%)","unclassified_Gammaproteobacteria (98.3%)","unclassified_Gammaproteobacteria (99.2%)","Pseudomonadales (91.2%)","Pseudomonadales (89.9%)","Pseudomonadales (93.7%)","Pseudomonadales (92%)","Pseudomonadales (85.7%)","Pseudomonadales (95.3%)","Pseudomonadales (89.6%)","Pseudomonadales (87%)","Pseudomonadales (95.1%)","Pseudomonadales (90.5%)","Pseudomonadales (94.5%)","Pseudomonadales (90.5%)","Pseudomonadales (93.7%)","Pseudomonadales (96.2%)","Pseudomonadales (94.1%)","Pseudomonadales (93.7%)","Pseudomonadales (95.7%)","Pseudomonadales (92.7%)","Pseudomonadales (90.3%)","Pseudomonadales (95.6%)","Pseudomonadales (89.8%)","Pseudomonadales (98.3%)","Pseudomonadales (97.9%)","Enterobacterales (82.7%)","Pseudomonadales (93.5%)","Enterobacterales (78%)","Enterobacterales (72.4%)","Enterobacterales (79.3%)","Enterobacterales (76.9%)","Enterobacterales (97.6%)","Pseudomonadales (95.9%)","Pseudomonadales (92.6%)","Pseudomonadales (95.4%)","Pseudomonadales (96.7%)","Enterobacterales (71.8%)","Pseudomonadales (90.1%)","Pseudomonadales (94.2%)","Pseudomonadales (94%)","Pseudomonadales (92.3%)","Pseudomonadales (95.4%)","Pseudomonadales (91.6%)","Pseudomonadales (92.6%)","Pseudomonadales (93.2%)","Pseudomonadales (94.2%)","Pseudomonadales (95.7%)","Pseudomonadales (92.5%)","unclassified_Gammaproteobacteria (91.8%)","Burkholderiales (98.4%)","Burkholderiales (98.9%)","Burkholderiales (98.7%)","Burkholderiales (98.5%)","Burkholderiales (100%)","Pseudomonadales (73.2%)","Pseudomonadales (79.8%)","Burkholderiales (95.8%)","Flavobacteriales (99.6%)","Flavobacteriales (99.5%)","Flavobacteriales (97.9%)","Flavobacteriales (96%)","Flavobacteriales (86.1%)","Flavobacteriales (99.5%)","Flavobacteriales (97.6%)","Flavobacteriales (99.4%)","Flavobacteriales (96.1%)","Burkholderiales (99.4%)","Pseudomonadales (98.8%)","Burkholderiales (98.9%)","Enterobacterales (79.2%)","Enterobacterales (76.8%)","Enterobacterales (69.9%)","Enterobacterales (70.6%)","Burkholderiales (98.6%)","Burkholderiales (98.6%)","Burkholderiales (98.3%)","Burkholderiales (97.6%)","Burkholderiales (99.1%)",null,"Enterobacterales (90.4%)","Enterobacterales (93.3%)","Pseudomonadales (72%)","unclassified_Gammaproteobacteria (97%)","Enterobacterales (66.5%)","Flavobacteriales (92.5%)","Sphingomonadales (98.1%)","Deinococcales (96.7%)","Deinococcales (96.9%)","Deinococcales (97.4%)","Burkholderiales (96.6%)","Enterobacterales (92%)","Burkholderiales (99.1%)","Burkholderiales (95.9%)","Burkholderiales (88.7%)","Enterobacterales (96.8%)","Pseudomonadales (72.3%)","Pseudomonadales (70.7%)"],["Exiguobacteriaceae (90.6%)","Exiguobacteriaceae (93.4%)","Exiguobacteriaceae (92.5%)","Exiguobacteriaceae (97.3%)","Streptococcaceae (88.1%)","Streptococcaceae (83.4%)","Streptococcaceae (79.8%)","Streptococcaceae (88.3%)","Streptococcaceae (74.3%)","Streptococcaceae (86.3%)","Streptococcaceae (70.7%)","Streptococcaceae (82%)","Streptococcaceae (88.7%)","Streptococcaceae (80.9%)","Streptococcaceae (88.3%)","Streptococcaceae (84.5%)","Streptococcaceae (85.7%)","Streptococcaceae (83.2%)","Streptococcaceae (77.6%)","Enterobacteriaceae (90%)","Enterobacteriaceae (89.5%)",null,null,"Pseudomonadaceae (91.2%)","Pseudomonadaceae (89.9%)","Pseudomonadaceae (92.8%)","Pseudomonadaceae (91.2%)","Pseudomonadaceae (85.7%)","Pseudomonadaceae (95.3%)","Pseudomonadaceae (89.6%)","Pseudomonadaceae (86.2%)","Pseudomonadaceae (95.1%)","Pseudomonadaceae (90.5%)","Pseudomonadaceae (94.5%)","Pseudomonadaceae (90.5%)","Pseudomonadaceae (93.7%)","Pseudomonadaceae (96.2%)","Pseudomonadaceae (94.1%)","Pseudomonadaceae (92.9%)","Pseudomonadaceae (95.7%)","Pseudomonadaceae (92.7%)","Pseudomonadaceae (89.4%)","Pseudomonadaceae (94.7%)","Pseudomonadaceae (89.8%)","Pseudomonadaceae (98.3%)","Pseudomonadaceae (97.9%)","Enterobacteriaceae (82.7%)","Pseudomonadaceae (91.8%)","Aeromonadaceae (78%)","Aeromonadaceae (72.4%)","Aeromonadaceae (79.3%)","Aeromonadaceae (76.9%)","Enterobacteriaceae (97.6%)","Pseudomonadaceae (95.9%)","Pseudomonadaceae (92.6%)","Pseudomonadaceae (92.4%)","Pseudomonadaceae (96.7%)","Aeromonadaceae (70.9%)","Pseudomonadaceae (83.7%)","Pseudomonadaceae (93.4%)","Pseudomonadaceae (94%)","Pseudomonadaceae (92.3%)","Pseudomonadaceae (95.4%)","Pseudomonadaceae (90.9%)","Pseudomonadaceae (91.6%)","Pseudomonadaceae (92.3%)","Pseudomonadaceae (94.2%)","Pseudomonadaceae (95.7%)","Pseudomonadaceae (92.5%)",null,"Chromobacteriaceae (98.4%)","Chromobacteriaceae (98.9%)","Chromobacteriaceae (98.7%)","Chromobacteriaceae (98.5%)","Chromobacteriaceae (100%)","Pseudomonadaceae (73.2%)","Pseudomonadaceae (77.5%)","Burkholderiaceae (95.8%)","Weeksellaceae (99.6%)","Weeksellaceae (99.5%)","Weeksellaceae (97.9%)","Weeksellaceae (96%)","Weeksellaceae (86.1%)","Weeksellaceae (99.5%)","Weeksellaceae (97.6%)","Weeksellaceae (99.4%)","Weeksellaceae (96.1%)","Burkholderiaceae (99.4%)","Moraxellaceae (98.8%)","Burkholderiaceae (98.9%)","Aeromonadaceae (79.2%)","Aeromonadaceae (76.8%)","Aeromonadaceae (69.2%)","Aeromonadaceae (69.7%)","Chromobacteriaceae (98.6%)","Chromobacteriaceae (98.6%)","Chromobacteriaceae (98.3%)","Chromobacteriaceae (97.6%)","Chromobacteriaceae (99.1%)",null,"Enterobacteriaceae (90.4%)","Enterobacteriaceae (93.3%)","Pseudomonadaceae (72%)",null,"Aeromonadaceae (66.5%)","Weeksellaceae (92.5%)","Sphingomonadaceae (98.1%)","Deinococcaceae (96.7%)","Deinococcaceae (96.9%)","Deinococcaceae (97.4%)","Burkholderiaceae (96.6%)","Enterobacteriaceae (92%)","Burkholderiaceae (99.1%)","Burkholderiaceae (95.9%)","Burkholderiaceae (88.7%)","Enterobacteriaceae (96.8%)","Pseudomonadaceae (69%)","Pseudomonadaceae (70%)"],["Exiguobacterium_A (90.6%)","Exiguobacterium_A (93.4%)","Exiguobacterium_A (92.5%)","Exiguobacterium_A (97.3%)","Lactococcus (88.1%)","Lactococcus (83.4%)","Lactococcus (79.8%)","Lactococcus (88.3%)","Lactococcus (74.3%)","Lactococcus (86.3%)","Lactococcus (70.7%)","Lactococcus (82%)","Lactococcus (88.7%)","Lactococcus (80.9%)","Lactococcus (88.3%)","Lactococcus (84.5%)","Lactococcus (85.7%)","Lactococcus (83.2%)","Lactococcus (77.6%)","unclassified_Enterobacteriaceae (90%)","unclassified_Enterobacteriaceae (89.5%)",null,null,"Pseudomonas_E (89.6%)","Pseudomonas_E (74.5%)","Pseudomonas_E (87.7%)","Pseudomonas_E (90.3%)","Pseudomonas_E (70.3%)","Pseudomonas_E (93.6%)","Pseudomonas_E (78.1%)","Pseudomonas_E (80.2%)","Pseudomonas_E (91.5%)","Pseudomonas_E (90.5%)","Pseudomonas_E (94.5%)","Pseudomonas_E (89.7%)","Pseudomonas_E (87.1%)","Pseudomonas_E (93.8%)","Pseudomonas_E (92.1%)","Pseudomonas_E (91.1%)","Pseudomonas_E (94.2%)","Pseudomonas_E (92.7%)","Pseudomonas_E (86.6%)","Pseudomonas_E (93.8%)","Pseudomonas_E (84.4%)","Pseudomonas_E (97.5%)","Pseudomonas_E (94.2%)","Pantoea (77.1%)","Pseudomonas_E (91.8%)","Aeromonas (78%)","Aeromonas (72.4%)","Aeromonas (79.3%)","Aeromonas (76.9%)","Lelliottia (60.9%)","Pseudomonas_E (93.3%)","Pseudomonas_E (91.8%)","Pseudomonas_E (89.8%)","Pseudomonas_E (94.9%)","Aeromonas (70.9%)","Pseudomonas_E (74.2%)","Pseudomonas_E (90.6%)","Pseudomonas_E (93.2%)","Pseudomonas_E (89.7%)","Pseudomonas_E (92.1%)","Pseudomonas_E (89.1%)","Pseudomonas_E (84.6%)","Pseudomonas_E (90.8%)","Pseudomonas_E (89%)","Pseudomonas_E (91.5%)","Pseudomonas_E (87%)",null,"Aquitalea (98.4%)","Aquitalea (98.9%)","Aquitalea (97.9%)","Aquitalea (98.5%)","Aquitalea (100%)","unclassified_Pseudomonadaceae (73.2%)","unclassified_Pseudomonadaceae (77.5%)","Pelomonas (64.4%)","Chryseobacterium (97.2%)","Chryseobacterium (99.5%)","Chryseobacterium (93.5%)","Chryseobacterium (92%)","Chryseobacterium (83.1%)","Chryseobacterium (99.5%)","Chryseobacterium (94.8%)","Chryseobacterium (99.4%)","Chryseobacterium (95.4%)","Comamonas (97.8%)","Acinetobacter (98.8%)","Roseateles (95.2%)","Aeromonas (79.2%)","Aeromonas (76.8%)","Aeromonas (69.2%)","Aeromonas (69.7%)","Chromobacterium (97.7%)","Chromobacterium (98.6%)","Chromobacterium (98.3%)","Chromobacterium (97.6%)","Chromobacterium (99.1%)",null,"Yersinia (86.2%)","Yersinia (85%)","Pseudomonas_E (70%)",null,"Aeromonas (66.5%)","Chryseobacterium (88.3%)","Sphingomonas (98.1%)","Deinococcus (96.7%)","Deinococcus (96.9%)","Deinococcus (97.4%)","Undibacterium (96.6%)","Erwinia (64%)","Herbaspirillum (93.8%)","Acidovorax (69.7%)","unclassified_Burkholderiaceae (88.7%)","Rouxiella (65.6%)","unclassified_Pseudomonadaceae (69%)","Pseudomonas_E (67.8%)"],["B05","D02","F08","F09","A05","B05","B07","C03","C04","C05","D04","E03","E04","F03","G03","G05","H02","H03","H04","E10","G09","G11","H01","B12","E12","G12","A01","A03","C01","E01","E06","H01","A06","B01","G01","B02","H10","A02","A09","B11","D04","F10","G10","H09","B10","D09","G02","F12","A03","C03","E03","H02","C02","E02","F04","F05","G04","C01","C12","D12","A02","B04","B06","D01","D03","D06","E07","F02","H06","E09","B03","B06","C07","D09","G02","D10","C06","H11","A07","D06","D07","E02","F02","F05","G06","H03","C09","A05","G06","F11","C05","C06","E04","H05","A01","B01","G03","H04","C08","B09","D03","G07","E05","G05","H06","B02","C11","C08","D08","E07","H07","F01","E06","F07","F06","H12","A10","E01"],["batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250606","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250620","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250606","batch20250606","batch20250606","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250606","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250620","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606","batch20250606"],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>strainID<\/th>\n      <th>google_drive_id<\/th>\n      <th>taxa_cluster<\/th>\n      <th>tree_name<\/th>\n      <th>bacteria_plate_well<\/th>\n      <th>location<\/th>\n      <th>ncbi_best_species<\/th>\n      <th>ncbi_taxid<\/th>\n      <th>rootrank<\/th>\n      <th>domain<\/th>\n      <th>phylum<\/th>\n      <th>class<\/th>\n      <th>order<\/th>\n      <th>family<\/th>\n      <th>genus<\/th>\n      <th>sanger_plate_well<\/th>\n      <th>sanger_batch<\/th>\n      <th>sangeranalyseR_success<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"ft","pageLength":-1,"scrollX":true,"fixedColumns":true,"columnDefs":[{"className":"dt-right","targets":[2,7]},{"name":"strainID","targets":0},{"name":"google_drive_id","targets":1},{"name":"taxa_cluster","targets":2},{"name":"tree_name","targets":3},{"name":"bacteria_plate_well","targets":4},{"name":"location","targets":5},{"name":"ncbi_best_species","targets":6},{"name":"ncbi_taxid","targets":7},{"name":"rootrank","targets":8},{"name":"domain","targets":9},{"name":"phylum","targets":10},{"name":"class","targets":11},{"name":"order","targets":12},{"name":"family","targets":13},{"name":"genus","targets":14},{"name":"sanger_plate_well","targets":15},{"name":"sanger_batch","targets":16},{"name":"sangeranalyseR_success","targets":17}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[-1,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="phylogenetic-tree-plotting" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Phylogenetic tree plotting</h1>
<p>Goal is to produce a tree that has for tip labels the Genus assigned by IDTAXA + the strain identifier, with different internal nodes labeled by coarser taxonomic grouping. The clusters assigned by <code>clusterizer</code> will be represented by different colors at the tips.</p>
<p>First lets look at the tree with just the identifiers on the leaves</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>) <span class="sc">+</span> </span>
<span id="cb3-2"><a href="#cb3-2"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb3-3"><a href="#cb3-3"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="fu">geom_treescale</span>(<span class="at">offset=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_tree_summary_table_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Maximum likelihood phylogenetic tree inferred from an alignment of the 16S rRNA gene amplified by Sanger sequencing from the duckweed SynCom. Scale bar at bottom represents 0.1 substitutions per site.
</figcaption>
</figure>
</div>
<p>We can also look at the internal node numbering by rendering the tree as a dataframe</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># the second tibble call is so quarto renders it properly</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>syncom16s_seqs_mltree_nwk <span class="sc">%&gt;%</span> treeio<span class="sc">::</span><span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">tibble</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["parent"],"name":[1],"type":["int"],"align":["right"]},{"label":["node"],"name":[2],"type":["int"],"align":["right"]},{"label":["branch.length"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["label"],"name":[4],"type":["chr"],"align":["left"]},{"label":["support"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"122","2":"1","3":"0.0000003023","4":"HA29","5":"NA"},{"1":"123","2":"2","3":"0.0000002833","4":"HA34","5":"NA"},{"1":"123","2":"3","3":"0.0000002881","4":"HA42","5":"NA"},{"1":"125","2":"4","3":"0.0000002980","4":"TL39","5":"NA"},{"1":"126","2":"5","3":"0.0000003030","4":"TL62","5":"NA"},{"1":"128","2":"6","3":"0.0000002454","4":"TL19","5":"NA"},{"1":"131","2":"7","3":"0.0000002680","4":"TL25","5":"NA"},{"1":"131","2":"8","3":"0.0010944810","4":"TL12","5":"NA"},{"1":"133","2":"9","3":"0.0000002877","4":"TL35","5":"NA"},{"1":"134","2":"10","3":"0.0000002409","4":"TL10","5":"NA"},{"1":"135","2":"11","3":"0.0000002823","4":"TL46","5":"NA"},{"1":"136","2":"12","3":"0.0000002755","4":"TL33","5":"NA"},{"1":"137","2":"13","3":"0.0000002930","4":"TL32","5":"NA"},{"1":"138","2":"14","3":"0.0000002915","4":"TL24","5":"NA"},{"1":"138","2":"15","3":"0.0000002949","4":"TL26","5":"NA"},{"1":"132","2":"16","3":"0.0000000238","4":"TL20","5":"NA"},{"1":"129","2":"17","3":"0.0000000558","4":"TL45","5":"NA"},{"1":"127","2":"18","3":"0.0000003023","4":"TL53","5":"NA"},{"1":"140","2":"19","3":"0.0000003028","4":"HA37","5":"NA"},{"1":"141","2":"20","3":"0.0000003016","4":"HA63","5":"NA"},{"1":"141","2":"21","3":"0.0000003034","4":"HA53","5":"NA"},{"1":"139","2":"22","3":"0.0011223550","4":"HA25","5":"NA"},{"1":"142","2":"23","3":"0.3473705000","4":"SJ02","5":"NA"},{"1":"146","2":"24","3":"0.0085621920","4":"HA18","5":"NA"},{"1":"147","2":"25","3":"0.0173163000","4":"HA27","5":"NA"},{"1":"147","2":"26","3":"0.0497882600","4":"HA69","5":"NA"},{"1":"149","2":"27","3":"0.0247243300","4":"SJ03","5":"NA"},{"1":"149","2":"28","3":"0.0400582200","4":"SJ14","5":"NA"},{"1":"150","2":"29","3":"0.0475276200","4":"HA16","5":"NA"},{"1":"150","2":"30","3":"0.0546802100","4":"HA52","5":"NA"},{"1":"153","2":"31","3":"0.0000002998","4":"HA32","5":"NA"},{"1":"154","2":"32","3":"0.0000002797","4":"HA35","5":"NA"},{"1":"154","2":"33","3":"0.0000002988","4":"HA68","5":"NA"},{"1":"155","2":"34","3":"0.0000003020","4":"HA46","5":"NA"},{"1":"155","2":"35","3":"0.0000003021","4":"HA65","5":"NA"},{"1":"157","2":"36","3":"0.0000003037","4":"HA01","5":"NA"},{"1":"157","2":"37","3":"0.0000003037","4":"HA11","5":"NA"},{"1":"158","2":"38","3":"0.0000003031","4":"HA72","5":"NA"},{"1":"159","2":"39","3":"0.0000003028","4":"HA06","5":"NA"},{"1":"159","2":"40","3":"0.0000003037","4":"HA47","5":"NA"},{"1":"161","2":"41","3":"0.0246131700","4":"HA82","5":"NA"},{"1":"161","2":"42","3":"0.0300392800","4":"TL64","5":"NA"},{"1":"164","2":"43","3":"0.0110978500","4":"HA44","5":"NA"},{"1":"169","2":"44","3":"0.0000003009","4":"HA61","5":"NA"},{"1":"169","2":"45","3":"0.0028810270","4":"HA14","5":"NA"},{"1":"168","2":"46","3":"0.0071653950","4":"HA10","5":"NA"},{"1":"167","2":"47","3":"0.3544444000","4":"TL02","5":"NA"},{"1":"170","2":"48","3":"0.0000002881","4":"SJ20","5":"NA"},{"1":"170","2":"49","3":"0.0000002893","4":"SJ08","5":"NA"},{"1":"171","2":"50","3":"0.0318726100","4":"TL09","5":"NA"},{"1":"171","2":"51","3":"0.0330256500","4":"TL38","5":"NA"},{"1":"173","2":"52","3":"0.0000003019","4":"HA13","5":"NA"},{"1":"173","2":"53","3":"0.0000003022","4":"HA04","5":"NA"},{"1":"175","2":"54","3":"0.0000003023","4":"HA03","5":"NA"},{"1":"175","2":"55","3":"0.0000003025","4":"HA23","5":"NA"},{"1":"177","2":"56","3":"0.0034335010","4":"HA17","5":"NA"},{"1":"178","2":"57","3":"0.0057523240","4":"HA48","5":"NA"},{"1":"179","2":"58","3":"0.0000002880","4":"HA40","5":"NA"},{"1":"179","2":"59","3":"0.0000002893","4":"HA50","5":"NA"},{"1":"180","2":"60","3":"0.0117376200","4":"HA21","5":"NA"},{"1":"182","2":"61","3":"0.0000003015","4":"SJ18","5":"NA"},{"1":"182","2":"62","3":"0.0000003035","4":"HA64","5":"NA"},{"1":"183","2":"63","3":"0.0000003027","4":"HA07","5":"NA"},{"1":"183","2":"64","3":"0.0012321170","4":"HA41","5":"NA"},{"1":"187","2":"65","3":"0.0000002951","4":"TL61","5":"NA"},{"1":"187","2":"66","3":"0.0011357230","4":"TL54","5":"NA"},{"1":"186","2":"67","3":"0.0000003034","4":"TL47","5":"NA"},{"1":"185","2":"68","3":"0.0000003037","4":"TL23","5":"NA"},{"1":"188","2":"69","3":"0.0061230540","4":"TL29","5":"NA"},{"1":"190","2":"70","3":"0.0130998200","4":"HA59","5":"NA"},{"1":"194","2":"71","3":"0.0233276100","4":"TL03","5":"NA"},{"1":"194","2":"72","3":"0.0377324500","4":"HA36","5":"NA"},{"1":"193","2":"73","3":"0.0012215510","4":"SJ17","5":"NA"},{"1":"195","2":"74","3":"0.0000003036","4":"SJ16","5":"NA"},{"1":"195","2":"75","3":"0.0000003037","4":"HA12","5":"NA"},{"1":"191","2":"76","3":"0.0063136800","4":"TL06","5":"NA"},{"1":"197","2":"77","3":"0.0000002870","4":"TL37","5":"NA"},{"1":"197","2":"78","3":"0.0011397930","4":"TL34","5":"NA"},{"1":"201","2":"79","3":"0.0000002871","4":"TL30","5":"NA"},{"1":"203","2":"80","3":"0.0000003003","4":"TL52","5":"NA"},{"1":"203","2":"81","3":"0.0009047834","4":"TL42","5":"NA"},{"1":"202","2":"82","3":"0.0000002943","4":"TL05","5":"NA"},{"1":"204","2":"83","3":"0.0122390800","4":"TL55","5":"NA"},{"1":"204","2":"84","3":"0.0157753600","4":"SJ05","5":"NA"},{"1":"206","2":"85","3":"0.0000002709","4":"TL04","5":"NA"},{"1":"206","2":"86","3":"0.0000002982","4":"TL50","5":"NA"},{"1":"208","2":"87","3":"0.0000000395","4":"TL65","5":"NA"},{"1":"209","2":"88","3":"0.0000000217","4":"TL31","5":"NA"},{"1":"210","2":"89","3":"0.0000000217","4":"TL07","5":"NA"},{"1":"211","2":"90","3":"0.0000000217","4":"TL16","5":"NA"},{"1":"212","2":"91","3":"0.0000000216","4":"TL51","5":"NA"},{"1":"214","2":"92","3":"0.0000000284","4":"TL44","5":"NA"},{"1":"215","2":"93","3":"0.0011435490","4":"TL11","5":"NA"},{"1":"216","2":"94","3":"0.0000002601","4":"TL14","5":"NA"},{"1":"218","2":"95","3":"0.0000003010","4":"TL41","5":"NA"},{"1":"218","2":"96","3":"0.0023091300","4":"TL22","5":"NA"},{"1":"217","2":"97","3":"0.0000002934","4":"TL63","5":"NA"},{"1":"213","2":"98","3":"0.0000000217","4":"TL01","5":"NA"},{"1":"207","2":"99","3":"0.0000006015","4":"TL18","5":"NA"},{"1":"219","2":"100","3":"0.0047454960","4":"SJ07","5":"NA"},{"1":"221","2":"101","3":"0.0011330150","4":"SJ23","5":"NA"},{"1":"222","2":"102","3":"0.0000003004","4":"HA81","5":"NA"},{"1":"226","2":"103","3":"0.0000002869","4":"SJ13","5":"NA"},{"1":"226","2":"104","3":"0.0000002895","4":"SJ12","5":"NA"},{"1":"225","2":"105","3":"0.0000003036","4":"HA02","5":"NA"},{"1":"224","2":"106","3":"0.0000003037","4":"HA39","5":"NA"},{"1":"223","2":"107","3":"0.0011499230","4":"HA30","5":"NA"},{"1":"220","2":"108","3":"0.0032442310","4":"SJ10","5":"NA"},{"1":"228","2":"109","3":"0.0000002952","4":"HA57","5":"NA"},{"1":"230","2":"110","3":"0.0000003017","4":"HA08","5":"NA"},{"1":"230","2":"111","3":"0.0000003035","4":"HA67","5":"NA"},{"1":"229","2":"112","3":"0.0000003025","4":"HA51","5":"NA"},{"1":"232","2":"113","3":"0.0000002641","4":"HA33","5":"NA"},{"1":"233","2":"114","3":"0.0000002958","4":"HA45","5":"NA"},{"1":"233","2":"115","3":"0.0137098400","4":"HA56","5":"NA"},{"1":"235","2":"116","3":"0.0000002792","4":"HA28","5":"NA"},{"1":"235","2":"117","3":"0.0000002886","4":"HA80","5":"NA"},{"1":"234","2":"118","3":"0.0444020500","4":"TL58","5":"NA"},{"1":"119","2":"119","3":"NA","4":"NA","5":"NA"},{"1":"119","2":"120","3":"0.0295668500","4":"NA","5":"1.00"},{"1":"120","2":"121","3":"0.1344812000","4":"NA","5":"1.00"},{"1":"121","2":"122","3":"0.7207153000","4":"NA","5":"1.00"},{"1":"122","2":"123","3":"0.0000000634","4":"NA","5":"0.33"},{"1":"121","2":"124","3":"0.2050687000","4":"NA","5":"1.00"},{"1":"124","2":"125","3":"0.2496145000","4":"NA","5":"1.00"},{"1":"125","2":"126","3":"0.0000000329","4":"NA","5":"0.33"},{"1":"126","2":"127","3":"0.0002296142","4":"NA","5":"0.33"},{"1":"127","2":"128","3":"0.0003592300","4":"NA","5":"0.34"},{"1":"128","2":"129","3":"0.0004506589","4":"NA","5":"0.33"},{"1":"129","2":"130","3":"0.0000154758","4":"NA","5":"0.33"},{"1":"130","2":"131","3":"0.0010917030","4":"NA","5":"1.00"},{"1":"130","2":"132","3":"0.0000095850","4":"NA","5":"0.33"},{"1":"132","2":"133","3":"0.0000131253","4":"NA","5":"0.33"},{"1":"133","2":"134","3":"0.0000097200","4":"NA","5":"0.33"},{"1":"134","2":"135","3":"0.0000097822","4":"NA","5":"0.33"},{"1":"135","2":"136","3":"0.0000103078","4":"NA","5":"0.33"},{"1":"136","2":"137","3":"0.0000002929","4":"NA","5":"0.33"},{"1":"137","2":"138","3":"0.0000002585","4":"NA","5":"0.33"},{"1":"124","2":"139","3":"0.2406384000","4":"NA","5":"1.00"},{"1":"139","2":"140","3":"0.0000016372","4":"NA","5":"0.33"},{"1":"140","2":"141","3":"0.0000003028","4":"NA","5":"0.33"},{"1":"120","2":"142","3":"0.0340158900","4":"NA","5":"0.56"},{"1":"142","2":"143","3":"0.1863477000","4":"NA","5":"1.00"},{"1":"143","2":"144","3":"0.1694784000","4":"NA","5":"1.00"},{"1":"144","2":"145","3":"0.0827489600","4":"NA","5":"1.00"},{"1":"145","2":"146","3":"0.0964129500","4":"NA","5":"1.00"},{"1":"146","2":"147","3":"0.0140819500","4":"NA","5":"0.96"},{"1":"145","2":"148","3":"0.0718772200","4":"NA","5":"1.00"},{"1":"148","2":"149","3":"0.0312316500","4":"NA","5":"1.00"},{"1":"148","2":"150","3":"0.0623829900","4":"NA","5":"1.00"},{"1":"144","2":"151","3":"0.0531105000","4":"NA","5":"1.00"},{"1":"151","2":"152","3":"0.0292747500","4":"NA","5":"1.00"},{"1":"152","2":"153","3":"0.0054894770","4":"NA","5":"0.97"},{"1":"153","2":"154","3":"0.0000000275","4":"NA","5":"0.33"},{"1":"152","2":"155","3":"0.0021253260","4":"NA","5":"0.61"},{"1":"151","2":"156","3":"0.0602668400","4":"NA","5":"1.00"},{"1":"156","2":"157","3":"0.0000002407","4":"NA","5":"0.33"},{"1":"156","2":"158","3":"0.0000002870","4":"NA","5":"0.33"},{"1":"158","2":"159","3":"0.0000003031","4":"NA","5":"0.33"},{"1":"143","2":"160","3":"0.0523796300","4":"NA","5":"1.00"},{"1":"160","2":"161","3":"0.1673914000","4":"NA","5":"1.00"},{"1":"160","2":"162","3":"0.0274929300","4":"NA","5":"0.52"},{"1":"162","2":"163","3":"0.1501122000","4":"NA","5":"1.00"},{"1":"163","2":"164","3":"0.1195508000","4":"NA","5":"1.00"},{"1":"164","2":"165","3":"0.0225670100","4":"NA","5":"1.00"},{"1":"165","2":"166","3":"0.0153033000","4":"NA","5":"0.99"},{"1":"166","2":"167","3":"0.0242769400","4":"NA","5":"0.70"},{"1":"167","2":"168","3":"0.0308099400","4":"NA","5":"0.74"},{"1":"168","2":"169","3":"0.0030935450","4":"NA","5":"0.37"},{"1":"166","2":"170","3":"0.0168779900","4":"NA","5":"1.00"},{"1":"165","2":"171","3":"0.0248347800","4":"NA","5":"1.00"},{"1":"163","2":"172","3":"0.0658691400","4":"NA","5":"1.00"},{"1":"172","2":"173","3":"0.0000003036","4":"NA","5":"0.33"},{"1":"172","2":"174","3":"0.0011466660","4":"NA","5":"0.67"},{"1":"174","2":"175","3":"0.0000003025","4":"NA","5":"0.33"},{"1":"174","2":"176","3":"0.0084532630","4":"NA","5":"1.00"},{"1":"176","2":"177","3":"0.0035324020","4":"NA","5":"0.85"},{"1":"177","2":"178","3":"0.0048174210","4":"NA","5":"0.55"},{"1":"178","2":"179","3":"0.0047444900","4":"NA","5":"0.98"},{"1":"176","2":"180","3":"0.0081387160","4":"NA","5":"1.00"},{"1":"180","2":"181","3":"0.0034620750","4":"NA","5":"0.98"},{"1":"181","2":"182","3":"0.0083832770","4":"NA","5":"1.00"},{"1":"181","2":"183","3":"0.0124521300","4":"NA","5":"1.00"},{"1":"162","2":"184","3":"0.0904250700","4":"NA","5":"1.00"},{"1":"184","2":"185","3":"0.0296197400","4":"NA","5":"1.00"},{"1":"185","2":"186","3":"0.0000000612","4":"NA","5":"0.33"},{"1":"186","2":"187","3":"0.0000003036","4":"NA","5":"0.33"},{"1":"184","2":"188","3":"0.0246482600","4":"NA","5":"1.00"},{"1":"188","2":"189","3":"0.0019484520","4":"NA","5":"0.50"},{"1":"189","2":"190","3":"0.0088157110","4":"NA","5":"1.00"},{"1":"190","2":"191","3":"0.0124073300","4":"NA","5":"1.00"},{"1":"191","2":"192","3":"0.0120418500","4":"NA","5":"0.41"},{"1":"192","2":"193","3":"0.1989057000","4":"NA","5":"1.00"},{"1":"193","2":"194","3":"0.0306488800","4":"NA","5":"1.00"},{"1":"192","2":"195","3":"0.0088394530","4":"NA","5":"0.37"},{"1":"189","2":"196","3":"0.0035110010","4":"NA","5":"1.00"},{"1":"196","2":"197","3":"0.0000002645","4":"NA","5":"0.33"},{"1":"196","2":"198","3":"0.0214573000","4":"NA","5":"1.00"},{"1":"198","2":"199","3":"0.0023250550","4":"NA","5":"0.87"},{"1":"199","2":"200","3":"0.0113730300","4":"NA","5":"1.00"},{"1":"200","2":"201","3":"0.0022491180","4":"NA","5":"0.58"},{"1":"201","2":"202","3":"0.0000001329","4":"NA","5":"0.33"},{"1":"202","2":"203","3":"0.0002201480","4":"NA","5":"0.33"},{"1":"200","2":"204","3":"0.0184862600","4":"NA","5":"1.00"},{"1":"199","2":"205","3":"0.0015586830","4":"NA","5":"0.92"},{"1":"205","2":"206","3":"0.0000002795","4":"NA","5":"0.33"},{"1":"205","2":"207","3":"0.0000000293","4":"NA","5":"0.33"},{"1":"207","2":"208","3":"0.0000002942","4":"NA","5":"0.33"},{"1":"208","2":"209","3":"0.0000002718","4":"NA","5":"0.33"},{"1":"209","2":"210","3":"0.0000003020","4":"NA","5":"0.33"},{"1":"210","2":"211","3":"0.0000002911","4":"NA","5":"0.33"},{"1":"211","2":"212","3":"0.0000002487","4":"NA","5":"0.33"},{"1":"212","2":"213","3":"0.0000002981","4":"NA","5":"0.33"},{"1":"213","2":"214","3":"0.0000002997","4":"NA","5":"0.33"},{"1":"214","2":"215","3":"0.0000002797","4":"NA","5":"0.33"},{"1":"215","2":"216","3":"0.0000000340","4":"NA","5":"0.33"},{"1":"216","2":"217","3":"0.0030062260","4":"NA","5":"1.00"},{"1":"217","2":"218","3":"0.0000000279","4":"NA","5":"0.33"},{"1":"198","2":"219","3":"0.0132719800","4":"NA","5":"1.00"},{"1":"219","2":"220","3":"0.0033889080","4":"NA","5":"0.99"},{"1":"220","2":"221","3":"0.0000001793","4":"NA","5":"0.33"},{"1":"221","2":"222","3":"0.0000003014","4":"NA","5":"0.33"},{"1":"222","2":"223","3":"0.0012263900","4":"NA","5":"1.00"},{"1":"223","2":"224","3":"0.0000000804","4":"NA","5":"0.33"},{"1":"224","2":"225","3":"0.0012282290","4":"NA","5":"1.00"},{"1":"225","2":"226","3":"0.0000000660","4":"NA","5":"0.33"},{"1":"119","2":"227","3":"0.9703983000","4":"NA","5":"1.00"},{"1":"227","2":"228","3":"0.0000002575","4":"NA","5":"0.33"},{"1":"228","2":"229","3":"0.0000003025","4":"NA","5":"0.33"},{"1":"229","2":"230","3":"0.0000003034","4":"NA","5":"0.33"},{"1":"227","2":"231","3":"0.0012570360","4":"NA","5":"0.48"},{"1":"231","2":"232","3":"0.0000000308","4":"NA","5":"0.33"},{"1":"232","2":"233","3":"0.0000002529","4":"NA","5":"0.33"},{"1":"231","2":"234","3":"0.0000002507","4":"NA","5":"0.33"},{"1":"234","2":"235","3":"0.0000002552","4":"NA","5":"0.33"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>First we need to get our naming scheme set up</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>newtipnames <span class="ot">&lt;-</span> final_table <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(genus, <span class="st">"(^[:alpha:]*).*"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(genus) <span class="sc">|</span> genus <span class="sc">==</span> <span class="st">"unclassified"</span> <span class="sc">~</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(ncbi_best_species, <span class="st">"^[:alpha:]*"</span>),</span>
<span id="cb5-4"><a href="#cb5-4"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> genus)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="fu">mutate</span>(<span class="at">newname =</span> <span class="fu">paste</span>(genus, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="fu">mutate</span>(<span class="at">taxa_cluster =</span> <span class="fu">paste0</span>(<span class="st">"c"</span>, taxa_cluster)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-7"><a href="#cb5-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">id=</span>strainID, newname, taxa_cluster, genus, order, family)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>First we need to generate a color palette that lets us distinguish the different colors</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="fu">set.seed</span>(<span class="dv">341234</span>)</span>
<span id="cb6-3"><a href="#cb6-3"></a>mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">max</span>(final_table<span class="sc">$</span>taxa_cluster), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we need to identify the internal nodes that we want to highlight. We can do this using the <code>MRCA</code> function from ggtree</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 184</code></pre>
</div>
</div>
<p>Code to plot the tree</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>offset <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a>p2 <span class="ot">&lt;-</span> p1 <span class="sc">%&lt;+%</span> newtipnames <span class="sc">+</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="fu">geom_tiplab</span>(<span class="fu">aes</span>(<span class="at">label =</span> newname), <span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> taxa_cluster), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal, <span class="at">guide=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb9-8"><a href="#cb9-8"></a>  <span class="co"># Pseudomonadaceae </span></span>
<span id="cb9-9"><a href="#cb9-9"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>), </span>
<span id="cb9-10"><a href="#cb9-10"></a>                <span class="at">label=</span><span class="st">"(Î³) Pseudomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-11"><a href="#cb9-11"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-12"><a href="#cb9-12"></a>  <span class="co"># Aeromonadaceae</span></span>
<span id="cb9-13"><a href="#cb9-13"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA41"</span>, <span class="st">"HA04"</span>), </span>
<span id="cb9-14"><a href="#cb9-14"></a>                <span class="at">label=</span><span class="st">"(Î³) Aeromonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-15"><a href="#cb9-15"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-16"><a href="#cb9-16"></a>  <span class="co"># Enterobacteriaceae</span></span>
<span id="cb9-17"><a href="#cb9-17"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA14"</span>, <span class="st">"HA44"</span>), </span>
<span id="cb9-18"><a href="#cb9-18"></a>                <span class="at">label=</span><span class="st">"(Î³) Enterobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-19"><a href="#cb9-19"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-20"><a href="#cb9-20"></a>  <span class="co"># Moraxellaceae</span></span>
<span id="cb9-21"><a href="#cb9-21"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL64"</span>, <span class="st">"HA82"</span>), </span>
<span id="cb9-22"><a href="#cb9-22"></a>                <span class="at">label=</span><span class="st">"(Î³) Moraxellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-23"><a href="#cb9-23"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-24"><a href="#cb9-24"></a>  <span class="co"># Chromobacteriaceae</span></span>
<span id="cb9-25"><a href="#cb9-25"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA47"</span>, <span class="st">"HA65"</span>), </span>
<span id="cb9-26"><a href="#cb9-26"></a>                <span class="at">label=</span><span class="st">"(Î³) Chromobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-27"><a href="#cb9-27"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-28"><a href="#cb9-28"></a>  <span class="co"># Burkholderiaceae</span></span>
<span id="cb9-29"><a href="#cb9-29"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA52"</span>, <span class="st">"HA18"</span>), </span>
<span id="cb9-30"><a href="#cb9-30"></a>                <span class="at">label=</span><span class="st">"(Î³) Burkholderiaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-31"><a href="#cb9-31"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-32"><a href="#cb9-32"></a>  <span class="co"># Sphingomonadaceae</span></span>
<span id="cb9-33"><a href="#cb9-33"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="dv">23</span>, </span>
<span id="cb9-34"><a href="#cb9-34"></a>                <span class="at">label=</span><span class="st">"(Î±) Sphingomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-35"><a href="#cb9-35"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-36"><a href="#cb9-36"></a>  <span class="co"># Streptococcaceae</span></span>
<span id="cb9-37"><a href="#cb9-37"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL12"</span>, <span class="st">"TL39"</span>), </span>
<span id="cb9-38"><a href="#cb9-38"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Streptococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-39"><a href="#cb9-39"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-40"><a href="#cb9-40"></a>  <span class="co"># Exiguobacteriaceae</span></span>
<span id="cb9-41"><a href="#cb9-41"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA63"</span>, <span class="st">"HA25"</span>), </span>
<span id="cb9-42"><a href="#cb9-42"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Exiguobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-43"><a href="#cb9-43"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb9-44"><a href="#cb9-44"></a>  <span class="co"># Deinococcaceae</span></span>
<span id="cb9-45"><a href="#cb9-45"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA42"</span>, <span class="st">"HA29"</span>), </span>
<span id="cb9-46"><a href="#cb9-46"></a>                <span class="at">label=</span><span class="st">"(Deinococci) Deinococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-47"><a href="#cb9-47"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.2</span>) <span class="sc">+</span></span>
<span id="cb9-48"><a href="#cb9-48"></a>  <span class="co"># Weeksellaceae</span></span>
<span id="cb9-49"><a href="#cb9-49"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA56"</span>, <span class="st">"HA57"</span>), </span>
<span id="cb9-50"><a href="#cb9-50"></a>                <span class="at">label=</span><span class="st">"(Bacteroidia) Weeksellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb9-51"><a href="#cb9-51"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.4</span>) <span class="sc">+</span></span>
<span id="cb9-52"><a href="#cb9-52"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-02" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_tree_summary_table_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: As in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>, but isolates have been labeled with their inferred genus (text) as well as their broader taxonomic class (right, vertical lines). Colors represent sequences clusters based on a 99% identity threshold (i.e., nodes of the same color have 16S rRNA gene sequences that are &gt; 99% similar)
</figcaption>
</figure>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb10" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb10-1"><a href="#cb10-1"></a><span class="co">---</span></span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="an">title:</span><span class="co"> "Plotting the phylogenetic tree and associated taxonmic information"</span></span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="an">subtitle:</span><span class="co"> ""</span></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="an">abstract:</span><span class="co"> "Display information from the previous steps in a nice format"</span></span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="co">---</span></span>
<span id="cb10-8"><a href="#cb10-8"></a></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="fu"># Setup</span></span>
<span id="cb10-10"><a href="#cb10-10"></a></span>
<span id="cb10-11"><a href="#cb10-11"></a>Loads required libraries</span>
<span id="cb10-12"><a href="#cb10-12"></a></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="in">```{r}</span></span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="co">#| output: false</span></span>
<span id="cb10-17"><a href="#cb10-17"></a><span class="fu">options</span>(<span class="at">repos =</span> <span class="fu">c</span>(<span class="at">CRAN =</span> <span class="st">"https://packagemanager.posit.co/cran/__linux__/jammy/latest"</span>))</span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb10-19"><a href="#cb10-19"></a><span class="fu">library</span>(here)</span>
<span id="cb10-20"><a href="#cb10-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb10-22"><a href="#cb10-22"></a><span class="fu">library</span>(treeio)</span>
<span id="cb10-23"><a href="#cb10-23"></a><span class="fu">library</span>(DT)</span>
<span id="cb10-24"><a href="#cb10-24"></a><span class="in">```</span></span>
<span id="cb10-25"><a href="#cb10-25"></a></span>
<span id="cb10-26"><a href="#cb10-26"></a><span class="fu"># Read and format</span></span>
<span id="cb10-27"><a href="#cb10-27"></a></span>
<span id="cb10-28"><a href="#cb10-28"></a>Read phylogenetic tree and associated data table from the previous step</span>
<span id="cb10-29"><a href="#cb10-29"></a></span>
<span id="cb10-32"><a href="#cb10-32"></a><span class="in">```{r}</span></span>
<span id="cb10-33"><a href="#cb10-33"></a><span class="co">#| echo: true</span></span>
<span id="cb10-34"><a href="#cb10-34"></a><span class="co">#| output: false</span></span>
<span id="cb10-35"><a href="#cb10-35"></a>syncom16s_seqs_mltree_nwk <span class="ot">&lt;-</span> treeio<span class="sc">::</span><span class="fu">read.newick</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.nwk"</span>),</span>
<span id="cb10-36"><a href="#cb10-36"></a>                            <span class="at">node.label=</span><span class="st">"support"</span>)</span>
<span id="cb10-37"><a href="#cb10-37"></a></span>
<span id="cb10-38"><a href="#cb10-38"></a>final_table <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strain_table_info.tsv"</span>))</span>
<span id="cb10-39"><a href="#cb10-39"></a><span class="in">```</span></span>
<span id="cb10-40"><a href="#cb10-40"></a></span>
<span id="cb10-41"><a href="#cb10-41"></a><span class="fu"># Summary table</span></span>
<span id="cb10-42"><a href="#cb10-42"></a></span>
<span id="cb10-43"><a href="#cb10-43"></a>Below is a compiled table with taxonomic info, isolation source, and mapping of new strainID to the original <span class="in">`google_drive_id.`</span> Taxa cluster refers to the 99% sequence identity cluster that each strain is a member of. Table is sorted by <span class="in">`taxa_cluster.`</span> Columns <span class="in">`domain`</span> through <span class="in">`genus`</span> are the taxonomic output from the IDTAXA algorithm using the <span class="co">[</span><span class="ot">Genome Taxonomy Database (GTDB)</span><span class="co">](https://gtdb.ecogenomic.org/)</span> 16S rRNA reference training set. The percentage following the assigned taxa classification represents the classification confidence/probability.</span>
<span id="cb10-44"><a href="#cb10-44"></a></span>
<span id="cb10-47"><a href="#cb10-47"></a><span class="in">```{r}</span></span>
<span id="cb10-48"><a href="#cb10-48"></a><span class="co">#| echo: false</span></span>
<span id="cb10-49"><a href="#cb10-49"></a><span class="co">#| output: true</span></span>
<span id="cb10-50"><a href="#cb10-50"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb10-51"><a href="#cb10-51"></a>  final_table,</span>
<span id="cb10-52"><a href="#cb10-52"></a>  <span class="at">class =</span> <span class="st">'cell-border stripe'</span>,</span>
<span id="cb10-53"><a href="#cb10-53"></a>  <span class="at">rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-54"><a href="#cb10-54"></a>  <span class="at">extensions =</span> <span class="st">'FixedColumns'</span>,</span>
<span id="cb10-55"><a href="#cb10-55"></a>  <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb10-56"><a href="#cb10-56"></a>    <span class="at">dom =</span> <span class="st">"ft"</span>,</span>
<span id="cb10-57"><a href="#cb10-57"></a>    <span class="at">pageLength =</span> <span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb10-58"><a href="#cb10-58"></a>    <span class="at">scrollX =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-59"><a href="#cb10-59"></a>    <span class="co">#scrollY = TRUE,</span></span>
<span id="cb10-60"><a href="#cb10-60"></a>    <span class="at">fixedColumns =</span> <span class="cn">TRUE</span></span>
<span id="cb10-61"><a href="#cb10-61"></a>  )</span>
<span id="cb10-62"><a href="#cb10-62"></a>)</span>
<span id="cb10-63"><a href="#cb10-63"></a><span class="in">```</span></span>
<span id="cb10-64"><a href="#cb10-64"></a></span>
<span id="cb10-65"><a href="#cb10-65"></a></span>
<span id="cb10-66"><a href="#cb10-66"></a><span class="fu"># Phylogenetic tree plotting</span></span>
<span id="cb10-67"><a href="#cb10-67"></a></span>
<span id="cb10-68"><a href="#cb10-68"></a>Goal is to produce a tree that has for tip labels the Genus assigned by IDTAXA + the strain identifier, with different internal nodes labeled by coarser taxonomic grouping. The clusters assigned by <span class="in">`clusterizer`</span> will be represented by different colors at the tips.</span>
<span id="cb10-69"><a href="#cb10-69"></a></span>
<span id="cb10-70"><a href="#cb10-70"></a>First lets look at the tree with just the identifiers on the leaves</span>
<span id="cb10-71"><a href="#cb10-71"></a></span>
<span id="cb10-72"><a href="#cb10-72"></a>::: {#fig-01}</span>
<span id="cb10-75"><a href="#cb10-75"></a><span class="in">```{r}</span></span>
<span id="cb10-76"><a href="#cb10-76"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb10-77"><a href="#cb10-77"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb10-78"><a href="#cb10-78"></a><span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>) <span class="sc">+</span> </span>
<span id="cb10-79"><a href="#cb10-79"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb10-80"><a href="#cb10-80"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb10-81"><a href="#cb10-81"></a>  <span class="fu">geom_treescale</span>(<span class="at">offset=</span><span class="dv">1</span>)</span>
<span id="cb10-82"><a href="#cb10-82"></a><span class="in">```</span></span>
<span id="cb10-83"><a href="#cb10-83"></a>Maximum likelihood phylogenetic tree inferred from an alignment of the 16S rRNA gene amplified by Sanger sequencing from the duckweed SynCom. Scale bar at bottom represents 0.1 substitutions per site.</span>
<span id="cb10-84"><a href="#cb10-84"></a>:::</span>
<span id="cb10-85"><a href="#cb10-85"></a></span>
<span id="cb10-86"><a href="#cb10-86"></a>We can also look at the internal node numbering by rendering the tree as a dataframe</span>
<span id="cb10-87"><a href="#cb10-87"></a></span>
<span id="cb10-90"><a href="#cb10-90"></a><span class="in">```{r}</span></span>
<span id="cb10-91"><a href="#cb10-91"></a><span class="co"># the second tibble call is so quarto renders it properly</span></span>
<span id="cb10-92"><a href="#cb10-92"></a>syncom16s_seqs_mltree_nwk <span class="sc">%&gt;%</span> treeio<span class="sc">::</span><span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">tibble</span>()</span>
<span id="cb10-93"><a href="#cb10-93"></a><span class="in">```</span></span>
<span id="cb10-94"><a href="#cb10-94"></a>First we need to get our naming scheme set up</span>
<span id="cb10-95"><a href="#cb10-95"></a></span>
<span id="cb10-98"><a href="#cb10-98"></a><span class="in">```{r}</span></span>
<span id="cb10-99"><a href="#cb10-99"></a>newtipnames <span class="ot">&lt;-</span> final_table <span class="sc">%&gt;%</span> </span>
<span id="cb10-100"><a href="#cb10-100"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(genus, <span class="st">"(^[:alpha:]*).*"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-101"><a href="#cb10-101"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(genus) <span class="sc">|</span> genus <span class="sc">==</span> <span class="st">"unclassified"</span> <span class="sc">~</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(ncbi_best_species, <span class="st">"^[:alpha:]*"</span>),</span>
<span id="cb10-102"><a href="#cb10-102"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> genus)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-103"><a href="#cb10-103"></a>  <span class="fu">mutate</span>(<span class="at">newname =</span> <span class="fu">paste</span>(genus, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-104"><a href="#cb10-104"></a>  <span class="fu">mutate</span>(<span class="at">taxa_cluster =</span> <span class="fu">paste0</span>(<span class="st">"c"</span>, taxa_cluster)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-105"><a href="#cb10-105"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">id=</span>strainID, newname, taxa_cluster, genus, order, family)</span>
<span id="cb10-106"><a href="#cb10-106"></a><span class="in">```</span></span>
<span id="cb10-107"><a href="#cb10-107"></a></span>
<span id="cb10-108"><a href="#cb10-108"></a>First we need to generate a color palette that lets us distinguish the different colors</span>
<span id="cb10-109"><a href="#cb10-109"></a></span>
<span id="cb10-112"><a href="#cb10-112"></a><span class="in">```{r}</span></span>
<span id="cb10-113"><a href="#cb10-113"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb10-114"><a href="#cb10-114"></a><span class="fu">set.seed</span>(<span class="dv">341234</span>)</span>
<span id="cb10-115"><a href="#cb10-115"></a>mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">max</span>(final_table<span class="sc">$</span>taxa_cluster), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb10-116"><a href="#cb10-116"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span>
<span id="cb10-117"><a href="#cb10-117"></a><span class="in">```</span></span>
<span id="cb10-118"><a href="#cb10-118"></a></span>
<span id="cb10-119"><a href="#cb10-119"></a>Now we need to identify the internal nodes that we want to highlight. We can do this using the <span class="in">`MRCA`</span> function from ggtree</span>
<span id="cb10-120"><a href="#cb10-120"></a></span>
<span id="cb10-123"><a href="#cb10-123"></a><span class="in">```{r}</span></span>
<span id="cb10-124"><a href="#cb10-124"></a><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>)</span>
<span id="cb10-125"><a href="#cb10-125"></a><span class="in">```</span></span>
<span id="cb10-126"><a href="#cb10-126"></a></span>
<span id="cb10-127"><a href="#cb10-127"></a>Code to plot the tree</span>
<span id="cb10-128"><a href="#cb10-128"></a></span>
<span id="cb10-131"><a href="#cb10-131"></a><span class="in">```{r}</span></span>
<span id="cb10-132"><a href="#cb10-132"></a>offset <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb10-133"><a href="#cb10-133"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>)</span>
<span id="cb10-134"><a href="#cb10-134"></a></span>
<span id="cb10-135"><a href="#cb10-135"></a>p2 <span class="ot">&lt;-</span> p1 <span class="sc">%&lt;+%</span> newtipnames <span class="sc">+</span> </span>
<span id="cb10-136"><a href="#cb10-136"></a>  <span class="fu">geom_tiplab</span>(<span class="fu">aes</span>(<span class="at">label =</span> newname), <span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb10-137"><a href="#cb10-137"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> taxa_cluster), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb10-138"><a href="#cb10-138"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal, <span class="at">guide=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb10-139"><a href="#cb10-139"></a>  <span class="co"># Pseudomonadaceae </span></span>
<span id="cb10-140"><a href="#cb10-140"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>), </span>
<span id="cb10-141"><a href="#cb10-141"></a>                <span class="at">label=</span><span class="st">"(Î³) Pseudomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-142"><a href="#cb10-142"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-143"><a href="#cb10-143"></a>  <span class="co"># Aeromonadaceae</span></span>
<span id="cb10-144"><a href="#cb10-144"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA41"</span>, <span class="st">"HA04"</span>), </span>
<span id="cb10-145"><a href="#cb10-145"></a>                <span class="at">label=</span><span class="st">"(Î³) Aeromonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-146"><a href="#cb10-146"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-147"><a href="#cb10-147"></a>  <span class="co"># Enterobacteriaceae</span></span>
<span id="cb10-148"><a href="#cb10-148"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA14"</span>, <span class="st">"HA44"</span>), </span>
<span id="cb10-149"><a href="#cb10-149"></a>                <span class="at">label=</span><span class="st">"(Î³) Enterobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-150"><a href="#cb10-150"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-151"><a href="#cb10-151"></a>  <span class="co"># Moraxellaceae</span></span>
<span id="cb10-152"><a href="#cb10-152"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL64"</span>, <span class="st">"HA82"</span>), </span>
<span id="cb10-153"><a href="#cb10-153"></a>                <span class="at">label=</span><span class="st">"(Î³) Moraxellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-154"><a href="#cb10-154"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-155"><a href="#cb10-155"></a>  <span class="co"># Chromobacteriaceae</span></span>
<span id="cb10-156"><a href="#cb10-156"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA47"</span>, <span class="st">"HA65"</span>), </span>
<span id="cb10-157"><a href="#cb10-157"></a>                <span class="at">label=</span><span class="st">"(Î³) Chromobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-158"><a href="#cb10-158"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-159"><a href="#cb10-159"></a>  <span class="co"># Burkholderiaceae</span></span>
<span id="cb10-160"><a href="#cb10-160"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA52"</span>, <span class="st">"HA18"</span>), </span>
<span id="cb10-161"><a href="#cb10-161"></a>                <span class="at">label=</span><span class="st">"(Î³) Burkholderiaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-162"><a href="#cb10-162"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-163"><a href="#cb10-163"></a>  <span class="co"># Sphingomonadaceae</span></span>
<span id="cb10-164"><a href="#cb10-164"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="dv">23</span>, </span>
<span id="cb10-165"><a href="#cb10-165"></a>                <span class="at">label=</span><span class="st">"(Î±) Sphingomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-166"><a href="#cb10-166"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-167"><a href="#cb10-167"></a>  <span class="co"># Streptococcaceae</span></span>
<span id="cb10-168"><a href="#cb10-168"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL12"</span>, <span class="st">"TL39"</span>), </span>
<span id="cb10-169"><a href="#cb10-169"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Streptococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-170"><a href="#cb10-170"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-171"><a href="#cb10-171"></a>  <span class="co"># Exiguobacteriaceae</span></span>
<span id="cb10-172"><a href="#cb10-172"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA63"</span>, <span class="st">"HA25"</span>), </span>
<span id="cb10-173"><a href="#cb10-173"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Exiguobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-174"><a href="#cb10-174"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb10-175"><a href="#cb10-175"></a>  <span class="co"># Deinococcaceae</span></span>
<span id="cb10-176"><a href="#cb10-176"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA42"</span>, <span class="st">"HA29"</span>), </span>
<span id="cb10-177"><a href="#cb10-177"></a>                <span class="at">label=</span><span class="st">"(Deinococci) Deinococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-178"><a href="#cb10-178"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.2</span>) <span class="sc">+</span></span>
<span id="cb10-179"><a href="#cb10-179"></a>  <span class="co"># Weeksellaceae</span></span>
<span id="cb10-180"><a href="#cb10-180"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA56"</span>, <span class="st">"HA57"</span>), </span>
<span id="cb10-181"><a href="#cb10-181"></a>                <span class="at">label=</span><span class="st">"(Bacteroidia) Weeksellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-182"><a href="#cb10-182"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.4</span>) <span class="sc">+</span></span>
<span id="cb10-183"><a href="#cb10-183"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>)</span>
<span id="cb10-184"><a href="#cb10-184"></a><span class="in">```</span></span>
<span id="cb10-185"><a href="#cb10-185"></a></span>
<span id="cb10-186"><a href="#cb10-186"></a>::: {#fig-02}</span>
<span id="cb10-189"><a href="#cb10-189"></a><span class="in">```{r}</span></span>
<span id="cb10-190"><a href="#cb10-190"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb10-191"><a href="#cb10-191"></a><span class="co">#| fig-height: 15</span></span>
<span id="cb10-192"><a href="#cb10-192"></a><span class="co">#| eval: true</span></span>
<span id="cb10-193"><a href="#cb10-193"></a><span class="co">#| echo: false</span></span>
<span id="cb10-194"><a href="#cb10-194"></a>p2</span>
<span id="cb10-195"><a href="#cb10-195"></a><span class="in">```</span></span>
<span id="cb10-196"><a href="#cb10-196"></a>As in @fig-01, but isolates have been labeled with their inferred genus (text) as well as their broader taxonomic class (right, vertical lines). Colors represent sequences clusters based on a 99% identity threshold (i.e., nodes of the same color have 16S rRNA gene sequences that are &gt; 99% similar)</span>
<span id="cb10-197"><a href="#cb10-197"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>