<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2025-07-18">

<title>Read and format Sanger 16S rRNA sequences from Duckweed SynCom – duckweedSynCom</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../R/01_read_format_sanger.html">1. 16S rRNA gene Sanger sequencing</a></li><li class="breadcrumb-item"><a href="../R/01_read_format_sanger.html">i) Process ab1files to fasta</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">duckweedSynCom</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. 16S rRNA gene Sanger sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../R/01_read_format_sanger.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">i) Process ab1files to fasta</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../R/02_seq_cluster_phylo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Clustering and phylogeny of 16S gene</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read-and-format" id="toc-read-and-format" class="nav-link" data-scroll-target="#read-and-format"><span class="header-section-number">2</span> Read and format</a>
  <ul class="collapse">
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata"><span class="header-section-number">2.1</span> Metadata</a></li>
  <li><a href="#batch20250606" id="toc-batch20250606" class="nav-link" data-scroll-target="#batch20250606"><span class="header-section-number">2.2</span> batch20250606</a>
  <ul class="collapse">
  <li><a href="#write-fasta" id="toc-write-fasta" class="nav-link" data-scroll-target="#write-fasta"><span class="header-section-number">2.2.1</span> Write fasta</a></li>
  </ul></li>
  <li><a href="#batch20250620" id="toc-batch20250620" class="nav-link" data-scroll-target="#batch20250620"><span class="header-section-number">2.3</span> batch20250620</a>
  <ul class="collapse">
  <li><a href="#write-fasta-1" id="toc-write-fasta-1" class="nav-link" data-scroll-target="#write-fasta-1"><span class="header-section-number">2.3.1</span> Write fasta</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#combine-batches" id="toc-combine-batches" class="nav-link" data-scroll-target="#combine-batches"><span class="header-section-number">3</span> Combine batches</a></li>
  <li><a href="#rename-fasta-files" id="toc-rename-fasta-files" class="nav-link" data-scroll-target="#rename-fasta-files"><span class="header-section-number">4</span> Rename fasta files</a>
  <ul class="collapse">
  <li><a href="#batch20250606-1" id="toc-batch20250606-1" class="nav-link" data-scroll-target="#batch20250606-1"><span class="header-section-number">4.1</span> batch20250606</a></li>
  <li><a href="#batch20250620-1" id="toc-batch20250620-1" class="nav-link" data-scroll-target="#batch20250620-1"><span class="header-section-number">4.2</span> batch20250620</a></li>
  <li><a href="#combine-and-write" id="toc-combine-and-write" class="nav-link" data-scroll-target="#combine-and-write"><span class="header-section-number">4.3</span> Combine and write</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../R/01_read_format_sanger.html">1. 16S rRNA gene Sanger sequencing</a></li><li class="breadcrumb-item"><a href="../R/01_read_format_sanger.html">i) Process ab1files to fasta</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Read and format Sanger 16S rRNA sequences from Duckweed SynCom</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 18, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This reads Fwd/Rev Sanger sequencing reads of the 16S rRNA gene from the bacterial isolates, trims them, overlaps them, and produces fasta condtigs for use in downstream analysises.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Loads required libraries</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">options</span>(<span class="at">repos =</span> <span class="fu">c</span>(<span class="at">CRAN =</span> <span class="st">"https://packagemanager.posit.co/cran/__linux__/jammy/latest"</span>))</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(here)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(sangeranalyseR)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(Biostrings)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-and-format" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read and format</h1>
<p>This code follows the <a href="https://sangeranalyser.readthedocs.io/en/latest/content/quickstart.html">instructions here</a>. Basically, the function reads Fwd/Rev traces trims them using “M1 - Modified Mott Trimming” algorithm, overlaps them then aligns them</p>
<section id="metadata" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="metadata"><span class="header-section-number">2.1</span> Metadata</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>sanger_batch_info <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"sanger_batch_info.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 171 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (5): sanger_batch, sanger_plate_well, bacteria_plate_well, location, goo...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
</section>
<section id="batch20250606" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="batch20250606"><span class="header-section-number">2.2</span> batch20250606</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>contigs_batch20250606 <span class="ot">&lt;-</span> sangeranalyseR<span class="sc">::</span><span class="fu">SangerAlignment</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="at">ABIF_Directory =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>, <span class="st">"ab1_files"</span>),</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="co"># Requires trimmed reads to be at least 400 bp. A good sanger trace has a trimmed</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="co"># length of about 800 bp</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="at">minReadLength =</span> <span class="dv">400</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="at">REGEX_SuffixForward =</span> <span class="st">"_F.ab1$"</span>,</span>
<span id="cb4-7"><a href="#cb4-7"></a>  <span class="at">REGEX_SuffixReverse =</span> <span class="st">"_R.ab1$"</span></span>
<span id="cb4-8"><a href="#cb4-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="write-fasta" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="write-fasta"><span class="header-section-number">2.2.1</span> Write fasta</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>sangeranalyseR<span class="sc">::</span><span class="fu">writeFasta</span>(contigs_batch20250606,</span>
<span id="cb5-2"><a href="#cb5-2"></a>                           here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="batch20250620" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="batch20250620"><span class="header-section-number">2.3</span> batch20250620</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>contigs_batch20250620 <span class="ot">&lt;-</span> sangeranalyseR<span class="sc">::</span><span class="fu">SangerAlignment</span>(</span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="at">ABIF_Directory =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>, <span class="st">"ab1_files"</span>),</span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="co"># Requires trimmed reads to be at least 400 bp. A good sanger trace has a trimmed</span></span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="co"># length of about 800 bp</span></span>
<span id="cb6-5"><a href="#cb6-5"></a>  <span class="at">minReadLength =</span> <span class="dv">400</span>,</span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="at">REGEX_SuffixForward =</span> <span class="st">"_F.ab1$"</span>,</span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="at">REGEX_SuffixReverse =</span> <span class="st">"_R.ab1$"</span></span>
<span id="cb6-8"><a href="#cb6-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="write-fasta-1" class="level3" data-number="2.3.1">
<h3 data-number="2.3.1" class="anchored" data-anchor-id="write-fasta-1"><span class="header-section-number">2.3.1</span> Write fasta</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>sangeranalyseR<span class="sc">::</span><span class="fu">writeFasta</span>(contigs_batch20250620,</span>
<span id="cb7-2"><a href="#cb7-2"></a>                           here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="combine-batches" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Combine batches</h1>
<p>Retrieve information about which sequences succeeded or failed</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>contigs_batch20250606_sf <span class="ot">&lt;-</span> contigs_batch20250606<span class="sc">@</span>objectResults<span class="sc">@</span>readResultTable <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(readName, creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">mutate</span>(<span class="at">sanger_batch =</span> <span class="st">"batch20250606"</span>,</span>
<span id="cb8-4"><a href="#cb8-4"></a>         <span class="at">sanger_plate_well =</span> <span class="fu">str_extract</span>(readName, <span class="st">"[:alnum:]*-([:upper:]*[:digit:]*)_[F|R].ab1"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, sanger_batch, <span class="at">sangeranalyseR_success=</span>creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>contigs_batch20250620_sf <span class="ot">&lt;-</span> contigs_batch20250620<span class="sc">@</span>objectResults<span class="sc">@</span>readResultTable <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(readName, creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="fu">mutate</span>(<span class="at">sanger_batch =</span> <span class="st">"batch20250620"</span>,</span>
<span id="cb9-4"><a href="#cb9-4"></a>         <span class="at">sanger_plate_well =</span> <span class="fu">str_extract</span>(readName, <span class="st">"[:alnum:]*-([:upper:]*[:digit:]*)_[F|R].ab1"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, sanger_batch, <span class="at">sangeranalyseR_success=</span>creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>sanger_batch_info_01 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(contigs_batch20250606_sf, contigs_batch20250620_sf) <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2"></a>  <span class="fu">right_join</span>(sanger_batch_info, <span class="at">by =</span> <span class="fu">join_by</span>(sanger_plate_well, sanger_batch)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="fu">arrange</span>(location, sanger_batch, bacteria_plate_well) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span> </span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="co"># create temporary variable holding the index in the group</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">mutate</span>(<span class="at">n =</span> stringr<span class="sc">::</span><span class="fu">str_pad</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">width =</span> <span class="dv">2</span>, <span class="at">side =</span> <span class="st">"left"</span>, <span class="at">pad =</span> <span class="st">"0"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="co"># create strainID of just alphanumeric. Strain is ordered by location, sanger_batch, and bacteria_plate_well</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>  <span class="fu">mutate</span>(<span class="at">strainID =</span> <span class="fu">case_when</span>(location <span class="sc">==</span> <span class="st">"Haunisten Allas"</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"HA"</span>, n),</span>
<span id="cb10-10"><a href="#cb10-10"></a>                              location <span class="sc">==</span> <span class="st">"Turun Linna"</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"TL"</span>, n),</span>
<span id="cb10-11"><a href="#cb10-11"></a>                              <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"SJ"</span>, n))) <span class="sc">%&gt;%</span> </span>
<span id="cb10-12"><a href="#cb10-12"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>n)</span>
<span id="cb10-13"><a href="#cb10-13"></a></span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="fu">write_tsv</span>(sanger_batch_info_01, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"sanger_batch_info_01.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="rename-fasta-files" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Rename fasta files</h1>
<p>Here we combine all the fastas into a single file and rename them.</p>
<section id="batch20250606-1" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="batch20250606-1"><span class="header-section-number">4.1</span> batch20250606</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>batch20250606_fa <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>, <span class="st">"Sanger_contigs_unalignment.fa"</span>))</span>
<span id="cb11-2"><a href="#cb11-2"></a></span>
<span id="cb11-3"><a href="#cb11-3"></a>batch20250606_fa_map <span class="ot">&lt;-</span> sanger_batch_info_01 <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="fu">filter</span>(sanger_batch <span class="sc">==</span> <span class="st">"batch20250606"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="fu">mutate</span>(<span class="at">sanger_plate_well =</span> <span class="fu">paste0</span>(<span class="st">"16S-"</span>, sanger_plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb11-8"><a href="#cb11-8"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sanger_plate_well"</span>)</span>
<span id="cb11-9"><a href="#cb11-9"></a></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="fu">names</span>(batch20250606_fa) <span class="ot">&lt;-</span> batch20250606_fa_map[<span class="fu">names</span>(batch20250606_fa),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="batch20250620-1" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="batch20250620-1"><span class="header-section-number">4.2</span> batch20250620</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>batch20250620_fa <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>, <span class="st">"Sanger_contigs_unalignment.fa"</span>))</span>
<span id="cb12-2"><a href="#cb12-2"></a></span>
<span id="cb12-3"><a href="#cb12-3"></a>batch20250620_fa_map <span class="ot">&lt;-</span> sanger_batch_info_01 <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">filter</span>(sanger_batch <span class="sc">==</span> <span class="st">"batch20250620"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="fu">mutate</span>(<span class="at">sanger_plate_well =</span> <span class="fu">paste0</span>(<span class="st">"16S-"</span>, sanger_plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sanger_plate_well"</span>)</span>
<span id="cb12-9"><a href="#cb12-9"></a></span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="fu">names</span>(batch20250620_fa) <span class="ot">&lt;-</span> batch20250620_fa_map[<span class="fu">names</span>(batch20250620_fa),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="combine-and-write" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="combine-and-write"><span class="header-section-number">4.3</span> Combine and write</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">writeXStringSet</span>(<span class="fu">c</span>(batch20250606_fa, batch20250620_fa),</span>
<span id="cb13-2"><a href="#cb13-2"></a>                here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"16S_rRNA_SynCom.fna"</span>),</span>
<span id="cb13-3"><a href="#cb13-3"></a>                <span class="at">format=</span><span class="st">'fasta'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb14" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb14-1"><a href="#cb14-1"></a><span class="co">---</span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="an">title:</span><span class="co"> "Read and format Sanger 16S rRNA sequences from Duckweed SynCom"</span></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="an">subtitle:</span><span class="co"> ""</span></span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb14-6"><a href="#cb14-6"></a><span class="an">abstract:</span><span class="co"> "This reads Fwd/Rev Sanger sequencing reads of the 16S rRNA gene from the bacterial isolates, trims them, overlaps them, and produces fasta condtigs for use in downstream analysises."</span></span>
<span id="cb14-7"><a href="#cb14-7"></a><span class="co">---</span></span>
<span id="cb14-8"><a href="#cb14-8"></a></span>
<span id="cb14-9"><a href="#cb14-9"></a><span class="fu"># Setup</span></span>
<span id="cb14-10"><a href="#cb14-10"></a></span>
<span id="cb14-11"><a href="#cb14-11"></a>Loads required libraries</span>
<span id="cb14-12"><a href="#cb14-12"></a></span>
<span id="cb14-15"><a href="#cb14-15"></a><span class="in">```{r}</span></span>
<span id="cb14-16"><a href="#cb14-16"></a><span class="co">#| output: false</span></span>
<span id="cb14-17"><a href="#cb14-17"></a><span class="fu">options</span>(<span class="at">repos =</span> <span class="fu">c</span>(<span class="at">CRAN =</span> <span class="st">"https://packagemanager.posit.co/cran/__linux__/jammy/latest"</span>))</span>
<span id="cb14-18"><a href="#cb14-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb14-19"><a href="#cb14-19"></a><span class="fu">library</span>(here)</span>
<span id="cb14-20"><a href="#cb14-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb14-21"><a href="#cb14-21"></a><span class="fu">library</span>(sangeranalyseR)</span>
<span id="cb14-22"><a href="#cb14-22"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb14-23"><a href="#cb14-23"></a><span class="in">```</span></span>
<span id="cb14-24"><a href="#cb14-24"></a></span>
<span id="cb14-25"><a href="#cb14-25"></a><span class="fu"># Read and format</span></span>
<span id="cb14-26"><a href="#cb14-26"></a></span>
<span id="cb14-27"><a href="#cb14-27"></a>This code follows the <span class="co">[</span><span class="ot">instructions here</span><span class="co">](https://sangeranalyser.readthedocs.io/en/latest/content/quickstart.html)</span>. Basically, the function reads Fwd/Rev traces trims them using "M1 - Modified Mott Trimming" algorithm, overlaps them then aligns them</span>
<span id="cb14-28"><a href="#cb14-28"></a></span>
<span id="cb14-29"><a href="#cb14-29"></a><span class="fu">## Metadata</span></span>
<span id="cb14-30"><a href="#cb14-30"></a></span>
<span id="cb14-33"><a href="#cb14-33"></a><span class="in">```{r}</span></span>
<span id="cb14-34"><a href="#cb14-34"></a>sanger_batch_info <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"sanger_batch_info.tsv"</span>))</span>
<span id="cb14-35"><a href="#cb14-35"></a><span class="in">```</span></span>
<span id="cb14-36"><a href="#cb14-36"></a></span>
<span id="cb14-37"><a href="#cb14-37"></a><span class="fu">## batch20250606</span></span>
<span id="cb14-38"><a href="#cb14-38"></a></span>
<span id="cb14-41"><a href="#cb14-41"></a><span class="in">```{r}</span></span>
<span id="cb14-42"><a href="#cb14-42"></a><span class="co">#| eval: false</span></span>
<span id="cb14-43"><a href="#cb14-43"></a><span class="co">#| include: true</span></span>
<span id="cb14-44"><a href="#cb14-44"></a>contigs_batch20250606 <span class="ot">&lt;-</span> sangeranalyseR<span class="sc">::</span><span class="fu">SangerAlignment</span>(</span>
<span id="cb14-45"><a href="#cb14-45"></a>  <span class="at">ABIF_Directory =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>, <span class="st">"ab1_files"</span>),</span>
<span id="cb14-46"><a href="#cb14-46"></a>  <span class="co"># Requires trimmed reads to be at least 400 bp. A good sanger trace has a trimmed</span></span>
<span id="cb14-47"><a href="#cb14-47"></a>  <span class="co"># length of about 800 bp</span></span>
<span id="cb14-48"><a href="#cb14-48"></a>  <span class="at">minReadLength =</span> <span class="dv">400</span>,</span>
<span id="cb14-49"><a href="#cb14-49"></a>  <span class="at">REGEX_SuffixForward =</span> <span class="st">"_F.ab1$"</span>,</span>
<span id="cb14-50"><a href="#cb14-50"></a>  <span class="at">REGEX_SuffixReverse =</span> <span class="st">"_R.ab1$"</span></span>
<span id="cb14-51"><a href="#cb14-51"></a>)</span>
<span id="cb14-52"><a href="#cb14-52"></a><span class="in">```</span></span>
<span id="cb14-53"><a href="#cb14-53"></a></span>
<span id="cb14-54"><a href="#cb14-54"></a><span class="fu">### Write fasta</span></span>
<span id="cb14-55"><a href="#cb14-55"></a></span>
<span id="cb14-58"><a href="#cb14-58"></a><span class="in">```{r}</span></span>
<span id="cb14-59"><a href="#cb14-59"></a><span class="co">#| eval: false</span></span>
<span id="cb14-60"><a href="#cb14-60"></a><span class="co">#| include: true</span></span>
<span id="cb14-61"><a href="#cb14-61"></a>sangeranalyseR<span class="sc">::</span><span class="fu">writeFasta</span>(contigs_batch20250606,</span>
<span id="cb14-62"><a href="#cb14-62"></a>                           here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>))</span>
<span id="cb14-63"><a href="#cb14-63"></a><span class="in">```</span></span>
<span id="cb14-64"><a href="#cb14-64"></a></span>
<span id="cb14-67"><a href="#cb14-67"></a><span class="in">```{r}</span></span>
<span id="cb14-68"><a href="#cb14-68"></a><span class="co">#| eval: false</span></span>
<span id="cb14-69"><a href="#cb14-69"></a><span class="co">#| include: false</span></span>
<span id="cb14-70"><a href="#cb14-70"></a><span class="co"># saving intermediate files</span></span>
<span id="cb14-71"><a href="#cb14-71"></a><span class="fu">write_rds</span>(contigs_batch20250606, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>, <span class="st">"batch20250606.rds"</span>))</span>
<span id="cb14-72"><a href="#cb14-72"></a><span class="in">```</span></span>
<span id="cb14-73"><a href="#cb14-73"></a></span>
<span id="cb14-76"><a href="#cb14-76"></a><span class="in">```{r}</span></span>
<span id="cb14-77"><a href="#cb14-77"></a><span class="co">#| eval: true</span></span>
<span id="cb14-78"><a href="#cb14-78"></a><span class="co">#| include: false</span></span>
<span id="cb14-79"><a href="#cb14-79"></a>contigs_batch20250606 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>,  <span class="st">"batch20250606"</span>, <span class="st">"batch20250606.rds"</span>))</span>
<span id="cb14-80"><a href="#cb14-80"></a><span class="in">```</span></span>
<span id="cb14-81"><a href="#cb14-81"></a></span>
<span id="cb14-82"><a href="#cb14-82"></a><span class="fu">## batch20250620</span></span>
<span id="cb14-83"><a href="#cb14-83"></a></span>
<span id="cb14-86"><a href="#cb14-86"></a><span class="in">```{r}</span></span>
<span id="cb14-87"><a href="#cb14-87"></a><span class="co">#| eval: false</span></span>
<span id="cb14-88"><a href="#cb14-88"></a><span class="co">#| include: true</span></span>
<span id="cb14-89"><a href="#cb14-89"></a>contigs_batch20250620 <span class="ot">&lt;-</span> sangeranalyseR<span class="sc">::</span><span class="fu">SangerAlignment</span>(</span>
<span id="cb14-90"><a href="#cb14-90"></a>  <span class="at">ABIF_Directory =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>, <span class="st">"ab1_files"</span>),</span>
<span id="cb14-91"><a href="#cb14-91"></a>  <span class="co"># Requires trimmed reads to be at least 400 bp. A good sanger trace has a trimmed</span></span>
<span id="cb14-92"><a href="#cb14-92"></a>  <span class="co"># length of about 800 bp</span></span>
<span id="cb14-93"><a href="#cb14-93"></a>  <span class="at">minReadLength =</span> <span class="dv">400</span>,</span>
<span id="cb14-94"><a href="#cb14-94"></a>  <span class="at">REGEX_SuffixForward =</span> <span class="st">"_F.ab1$"</span>,</span>
<span id="cb14-95"><a href="#cb14-95"></a>  <span class="at">REGEX_SuffixReverse =</span> <span class="st">"_R.ab1$"</span></span>
<span id="cb14-96"><a href="#cb14-96"></a>)</span>
<span id="cb14-97"><a href="#cb14-97"></a><span class="in">```</span></span>
<span id="cb14-98"><a href="#cb14-98"></a></span>
<span id="cb14-99"><a href="#cb14-99"></a><span class="fu">### Write fasta</span></span>
<span id="cb14-100"><a href="#cb14-100"></a></span>
<span id="cb14-103"><a href="#cb14-103"></a><span class="in">```{r}</span></span>
<span id="cb14-104"><a href="#cb14-104"></a><span class="co">#| eval: false</span></span>
<span id="cb14-105"><a href="#cb14-105"></a><span class="co">#| include: true</span></span>
<span id="cb14-106"><a href="#cb14-106"></a>sangeranalyseR<span class="sc">::</span><span class="fu">writeFasta</span>(contigs_batch20250620,</span>
<span id="cb14-107"><a href="#cb14-107"></a>                           here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>))</span>
<span id="cb14-108"><a href="#cb14-108"></a><span class="in">```</span></span>
<span id="cb14-109"><a href="#cb14-109"></a></span>
<span id="cb14-112"><a href="#cb14-112"></a><span class="in">```{r}</span></span>
<span id="cb14-113"><a href="#cb14-113"></a><span class="co">#| eval: false</span></span>
<span id="cb14-114"><a href="#cb14-114"></a><span class="co">#| include: false</span></span>
<span id="cb14-115"><a href="#cb14-115"></a><span class="co"># saving intermediate files</span></span>
<span id="cb14-116"><a href="#cb14-116"></a><span class="fu">write_rds</span>(contigs_batch20250620, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>, <span class="st">"batch20250620.rds"</span>))</span>
<span id="cb14-117"><a href="#cb14-117"></a><span class="in">```</span></span>
<span id="cb14-118"><a href="#cb14-118"></a></span>
<span id="cb14-121"><a href="#cb14-121"></a><span class="in">```{r}</span></span>
<span id="cb14-122"><a href="#cb14-122"></a><span class="co">#| eval: true</span></span>
<span id="cb14-123"><a href="#cb14-123"></a><span class="co">#| include: false</span></span>
<span id="cb14-124"><a href="#cb14-124"></a>contigs_batch20250620 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>,  <span class="st">"batch20250620"</span>, <span class="st">"batch20250620.rds"</span>))</span>
<span id="cb14-125"><a href="#cb14-125"></a><span class="in">```</span></span>
<span id="cb14-126"><a href="#cb14-126"></a></span>
<span id="cb14-127"><a href="#cb14-127"></a><span class="fu"># Combine batches</span></span>
<span id="cb14-128"><a href="#cb14-128"></a></span>
<span id="cb14-129"><a href="#cb14-129"></a>Retrieve information about which sequences succeeded or failed</span>
<span id="cb14-130"><a href="#cb14-130"></a></span>
<span id="cb14-133"><a href="#cb14-133"></a><span class="in">```{r}</span></span>
<span id="cb14-134"><a href="#cb14-134"></a>contigs_batch20250606_sf <span class="ot">&lt;-</span> contigs_batch20250606<span class="sc">@</span>objectResults<span class="sc">@</span>readResultTable <span class="sc">%&gt;%</span> </span>
<span id="cb14-135"><a href="#cb14-135"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(readName, creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb14-136"><a href="#cb14-136"></a>  <span class="fu">mutate</span>(<span class="at">sanger_batch =</span> <span class="st">"batch20250606"</span>,</span>
<span id="cb14-137"><a href="#cb14-137"></a>         <span class="at">sanger_plate_well =</span> <span class="fu">str_extract</span>(readName, <span class="st">"[:alnum:]*-([:upper:]*[:digit:]*)_[F|R].ab1"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-138"><a href="#cb14-138"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, sanger_batch, <span class="at">sangeranalyseR_success=</span>creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb14-139"><a href="#cb14-139"></a>  <span class="fu">distinct</span>()</span>
<span id="cb14-140"><a href="#cb14-140"></a><span class="in">```</span></span>
<span id="cb14-141"><a href="#cb14-141"></a></span>
<span id="cb14-144"><a href="#cb14-144"></a><span class="in">```{r}</span></span>
<span id="cb14-145"><a href="#cb14-145"></a>contigs_batch20250620_sf <span class="ot">&lt;-</span> contigs_batch20250620<span class="sc">@</span>objectResults<span class="sc">@</span>readResultTable <span class="sc">%&gt;%</span> </span>
<span id="cb14-146"><a href="#cb14-146"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(readName, creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb14-147"><a href="#cb14-147"></a>  <span class="fu">mutate</span>(<span class="at">sanger_batch =</span> <span class="st">"batch20250620"</span>,</span>
<span id="cb14-148"><a href="#cb14-148"></a>         <span class="at">sanger_plate_well =</span> <span class="fu">str_extract</span>(readName, <span class="st">"[:alnum:]*-([:upper:]*[:digit:]*)_[F|R].ab1"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-149"><a href="#cb14-149"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, sanger_batch, <span class="at">sangeranalyseR_success=</span>creationResult) <span class="sc">%&gt;%</span> </span>
<span id="cb14-150"><a href="#cb14-150"></a>  <span class="fu">distinct</span>()</span>
<span id="cb14-151"><a href="#cb14-151"></a><span class="in">```</span></span>
<span id="cb14-152"><a href="#cb14-152"></a></span>
<span id="cb14-155"><a href="#cb14-155"></a><span class="in">```{r}</span></span>
<span id="cb14-156"><a href="#cb14-156"></a>sanger_batch_info_01 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(contigs_batch20250606_sf, contigs_batch20250620_sf) <span class="sc">%&gt;%</span> </span>
<span id="cb14-157"><a href="#cb14-157"></a>  <span class="fu">right_join</span>(sanger_batch_info, <span class="at">by =</span> <span class="fu">join_by</span>(sanger_plate_well, sanger_batch)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-158"><a href="#cb14-158"></a>  <span class="fu">arrange</span>(location, sanger_batch, bacteria_plate_well) <span class="sc">%&gt;%</span> </span>
<span id="cb14-159"><a href="#cb14-159"></a>  <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span> </span>
<span id="cb14-160"><a href="#cb14-160"></a>  <span class="co"># create temporary variable holding the index in the group</span></span>
<span id="cb14-161"><a href="#cb14-161"></a>  <span class="fu">mutate</span>(<span class="at">n =</span> stringr<span class="sc">::</span><span class="fu">str_pad</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">width =</span> <span class="dv">2</span>, <span class="at">side =</span> <span class="st">"left"</span>, <span class="at">pad =</span> <span class="st">"0"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-162"><a href="#cb14-162"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-163"><a href="#cb14-163"></a>  <span class="co"># create strainID of just alphanumeric. Strain is ordered by location, sanger_batch, and bacteria_plate_well</span></span>
<span id="cb14-164"><a href="#cb14-164"></a>  <span class="fu">mutate</span>(<span class="at">strainID =</span> <span class="fu">case_when</span>(location <span class="sc">==</span> <span class="st">"Haunisten Allas"</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"HA"</span>, n),</span>
<span id="cb14-165"><a href="#cb14-165"></a>                              location <span class="sc">==</span> <span class="st">"Turun Linna"</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"TL"</span>, n),</span>
<span id="cb14-166"><a href="#cb14-166"></a>                              <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"SJ"</span>, n))) <span class="sc">%&gt;%</span> </span>
<span id="cb14-167"><a href="#cb14-167"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>n)</span>
<span id="cb14-168"><a href="#cb14-168"></a></span>
<span id="cb14-169"><a href="#cb14-169"></a><span class="fu">write_tsv</span>(sanger_batch_info_01, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"sanger_batch_info_01.tsv"</span>))</span>
<span id="cb14-170"><a href="#cb14-170"></a><span class="in">```</span></span>
<span id="cb14-171"><a href="#cb14-171"></a></span>
<span id="cb14-172"><a href="#cb14-172"></a><span class="fu"># Rename fasta files</span></span>
<span id="cb14-173"><a href="#cb14-173"></a></span>
<span id="cb14-174"><a href="#cb14-174"></a>Here we combine all the fastas into a single file and rename them.</span>
<span id="cb14-175"><a href="#cb14-175"></a></span>
<span id="cb14-176"><a href="#cb14-176"></a><span class="fu">## batch20250606</span></span>
<span id="cb14-177"><a href="#cb14-177"></a></span>
<span id="cb14-180"><a href="#cb14-180"></a><span class="in">```{r}</span></span>
<span id="cb14-181"><a href="#cb14-181"></a>batch20250606_fa <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250606"</span>, <span class="st">"Sanger_contigs_unalignment.fa"</span>))</span>
<span id="cb14-182"><a href="#cb14-182"></a></span>
<span id="cb14-183"><a href="#cb14-183"></a>batch20250606_fa_map <span class="ot">&lt;-</span> sanger_batch_info_01 <span class="sc">%&gt;%</span> </span>
<span id="cb14-184"><a href="#cb14-184"></a>  <span class="fu">filter</span>(sanger_batch <span class="sc">==</span> <span class="st">"batch20250606"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-185"><a href="#cb14-185"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb14-186"><a href="#cb14-186"></a>  <span class="fu">mutate</span>(<span class="at">sanger_plate_well =</span> <span class="fu">paste0</span>(<span class="st">"16S-"</span>, sanger_plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-187"><a href="#cb14-187"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-188"><a href="#cb14-188"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sanger_plate_well"</span>)</span>
<span id="cb14-189"><a href="#cb14-189"></a></span>
<span id="cb14-190"><a href="#cb14-190"></a><span class="fu">names</span>(batch20250606_fa) <span class="ot">&lt;-</span> batch20250606_fa_map[<span class="fu">names</span>(batch20250606_fa),]</span>
<span id="cb14-191"><a href="#cb14-191"></a><span class="in">```</span></span>
<span id="cb14-192"><a href="#cb14-192"></a></span>
<span id="cb14-193"><a href="#cb14-193"></a><span class="fu">## batch20250620</span></span>
<span id="cb14-194"><a href="#cb14-194"></a></span>
<span id="cb14-197"><a href="#cb14-197"></a><span class="in">```{r}</span></span>
<span id="cb14-198"><a href="#cb14-198"></a>batch20250620_fa <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"batch20250620"</span>, <span class="st">"Sanger_contigs_unalignment.fa"</span>))</span>
<span id="cb14-199"><a href="#cb14-199"></a></span>
<span id="cb14-200"><a href="#cb14-200"></a>batch20250620_fa_map <span class="ot">&lt;-</span> sanger_batch_info_01 <span class="sc">%&gt;%</span> </span>
<span id="cb14-201"><a href="#cb14-201"></a>  <span class="fu">filter</span>(sanger_batch <span class="sc">==</span> <span class="st">"batch20250620"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-202"><a href="#cb14-202"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sanger_plate_well, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb14-203"><a href="#cb14-203"></a>  <span class="fu">mutate</span>(<span class="at">sanger_plate_well =</span> <span class="fu">paste0</span>(<span class="st">"16S-"</span>, sanger_plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-204"><a href="#cb14-204"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-205"><a href="#cb14-205"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sanger_plate_well"</span>)</span>
<span id="cb14-206"><a href="#cb14-206"></a></span>
<span id="cb14-207"><a href="#cb14-207"></a><span class="fu">names</span>(batch20250620_fa) <span class="ot">&lt;-</span> batch20250620_fa_map[<span class="fu">names</span>(batch20250620_fa),]</span>
<span id="cb14-208"><a href="#cb14-208"></a><span class="in">```</span></span>
<span id="cb14-209"><a href="#cb14-209"></a></span>
<span id="cb14-210"><a href="#cb14-210"></a><span class="fu">## Combine and write</span></span>
<span id="cb14-211"><a href="#cb14-211"></a></span>
<span id="cb14-214"><a href="#cb14-214"></a><span class="in">```{r}</span></span>
<span id="cb14-215"><a href="#cb14-215"></a><span class="fu">writeXStringSet</span>(<span class="fu">c</span>(batch20250606_fa, batch20250620_fa),</span>
<span id="cb14-216"><a href="#cb14-216"></a>                here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"16S_rRNA_SynCom.fna"</span>),</span>
<span id="cb14-217"><a href="#cb14-217"></a>                <span class="at">format=</span><span class="st">'fasta'</span>)</span>
<span id="cb14-218"><a href="#cb14-218"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>