<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2025-07-18">

<title>Clustering/phylogenetics of 16S rRNA sequences from duckweed SynCom – duckweedSynCom</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../R/01_read_format_sanger.html">1. 16S rRNA gene Sanger sequencing</a></li><li class="breadcrumb-item"><a href="../R/02_seq_cluster_phylo.html">ii) Clustering and phylogeny of 16S gene</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">duckweedSynCom</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. 16S rRNA gene Sanger sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../R/01_read_format_sanger.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process ab1files to fasta</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../R/02_seq_cluster_phylo.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">ii) Clustering and phylogeny of 16S gene</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read-and-format" id="toc-read-and-format" class="nav-link" data-scroll-target="#read-and-format"><span class="header-section-number">2</span> Read and format</a></li>
  <li><a href="#redundancy" id="toc-redundancy" class="nav-link" data-scroll-target="#redundancy"><span class="header-section-number">3</span> Redundancy</a></li>
  <li><a href="#cluster" id="toc-cluster" class="nav-link" data-scroll-target="#cluster"><span class="header-section-number">4</span> Cluster</a></li>
  <li><a href="#classify-idtaxa" id="toc-classify-idtaxa" class="nav-link" data-scroll-target="#classify-idtaxa"><span class="header-section-number">5</span> Classify (IDTAXA)</a></li>
  <li><a href="#classify-blastn" id="toc-classify-blastn" class="nav-link" data-scroll-target="#classify-blastn"><span class="header-section-number">6</span> Classify (Blastn)</a></li>
  <li><a href="#combine-clustering-and-classification" id="toc-combine-clustering-and-classification" class="nav-link" data-scroll-target="#combine-clustering-and-classification"><span class="header-section-number">7</span> Combine clustering and classification</a></li>
  <li><a href="#s-rrna-sequence-alignment" id="toc-s-rrna-sequence-alignment" class="nav-link" data-scroll-target="#s-rrna-sequence-alignment"><span class="header-section-number">8</span> 16S rRNA sequence alignment</a></li>
  <li><a href="#phylogenetic-tree-building" id="toc-phylogenetic-tree-building" class="nav-link" data-scroll-target="#phylogenetic-tree-building"><span class="header-section-number">9</span> Phylogenetic tree building</a></li>
  <li><a href="#phylogenetic-tree-plotting" id="toc-phylogenetic-tree-plotting" class="nav-link" data-scroll-target="#phylogenetic-tree-plotting"><span class="header-section-number">10</span> Phylogenetic tree plotting</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../R/01_read_format_sanger.html">1. 16S rRNA gene Sanger sequencing</a></li><li class="breadcrumb-item"><a href="../R/02_seq_cluster_phylo.html">ii) Clustering and phylogeny of 16S gene</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Clustering/phylogenetics of 16S rRNA sequences from duckweed SynCom</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 18, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    Various operations to group the isolates into clusters using clustering algorithms and phylogenetic inference. At the end a phylogenetic tree for the isolates is plotting with associated taxonomic information as well as clustering identity based on shared kmers.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Loads required libraries</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">options</span>(<span class="at">repos =</span> <span class="fu">c</span>(<span class="at">CRAN =</span> <span class="st">"https://packagemanager.posit.co/cran/__linux__/jammy/latest"</span>))</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(here)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(DECIPHER)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-and-format" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read and format</h1>
<p>Read 16S rRNA fasta sequences from the prior step.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>syncom16s_fna <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"16S_rRNA_SynCom.fna"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Convert to a DNA stringset object</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>syncom16s_seqs <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(syncom16s_fna)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="redundancy" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Redundancy</h1>
<p>All sequences are unique which is expected</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">duplicated</span>(syncom16s_seqs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [85] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [97] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</code></pre>
</div>
</div>
</section>
<section id="cluster" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Cluster</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">set.seed</span>(<span class="dv">36742</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a>syncom16s_clusters <span class="ot">&lt;-</span> <span class="fu">Clusterize</span>(syncom16s_seqs,</span>
<span id="cb6-3"><a href="#cb6-3"></a>                       <span class="at">cutoff=</span><span class="fl">0.01</span>,     <span class="co"># at ~ species level (ie 99% identity)</span></span>
<span id="cb6-4"><a href="#cb6-4"></a>                       <span class="at">minCoverage=</span><span class="fl">0.5</span>, <span class="co"># &gt; 50% coverage</span></span>
<span id="cb6-5"><a href="#cb6-5"></a>                       <span class="at">processors=</span><span class="cn">NULL</span>) <span class="co"># use all CPUs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Partitioning sequences by 4-mer similarity:
================================================================================

Time difference of 0.05 secs

Sorting by relatedness within 32 groups:

iteration 1 of up to 43 (100.0% stability) 

Time difference of 0.05 secs

Clustering sequences by 10-mer similarity:
================================================================================

Time difference of 0.22 secs

Clusters via relatedness sorting: 100% (0% exclusively)
Clusters via rare 4-mers: 100% (0% exclusively)
Estimated clustering effectiveness: 100%</code></pre>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># reset random seed</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="classify-idtaxa" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Classify (IDTAXA)</h1>
<p>Using <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0521-5">IDTAXA</a> following the <a href="https://www2.decipher.codes/ClassifyOrganisms.html">approach shown here</a>.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">set.seed</span>(<span class="dv">234676</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a></span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co"># load the training data set</span></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="fu">load</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"idtaxa"</span>, <span class="st">"GTDB_r226-mod_April2025.RData"</span>))</span>
<span id="cb9-5"><a href="#cb9-5"></a></span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co"># classify the sequences</span></span>
<span id="cb9-7"><a href="#cb9-7"></a>syncom16s_ids <span class="ot">&lt;-</span> <span class="fu">IdTaxa</span>(syncom16s_seqs,</span>
<span id="cb9-8"><a href="#cb9-8"></a>   trainingSet,</span>
<span id="cb9-9"><a href="#cb9-9"></a>   <span class="at">strand=</span><span class="st">"both"</span>, <span class="co"># or "top" if same as trainingSet</span></span>
<span id="cb9-10"><a href="#cb9-10"></a>   <span class="at">threshold=</span><span class="dv">60</span>, <span class="co"># 60 (cautious) or 50 (sensible)</span></span>
<span id="cb9-11"><a href="#cb9-11"></a>   <span class="at">processors=</span><span class="cn">NULL</span>) <span class="co"># use all available processors</span></span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co"># reset random seed</span></span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>syncom16s_idtx <span class="ot">&lt;-</span> <span class="fu">imap</span>(</span>
<span id="cb10-2"><a href="#cb10-2"></a>  syncom16s_ids,</span>
<span id="cb10-3"><a href="#cb10-3"></a>  \(x, idx) <span class="fu">tibble</span>(<span class="at">a =</span> x<span class="sc">$</span>rank, <span class="at">b =</span> <span class="fu">paste</span>(x<span class="sc">$</span>taxon,<span class="st">" ("</span>,<span class="fu">round</span>(x<span class="sc">$</span>confidence, <span class="at">digits =</span> <span class="dv">1</span>),<span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="#cb10-4"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> a, <span class="at">values_from =</span> b) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>    <span class="fu">mutate</span>(<span class="at">id =</span> idx)</span>
<span id="cb10-6"><a href="#cb10-6"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="fu">list_rbind</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="fu">relocate</span>(id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="classify-blastn" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Classify (Blastn)</h1>
<p>These steps need to be run outside of R/Rstudio</p>
<p>First we need to <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST/">download</a> and <a href="https://www.ncbi.nlm.nih.gov/books/NBK52640/">install</a> NCBI blast+ suite (I am using v2.16.0) and <a href="https://ftp.ncbi.nlm.nih.gov/blast/db/">download</a> the 16S_ribosomal_RNA database.</p>
<pre><code>blastn -num_threads 12 -db 16S_ribosomal_RNA -query 16S_rRNA_SynCom.fna -dust no -max_target_seqs 5 -out 16S_rRNA_SynCom_v_blast16SDB.tsv -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore staxid"</code></pre>
<p>The last column contains the taxonomy ID from the NCBI 16S Ribosomal database. We will just take the taxonomic lineage of the best hit</p>
<pre><code>csvtk tab2csv 16S_rRNA_SynCom_v_blast16SDB.tsv | csvtk cut -f1,13 | csvtk spread -k1 -v2 | sed 's/^,//g' | csvtk transpose | cut -d ";" -f1 | csvtk cut -f2,1 | csvtk csv2tab &gt; a
csvtk tab2csv 16S_rRNA_SynCom_v_blast16SDB.tsv | csvtk cut -f1,13 | csvtk spread -k1 -v2 | sed 's/^,//g' | csvtk transpose | cut -d ";" -f1 | csvtk cut -f2 | taxonkit lineage -n -L &gt; b
# combine into one file
paste a b &gt; blastn_lineage_best_hit.tsv
rm a b</code></pre>
<p>Now read this file back into R</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>syncom16s_blastn <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"blastn_lineage_best_hit.tsv"</span>),</span>
<span id="cb13-2"><a href="#cb13-2"></a>                <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">id =</span> X2, <span class="at">ncbi_taxid =</span> X1, <span class="at">ncbi_best_species =</span> X4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 118 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): X2, X4
dbl (2): X1, X3

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
</section>
<section id="combine-clustering-and-classification" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Combine clustering and classification</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>syncom16s_idtx_lineage_info <span class="ot">&lt;-</span> syncom16s_idtx <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="fu">left_join</span>(<span class="fu">rownames_to_column</span>(syncom16s_clusters, <span class="at">var =</span> <span class="st">"id"</span>), <span class="at">by =</span> <span class="fu">join_by</span>(id)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">left_join</span>(syncom16s_blastn, <span class="at">by =</span> <span class="fu">join_by</span>(id)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">relocate</span>(ncbi_best_species, <span class="at">.after=</span><span class="st">"genus"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  <span class="fu">relocate</span>(cluster, <span class="at">.after=</span><span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6"></a>  <span class="fu">arrange</span>(cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="s-rrna-sequence-alignment" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> 16S rRNA sequence alignment</h1>
<p>We start with DNA sequences and convert to RNA so that DECIPHER can leverage structural information from the RNA</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>syncom16s_seqs_rna <span class="ot">&lt;-</span> <span class="fu">RNAStringSet</span>(syncom16s_seqs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Align the sequences</p>
<p>From <a href="https://bioconductor.org/packages/devel/bioc/vignettes/DECIPHER/inst/doc/ArtOfAlignmentInR.pdf">DECIPHER documentation on sequence alignment</a></p>
<blockquote class="blockquote">
<p>5.2 Example: Non-coding RNA sequences Much like proteins, non-coding RNAs often have a conserved secondary structure that can be used to improve their alignment. The PredictDBN function will predict base pairings from a sequence alignment by calculating the mutual information between pairs of positions. If RNA sequences are given as input, AlignSeqs will automatically use the output of PredictDBN to iteratively improve the alignment. Providing an RNAStringSet also causes single-base and double-base substitution matrices to be used, and is preferable to providing a DNAStringSet when the sequences are non-coding RNA. The type of the input sequences can easily be converted to RNA, as shown below.</p>
</blockquote>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">set.seed</span>(<span class="dv">2342</span>)</span>
<span id="cb17-2"><a href="#cb17-2"></a>syncom16s_seqs_rna_aligned <span class="ot">&lt;-</span> <span class="fu">AlignSeqs</span>(syncom16s_seqs_rna)</span>
<span id="cb17-3"><a href="#cb17-3"></a><span class="co"># reset random seed</span></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># Write as aligned fasta DNA file </span></span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="fu">writeXStringSet</span>(<span class="fu">DNAStringSet</span>(syncom16s_seqs_rna_aligned),</span>
<span id="cb18-3"><a href="#cb18-3"></a>                here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"16S_rRNA_SynCom_aligned.fna"</span>),</span>
<span id="cb18-4"><a href="#cb18-4"></a>                <span class="at">format=</span><span class="st">'fasta'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="phylogenetic-tree-building" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Phylogenetic tree building</h1>
<p>Following maximum likelihood tutorial <a href="https://bioconductor.org/packages/devel/bioc/vignettes/DECIPHER/inst/doc/GrowingTrees.pdf">DECIPHER documentation on phylogenetic trees</a></p>
<p>This step constructs a phylogenetic tree using the general time reversable (GTR) model with a discrete Gamma model <a href="https://link.springer.com/article/10.1007/BF00160154">(Yang, 1994)</a> with the default 4 rate categories. Branch support is depicted using aBayes probabilities, which is a Bayesian-like transformation of an approximate Likelihood Ratio Test <a href="https://academic.oup.com/sysbio/article/60/5/685/1644562">(Anisimova et al., 2011)</a>.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="fu">set.seed</span>(<span class="dv">26345</span>)</span>
<span id="cb19-2"><a href="#cb19-2"></a>syncom16s_seqs_mltree <span class="ot">&lt;-</span> <span class="fu">Treeline</span>(</span>
<span id="cb19-3"><a href="#cb19-3"></a>  syncom16s_seqs_rna_aligned,</span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="at">method =</span> <span class="st">"ML"</span>,</span>
<span id="cb19-5"><a href="#cb19-5"></a>  <span class="at">model =</span> <span class="st">"GTR+G4"</span>,</span>
<span id="cb19-6"><a href="#cb19-6"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-7"><a href="#cb19-7"></a>  <span class="at">processors =</span> <span class="dv">12</span></span>
<span id="cb19-8"><a href="#cb19-8"></a>)</span>
<span id="cb19-9"><a href="#cb19-9"></a><span class="co"># reset random seed</span></span>
<span id="cb19-10"><a href="#cb19-10"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="co"># this function adds an 'edgetext' attribute that is recognized by WriteDendrogram so that</span></span>
<span id="cb20-2"><a href="#cb20-2"></a><span class="co"># branch support values (aBayes) are written properly</span></span>
<span id="cb20-3"><a href="#cb20-3"></a>add_boots <span class="ot">&lt;-</span> <span class="cf">function</span>(node){</span>
<span id="cb20-4"><a href="#cb20-4"></a>  s <span class="ot">&lt;-</span> <span class="fu">attr</span>(node, <span class="st">"probability"</span>)</span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(s))</span>
<span id="cb20-6"><a href="#cb20-6"></a>    <span class="fu">attr</span>(node, <span class="st">"edgetext"</span>) <span class="ot">&lt;-</span> <span class="fu">formatC</span>(<span class="fu">as.numeric</span>(s), <span class="at">digits=</span><span class="dv">2</span>, <span class="at">format=</span><span class="st">"f"</span>)</span>
<span id="cb20-7"><a href="#cb20-7"></a>  <span class="fu">return</span>(node)</span>
<span id="cb20-8"><a href="#cb20-8"></a>}</span>
<span id="cb20-9"><a href="#cb20-9"></a></span>
<span id="cb20-10"><a href="#cb20-10"></a>syncom16s_seqs_mltree <span class="ot">&lt;-</span> <span class="fu">dendrapply</span>(syncom16s_seqs_mltree, add_boots)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Save ML tree as newick file</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="fu">WriteDendrogram</span>(syncom16s_seqs_mltree, </span>
<span id="cb21-2"><a href="#cb21-2"></a>                <span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.nwk"</span>),</span>
<span id="cb21-3"><a href="#cb21-3"></a>                <span class="co"># quote ="" ensure that bootstraps are not written quoted as strings</span></span>
<span id="cb21-4"><a href="#cb21-4"></a>                <span class="co"># so that ggtree doesn't fail to parse them</span></span>
<span id="cb21-5"><a href="#cb21-5"></a>                <span class="at">quote=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Tree building results</p>
<pre><code>Fitting initial tree to model:
GTR+G4 -ln(L) = 15520, AICc = 31633, BIC = 32781

Optimizing up to 400 candidate trees:
Tree #136. -ln(L) = 15395.993 (0.000%), 9 Climbs, 0 Grafts of 7                                                                                                       

Finalizing the best tree (#130):
-ln(L) = 15395.993 (0.000%), 1 Climb                                                                                                                                  

Model parameters:
Frequency(A) = 0.244
Frequency(C) = 0.223
Frequency(G) = 0.312
Frequency(T) = 0.221
Rate A &lt;-&gt; C = 0.816
Rate A &lt;-&gt; G = 1.978
Rate A &lt;-&gt; T = 1.262
Rate C &lt;-&gt; G = 0.677
Rate C &lt;-&gt; T = 3.029
Rate G &lt;-&gt; T = 1.000
Alpha = 0.302

Time difference of 1090.3 secs</code></pre>
</section>
<section id="phylogenetic-tree-plotting" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Phylogenetic tree plotting</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="fu">library</span>(ggtree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>ggtree v3.17.1.001 Learn more at https://yulab-smu.top/contribution-tree-data/

Please cite:

S Xu, Z Dai, P Guo, X Fu, S Liu, L Zhou, W Tang, T Feng, M Chen, L
Zhan, T Wu, E Hu, Y Jiang, X Bo, G Yu. ggtreeExtra: Compact
visualization of richly annotated phylogenetic data. Molecular Biology
and Evolution. 2021, 38(9):4039-4042. doi: 10.1093/molbev/msab166</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggtree'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biostrings':

    collapse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    collapse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="fu">library</span>(treeio)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>treeio v1.32.0 Learn more at https://yulab-smu.top/contribution-tree-data/

Please cite:

LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR
Jones, T Bradley, H Zhu, Y Guan, Y Jiang, G Yu. treeio: an R package
for phylogenetic tree input and output with richly annotated and
associated data. Molecular Biology and Evolution. 2020, 37(2):599-603.
doi: 10.1093/molbev/msz240</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'treeio'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biostrings':

    mask</code></pre>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>syncom16s_seqs_mltree_nwk <span class="ot">&lt;-</span> treeio<span class="sc">::</span><span class="fu">read.newick</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.nwk"</span>),</span>
<span id="cb34-2"><a href="#cb34-2"></a>                            <span class="at">node.label=</span><span class="st">"support"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Goal is to produce a tree that has for tip labels the Genus assigned by IDTAXA + the strain identifier, with different internal nodes labeled by coarser taxonomic grouping. The clusters assigned by <code>clusterizer</code> will be represented by different colors at the tips.</p>
<p>First lets look at the tree with just the identifiers on the leaves</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>) <span class="sc">+</span> </span>
<span id="cb35-2"><a href="#cb35-2"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb35-3"><a href="#cb35-3"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4"></a>  <span class="fu">geom_treescale</span>(<span class="at">offset=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_seq_cluster_phylo_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Maximum likelihood phylogenetic tree inferred from an alignment of the 16S rRNA gene amplified by Sanger sequencing from the duckweed SynCom. Scale bar at bottom represents 0.1 substitutions per site.
</figcaption>
</figure>
</div>
<p>We can also look at the internal node numbering by rendering the tree as a dataframe</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="co"># the second tibble call is so quarto renders it properly</span></span>
<span id="cb36-2"><a href="#cb36-2"></a>syncom16s_seqs_mltree_nwk <span class="sc">%&gt;%</span> treeio<span class="sc">::</span><span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">tibble</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["parent"],"name":[1],"type":["int"],"align":["right"]},{"label":["node"],"name":[2],"type":["int"],"align":["right"]},{"label":["branch.length"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["label"],"name":[4],"type":["chr"],"align":["left"]},{"label":["support"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"122","2":"1","3":"0.0000003023","4":"HA29","5":"NA"},{"1":"123","2":"2","3":"0.0000002833","4":"HA34","5":"NA"},{"1":"123","2":"3","3":"0.0000002881","4":"HA42","5":"NA"},{"1":"125","2":"4","3":"0.0000002980","4":"TL39","5":"NA"},{"1":"126","2":"5","3":"0.0000003030","4":"TL62","5":"NA"},{"1":"128","2":"6","3":"0.0000002454","4":"TL19","5":"NA"},{"1":"131","2":"7","3":"0.0000002680","4":"TL25","5":"NA"},{"1":"131","2":"8","3":"0.0010944810","4":"TL12","5":"NA"},{"1":"133","2":"9","3":"0.0000002877","4":"TL35","5":"NA"},{"1":"134","2":"10","3":"0.0000002409","4":"TL10","5":"NA"},{"1":"135","2":"11","3":"0.0000002823","4":"TL46","5":"NA"},{"1":"136","2":"12","3":"0.0000002755","4":"TL33","5":"NA"},{"1":"137","2":"13","3":"0.0000002930","4":"TL32","5":"NA"},{"1":"138","2":"14","3":"0.0000002915","4":"TL24","5":"NA"},{"1":"138","2":"15","3":"0.0000002949","4":"TL26","5":"NA"},{"1":"132","2":"16","3":"0.0000000238","4":"TL20","5":"NA"},{"1":"129","2":"17","3":"0.0000000558","4":"TL45","5":"NA"},{"1":"127","2":"18","3":"0.0000003023","4":"TL53","5":"NA"},{"1":"140","2":"19","3":"0.0000003028","4":"HA37","5":"NA"},{"1":"141","2":"20","3":"0.0000003016","4":"HA63","5":"NA"},{"1":"141","2":"21","3":"0.0000003034","4":"HA53","5":"NA"},{"1":"139","2":"22","3":"0.0011223550","4":"HA25","5":"NA"},{"1":"142","2":"23","3":"0.3473705000","4":"SJ02","5":"NA"},{"1":"146","2":"24","3":"0.0085621920","4":"HA18","5":"NA"},{"1":"147","2":"25","3":"0.0173163000","4":"HA27","5":"NA"},{"1":"147","2":"26","3":"0.0497882600","4":"HA69","5":"NA"},{"1":"149","2":"27","3":"0.0247243300","4":"SJ03","5":"NA"},{"1":"149","2":"28","3":"0.0400582200","4":"SJ14","5":"NA"},{"1":"150","2":"29","3":"0.0475276200","4":"HA16","5":"NA"},{"1":"150","2":"30","3":"0.0546802100","4":"HA52","5":"NA"},{"1":"153","2":"31","3":"0.0000002998","4":"HA32","5":"NA"},{"1":"154","2":"32","3":"0.0000002797","4":"HA35","5":"NA"},{"1":"154","2":"33","3":"0.0000002988","4":"HA68","5":"NA"},{"1":"155","2":"34","3":"0.0000003020","4":"HA46","5":"NA"},{"1":"155","2":"35","3":"0.0000003021","4":"HA65","5":"NA"},{"1":"157","2":"36","3":"0.0000003037","4":"HA01","5":"NA"},{"1":"157","2":"37","3":"0.0000003037","4":"HA11","5":"NA"},{"1":"158","2":"38","3":"0.0000003031","4":"HA72","5":"NA"},{"1":"159","2":"39","3":"0.0000003028","4":"HA06","5":"NA"},{"1":"159","2":"40","3":"0.0000003037","4":"HA47","5":"NA"},{"1":"161","2":"41","3":"0.0246131700","4":"HA82","5":"NA"},{"1":"161","2":"42","3":"0.0300392800","4":"TL64","5":"NA"},{"1":"164","2":"43","3":"0.0110978500","4":"HA44","5":"NA"},{"1":"169","2":"44","3":"0.0000003009","4":"HA61","5":"NA"},{"1":"169","2":"45","3":"0.0028810270","4":"HA14","5":"NA"},{"1":"168","2":"46","3":"0.0071653950","4":"HA10","5":"NA"},{"1":"167","2":"47","3":"0.3544444000","4":"TL02","5":"NA"},{"1":"170","2":"48","3":"0.0000002881","4":"SJ20","5":"NA"},{"1":"170","2":"49","3":"0.0000002893","4":"SJ08","5":"NA"},{"1":"171","2":"50","3":"0.0318726100","4":"TL09","5":"NA"},{"1":"171","2":"51","3":"0.0330256500","4":"TL38","5":"NA"},{"1":"173","2":"52","3":"0.0000003019","4":"HA13","5":"NA"},{"1":"173","2":"53","3":"0.0000003022","4":"HA04","5":"NA"},{"1":"175","2":"54","3":"0.0000003023","4":"HA03","5":"NA"},{"1":"175","2":"55","3":"0.0000003025","4":"HA23","5":"NA"},{"1":"177","2":"56","3":"0.0034335010","4":"HA17","5":"NA"},{"1":"178","2":"57","3":"0.0057523240","4":"HA48","5":"NA"},{"1":"179","2":"58","3":"0.0000002880","4":"HA40","5":"NA"},{"1":"179","2":"59","3":"0.0000002893","4":"HA50","5":"NA"},{"1":"180","2":"60","3":"0.0117376200","4":"HA21","5":"NA"},{"1":"182","2":"61","3":"0.0000003015","4":"SJ18","5":"NA"},{"1":"182","2":"62","3":"0.0000003035","4":"HA64","5":"NA"},{"1":"183","2":"63","3":"0.0000003027","4":"HA07","5":"NA"},{"1":"183","2":"64","3":"0.0012321170","4":"HA41","5":"NA"},{"1":"187","2":"65","3":"0.0000002951","4":"TL61","5":"NA"},{"1":"187","2":"66","3":"0.0011357230","4":"TL54","5":"NA"},{"1":"186","2":"67","3":"0.0000003034","4":"TL47","5":"NA"},{"1":"185","2":"68","3":"0.0000003037","4":"TL23","5":"NA"},{"1":"188","2":"69","3":"0.0061230540","4":"TL29","5":"NA"},{"1":"190","2":"70","3":"0.0130998200","4":"HA59","5":"NA"},{"1":"194","2":"71","3":"0.0233276100","4":"TL03","5":"NA"},{"1":"194","2":"72","3":"0.0377324500","4":"HA36","5":"NA"},{"1":"193","2":"73","3":"0.0012215510","4":"SJ17","5":"NA"},{"1":"195","2":"74","3":"0.0000003036","4":"SJ16","5":"NA"},{"1":"195","2":"75","3":"0.0000003037","4":"HA12","5":"NA"},{"1":"191","2":"76","3":"0.0063136800","4":"TL06","5":"NA"},{"1":"197","2":"77","3":"0.0000002870","4":"TL37","5":"NA"},{"1":"197","2":"78","3":"0.0011397930","4":"TL34","5":"NA"},{"1":"201","2":"79","3":"0.0000002871","4":"TL30","5":"NA"},{"1":"203","2":"80","3":"0.0000003003","4":"TL52","5":"NA"},{"1":"203","2":"81","3":"0.0009047834","4":"TL42","5":"NA"},{"1":"202","2":"82","3":"0.0000002943","4":"TL05","5":"NA"},{"1":"204","2":"83","3":"0.0122390800","4":"TL55","5":"NA"},{"1":"204","2":"84","3":"0.0157753600","4":"SJ05","5":"NA"},{"1":"206","2":"85","3":"0.0000002709","4":"TL04","5":"NA"},{"1":"206","2":"86","3":"0.0000002982","4":"TL50","5":"NA"},{"1":"208","2":"87","3":"0.0000000395","4":"TL65","5":"NA"},{"1":"209","2":"88","3":"0.0000000217","4":"TL31","5":"NA"},{"1":"210","2":"89","3":"0.0000000217","4":"TL07","5":"NA"},{"1":"211","2":"90","3":"0.0000000217","4":"TL16","5":"NA"},{"1":"212","2":"91","3":"0.0000000216","4":"TL51","5":"NA"},{"1":"214","2":"92","3":"0.0000000284","4":"TL44","5":"NA"},{"1":"215","2":"93","3":"0.0011435490","4":"TL11","5":"NA"},{"1":"216","2":"94","3":"0.0000002601","4":"TL14","5":"NA"},{"1":"218","2":"95","3":"0.0000003010","4":"TL41","5":"NA"},{"1":"218","2":"96","3":"0.0023091300","4":"TL22","5":"NA"},{"1":"217","2":"97","3":"0.0000002934","4":"TL63","5":"NA"},{"1":"213","2":"98","3":"0.0000000217","4":"TL01","5":"NA"},{"1":"207","2":"99","3":"0.0000006015","4":"TL18","5":"NA"},{"1":"219","2":"100","3":"0.0047454960","4":"SJ07","5":"NA"},{"1":"221","2":"101","3":"0.0011330150","4":"SJ23","5":"NA"},{"1":"222","2":"102","3":"0.0000003004","4":"HA81","5":"NA"},{"1":"226","2":"103","3":"0.0000002869","4":"SJ13","5":"NA"},{"1":"226","2":"104","3":"0.0000002895","4":"SJ12","5":"NA"},{"1":"225","2":"105","3":"0.0000003036","4":"HA02","5":"NA"},{"1":"224","2":"106","3":"0.0000003037","4":"HA39","5":"NA"},{"1":"223","2":"107","3":"0.0011499230","4":"HA30","5":"NA"},{"1":"220","2":"108","3":"0.0032442310","4":"SJ10","5":"NA"},{"1":"228","2":"109","3":"0.0000002952","4":"HA57","5":"NA"},{"1":"230","2":"110","3":"0.0000003017","4":"HA08","5":"NA"},{"1":"230","2":"111","3":"0.0000003035","4":"HA67","5":"NA"},{"1":"229","2":"112","3":"0.0000003025","4":"HA51","5":"NA"},{"1":"232","2":"113","3":"0.0000002641","4":"HA33","5":"NA"},{"1":"233","2":"114","3":"0.0000002958","4":"HA45","5":"NA"},{"1":"233","2":"115","3":"0.0137098400","4":"HA56","5":"NA"},{"1":"235","2":"116","3":"0.0000002792","4":"HA28","5":"NA"},{"1":"235","2":"117","3":"0.0000002886","4":"HA80","5":"NA"},{"1":"234","2":"118","3":"0.0444020500","4":"TL58","5":"NA"},{"1":"119","2":"119","3":"NA","4":"NA","5":"NA"},{"1":"119","2":"120","3":"0.0295668500","4":"NA","5":"1.00"},{"1":"120","2":"121","3":"0.1344812000","4":"NA","5":"1.00"},{"1":"121","2":"122","3":"0.7207153000","4":"NA","5":"1.00"},{"1":"122","2":"123","3":"0.0000000634","4":"NA","5":"0.33"},{"1":"121","2":"124","3":"0.2050687000","4":"NA","5":"1.00"},{"1":"124","2":"125","3":"0.2496145000","4":"NA","5":"1.00"},{"1":"125","2":"126","3":"0.0000000329","4":"NA","5":"0.33"},{"1":"126","2":"127","3":"0.0002296142","4":"NA","5":"0.33"},{"1":"127","2":"128","3":"0.0003592300","4":"NA","5":"0.34"},{"1":"128","2":"129","3":"0.0004506589","4":"NA","5":"0.33"},{"1":"129","2":"130","3":"0.0000154758","4":"NA","5":"0.33"},{"1":"130","2":"131","3":"0.0010917030","4":"NA","5":"1.00"},{"1":"130","2":"132","3":"0.0000095850","4":"NA","5":"0.33"},{"1":"132","2":"133","3":"0.0000131253","4":"NA","5":"0.33"},{"1":"133","2":"134","3":"0.0000097200","4":"NA","5":"0.33"},{"1":"134","2":"135","3":"0.0000097822","4":"NA","5":"0.33"},{"1":"135","2":"136","3":"0.0000103078","4":"NA","5":"0.33"},{"1":"136","2":"137","3":"0.0000002929","4":"NA","5":"0.33"},{"1":"137","2":"138","3":"0.0000002585","4":"NA","5":"0.33"},{"1":"124","2":"139","3":"0.2406384000","4":"NA","5":"1.00"},{"1":"139","2":"140","3":"0.0000016372","4":"NA","5":"0.33"},{"1":"140","2":"141","3":"0.0000003028","4":"NA","5":"0.33"},{"1":"120","2":"142","3":"0.0340158900","4":"NA","5":"0.56"},{"1":"142","2":"143","3":"0.1863477000","4":"NA","5":"1.00"},{"1":"143","2":"144","3":"0.1694784000","4":"NA","5":"1.00"},{"1":"144","2":"145","3":"0.0827489600","4":"NA","5":"1.00"},{"1":"145","2":"146","3":"0.0964129500","4":"NA","5":"1.00"},{"1":"146","2":"147","3":"0.0140819500","4":"NA","5":"0.96"},{"1":"145","2":"148","3":"0.0718772200","4":"NA","5":"1.00"},{"1":"148","2":"149","3":"0.0312316500","4":"NA","5":"1.00"},{"1":"148","2":"150","3":"0.0623829900","4":"NA","5":"1.00"},{"1":"144","2":"151","3":"0.0531105000","4":"NA","5":"1.00"},{"1":"151","2":"152","3":"0.0292747500","4":"NA","5":"1.00"},{"1":"152","2":"153","3":"0.0054894770","4":"NA","5":"0.97"},{"1":"153","2":"154","3":"0.0000000275","4":"NA","5":"0.33"},{"1":"152","2":"155","3":"0.0021253260","4":"NA","5":"0.61"},{"1":"151","2":"156","3":"0.0602668400","4":"NA","5":"1.00"},{"1":"156","2":"157","3":"0.0000002407","4":"NA","5":"0.33"},{"1":"156","2":"158","3":"0.0000002870","4":"NA","5":"0.33"},{"1":"158","2":"159","3":"0.0000003031","4":"NA","5":"0.33"},{"1":"143","2":"160","3":"0.0523796300","4":"NA","5":"1.00"},{"1":"160","2":"161","3":"0.1673914000","4":"NA","5":"1.00"},{"1":"160","2":"162","3":"0.0274929300","4":"NA","5":"0.52"},{"1":"162","2":"163","3":"0.1501122000","4":"NA","5":"1.00"},{"1":"163","2":"164","3":"0.1195508000","4":"NA","5":"1.00"},{"1":"164","2":"165","3":"0.0225670100","4":"NA","5":"1.00"},{"1":"165","2":"166","3":"0.0153033000","4":"NA","5":"0.99"},{"1":"166","2":"167","3":"0.0242769400","4":"NA","5":"0.70"},{"1":"167","2":"168","3":"0.0308099400","4":"NA","5":"0.74"},{"1":"168","2":"169","3":"0.0030935450","4":"NA","5":"0.37"},{"1":"166","2":"170","3":"0.0168779900","4":"NA","5":"1.00"},{"1":"165","2":"171","3":"0.0248347800","4":"NA","5":"1.00"},{"1":"163","2":"172","3":"0.0658691400","4":"NA","5":"1.00"},{"1":"172","2":"173","3":"0.0000003036","4":"NA","5":"0.33"},{"1":"172","2":"174","3":"0.0011466660","4":"NA","5":"0.67"},{"1":"174","2":"175","3":"0.0000003025","4":"NA","5":"0.33"},{"1":"174","2":"176","3":"0.0084532630","4":"NA","5":"1.00"},{"1":"176","2":"177","3":"0.0035324020","4":"NA","5":"0.85"},{"1":"177","2":"178","3":"0.0048174210","4":"NA","5":"0.55"},{"1":"178","2":"179","3":"0.0047444900","4":"NA","5":"0.98"},{"1":"176","2":"180","3":"0.0081387160","4":"NA","5":"1.00"},{"1":"180","2":"181","3":"0.0034620750","4":"NA","5":"0.98"},{"1":"181","2":"182","3":"0.0083832770","4":"NA","5":"1.00"},{"1":"181","2":"183","3":"0.0124521300","4":"NA","5":"1.00"},{"1":"162","2":"184","3":"0.0904250700","4":"NA","5":"1.00"},{"1":"184","2":"185","3":"0.0296197400","4":"NA","5":"1.00"},{"1":"185","2":"186","3":"0.0000000612","4":"NA","5":"0.33"},{"1":"186","2":"187","3":"0.0000003036","4":"NA","5":"0.33"},{"1":"184","2":"188","3":"0.0246482600","4":"NA","5":"1.00"},{"1":"188","2":"189","3":"0.0019484520","4":"NA","5":"0.50"},{"1":"189","2":"190","3":"0.0088157110","4":"NA","5":"1.00"},{"1":"190","2":"191","3":"0.0124073300","4":"NA","5":"1.00"},{"1":"191","2":"192","3":"0.0120418500","4":"NA","5":"0.41"},{"1":"192","2":"193","3":"0.1989057000","4":"NA","5":"1.00"},{"1":"193","2":"194","3":"0.0306488800","4":"NA","5":"1.00"},{"1":"192","2":"195","3":"0.0088394530","4":"NA","5":"0.37"},{"1":"189","2":"196","3":"0.0035110010","4":"NA","5":"1.00"},{"1":"196","2":"197","3":"0.0000002645","4":"NA","5":"0.33"},{"1":"196","2":"198","3":"0.0214573000","4":"NA","5":"1.00"},{"1":"198","2":"199","3":"0.0023250550","4":"NA","5":"0.87"},{"1":"199","2":"200","3":"0.0113730300","4":"NA","5":"1.00"},{"1":"200","2":"201","3":"0.0022491180","4":"NA","5":"0.58"},{"1":"201","2":"202","3":"0.0000001329","4":"NA","5":"0.33"},{"1":"202","2":"203","3":"0.0002201480","4":"NA","5":"0.33"},{"1":"200","2":"204","3":"0.0184862600","4":"NA","5":"1.00"},{"1":"199","2":"205","3":"0.0015586830","4":"NA","5":"0.92"},{"1":"205","2":"206","3":"0.0000002795","4":"NA","5":"0.33"},{"1":"205","2":"207","3":"0.0000000293","4":"NA","5":"0.33"},{"1":"207","2":"208","3":"0.0000002942","4":"NA","5":"0.33"},{"1":"208","2":"209","3":"0.0000002718","4":"NA","5":"0.33"},{"1":"209","2":"210","3":"0.0000003020","4":"NA","5":"0.33"},{"1":"210","2":"211","3":"0.0000002911","4":"NA","5":"0.33"},{"1":"211","2":"212","3":"0.0000002487","4":"NA","5":"0.33"},{"1":"212","2":"213","3":"0.0000002981","4":"NA","5":"0.33"},{"1":"213","2":"214","3":"0.0000002997","4":"NA","5":"0.33"},{"1":"214","2":"215","3":"0.0000002797","4":"NA","5":"0.33"},{"1":"215","2":"216","3":"0.0000000340","4":"NA","5":"0.33"},{"1":"216","2":"217","3":"0.0030062260","4":"NA","5":"1.00"},{"1":"217","2":"218","3":"0.0000000279","4":"NA","5":"0.33"},{"1":"198","2":"219","3":"0.0132719800","4":"NA","5":"1.00"},{"1":"219","2":"220","3":"0.0033889080","4":"NA","5":"0.99"},{"1":"220","2":"221","3":"0.0000001793","4":"NA","5":"0.33"},{"1":"221","2":"222","3":"0.0000003014","4":"NA","5":"0.33"},{"1":"222","2":"223","3":"0.0012263900","4":"NA","5":"1.00"},{"1":"223","2":"224","3":"0.0000000804","4":"NA","5":"0.33"},{"1":"224","2":"225","3":"0.0012282290","4":"NA","5":"1.00"},{"1":"225","2":"226","3":"0.0000000660","4":"NA","5":"0.33"},{"1":"119","2":"227","3":"0.9703983000","4":"NA","5":"1.00"},{"1":"227","2":"228","3":"0.0000002575","4":"NA","5":"0.33"},{"1":"228","2":"229","3":"0.0000003025","4":"NA","5":"0.33"},{"1":"229","2":"230","3":"0.0000003034","4":"NA","5":"0.33"},{"1":"227","2":"231","3":"0.0012570360","4":"NA","5":"0.48"},{"1":"231","2":"232","3":"0.0000000308","4":"NA","5":"0.33"},{"1":"232","2":"233","3":"0.0000002529","4":"NA","5":"0.33"},{"1":"231","2":"234","3":"0.0000002507","4":"NA","5":"0.33"},{"1":"234","2":"235","3":"0.0000002552","4":"NA","5":"0.33"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>First we need to get our naming scheme set up</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>newtipnames <span class="ot">&lt;-</span> syncom16s_idtx_lineage_info <span class="sc">%&gt;%</span> </span>
<span id="cb37-2"><a href="#cb37-2"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(genus, <span class="st">"(^[:alpha:]*).*"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-3"><a href="#cb37-3"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(genus) <span class="sc">|</span> genus <span class="sc">==</span> <span class="st">"unclassified"</span> <span class="sc">~</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(ncbi_best_species, <span class="st">"^[:alpha:]*"</span>),</span>
<span id="cb37-4"><a href="#cb37-4"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> genus)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-5"><a href="#cb37-5"></a>  <span class="fu">mutate</span>(<span class="at">newname =</span> <span class="fu">paste</span>(genus, id)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-6"><a href="#cb37-6"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> <span class="fu">paste0</span>(<span class="st">"c"</span>, cluster)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-7"><a href="#cb37-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(id, newname, cluster, genus, order, family)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>First we need to generate a color palette that lets us distinguish the different colors</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb38-2"><a href="#cb38-2"></a><span class="fu">set.seed</span>(<span class="dv">341234</span>)</span>
<span id="cb38-3"><a href="#cb38-3"></a>mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">max</span>(syncom16s_idtx_lineage_info<span class="sc">$</span>cluster), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb38-4"><a href="#cb38-4"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we need to identify the internal nodes that we want to highlight. We can do this using the <code>MRCA</code> function from ggtree</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 184</code></pre>
</div>
</div>
<p>Code to plot the tree</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>offset <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb41-2"><a href="#cb41-2"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>)</span>
<span id="cb41-3"><a href="#cb41-3"></a></span>
<span id="cb41-4"><a href="#cb41-4"></a>p2 <span class="ot">&lt;-</span> p1 <span class="sc">%&lt;+%</span> newtipnames <span class="sc">+</span> </span>
<span id="cb41-5"><a href="#cb41-5"></a>  <span class="fu">geom_tiplab</span>(<span class="fu">aes</span>(<span class="at">label =</span> newname), <span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb41-6"><a href="#cb41-6"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> cluster), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal, <span class="at">guide=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb41-8"><a href="#cb41-8"></a>  <span class="co"># Pseudomonadaceae </span></span>
<span id="cb41-9"><a href="#cb41-9"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>), </span>
<span id="cb41-10"><a href="#cb41-10"></a>                <span class="at">label=</span><span class="st">"(γ) Pseudomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-11"><a href="#cb41-11"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-12"><a href="#cb41-12"></a>  <span class="co"># Aeromonadaceae</span></span>
<span id="cb41-13"><a href="#cb41-13"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA41"</span>, <span class="st">"HA04"</span>), </span>
<span id="cb41-14"><a href="#cb41-14"></a>                <span class="at">label=</span><span class="st">"(γ) Aeromonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-15"><a href="#cb41-15"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-16"><a href="#cb41-16"></a>  <span class="co"># Enterobacteriaceae</span></span>
<span id="cb41-17"><a href="#cb41-17"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA14"</span>, <span class="st">"HA44"</span>), </span>
<span id="cb41-18"><a href="#cb41-18"></a>                <span class="at">label=</span><span class="st">"(γ) Enterobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-19"><a href="#cb41-19"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-20"><a href="#cb41-20"></a>  <span class="co"># Moraxellaceae</span></span>
<span id="cb41-21"><a href="#cb41-21"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL64"</span>, <span class="st">"HA82"</span>), </span>
<span id="cb41-22"><a href="#cb41-22"></a>                <span class="at">label=</span><span class="st">"(γ) Moraxellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-23"><a href="#cb41-23"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-24"><a href="#cb41-24"></a>  <span class="co"># Chromobacteriaceae</span></span>
<span id="cb41-25"><a href="#cb41-25"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA47"</span>, <span class="st">"HA65"</span>), </span>
<span id="cb41-26"><a href="#cb41-26"></a>                <span class="at">label=</span><span class="st">"(γ) Chromobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-27"><a href="#cb41-27"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-28"><a href="#cb41-28"></a>  <span class="co"># Burkholderiaceae</span></span>
<span id="cb41-29"><a href="#cb41-29"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA52"</span>, <span class="st">"HA18"</span>), </span>
<span id="cb41-30"><a href="#cb41-30"></a>                <span class="at">label=</span><span class="st">"(γ) Burkholderiaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-31"><a href="#cb41-31"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-32"><a href="#cb41-32"></a>  <span class="co"># Sphingomonadaceae</span></span>
<span id="cb41-33"><a href="#cb41-33"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="dv">23</span>, </span>
<span id="cb41-34"><a href="#cb41-34"></a>                <span class="at">label=</span><span class="st">"(α) Sphingomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-35"><a href="#cb41-35"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-36"><a href="#cb41-36"></a>  <span class="co"># Streptococcaceae</span></span>
<span id="cb41-37"><a href="#cb41-37"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL12"</span>, <span class="st">"TL39"</span>), </span>
<span id="cb41-38"><a href="#cb41-38"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Streptococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-39"><a href="#cb41-39"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-40"><a href="#cb41-40"></a>  <span class="co"># Exiguobacteriaceae</span></span>
<span id="cb41-41"><a href="#cb41-41"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA63"</span>, <span class="st">"HA25"</span>), </span>
<span id="cb41-42"><a href="#cb41-42"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Exiguobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-43"><a href="#cb41-43"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb41-44"><a href="#cb41-44"></a>  <span class="co"># Deinococcaceae</span></span>
<span id="cb41-45"><a href="#cb41-45"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA42"</span>, <span class="st">"HA29"</span>), </span>
<span id="cb41-46"><a href="#cb41-46"></a>                <span class="at">label=</span><span class="st">"(Deinococci) Deinococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-47"><a href="#cb41-47"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.2</span>) <span class="sc">+</span></span>
<span id="cb41-48"><a href="#cb41-48"></a>  <span class="co"># Weeksellaceae</span></span>
<span id="cb41-49"><a href="#cb41-49"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA56"</span>, <span class="st">"HA57"</span>), </span>
<span id="cb41-50"><a href="#cb41-50"></a>                <span class="at">label=</span><span class="st">"(Bacteroidia) Weeksellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb41-51"><a href="#cb41-51"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.4</span>) <span class="sc">+</span></span>
<span id="cb41-52"><a href="#cb41-52"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-02" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_seq_cluster_phylo_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: As in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>, but isolates have been labeled with their inferred genus (text) as well as their broader taxonomic class (right, vertical lines). Colors represent sequences clusters based on a 99% identity threshold (i.e., nodes of the same color have 16S rRNA gene sequences that are &gt; 99% similar)
</figcaption>
</figure>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb42" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb42-1"><a href="#cb42-1"></a><span class="co">---</span></span>
<span id="cb42-2"><a href="#cb42-2"></a><span class="an">title:</span><span class="co"> "Clustering/phylogenetics of 16S rRNA sequences from duckweed SynCom"</span></span>
<span id="cb42-3"><a href="#cb42-3"></a><span class="an">subtitle:</span><span class="co"> ""</span></span>
<span id="cb42-4"><a href="#cb42-4"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb42-5"><a href="#cb42-5"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb42-6"><a href="#cb42-6"></a><span class="an">abstract:</span><span class="co"> "Various operations to group the isolates into clusters using clustering algorithms and phylogenetic inference. At the end a phylogenetic tree for the isolates is plotting with associated taxonomic information as well as clustering identity based on shared kmers."</span></span>
<span id="cb42-7"><a href="#cb42-7"></a><span class="co">---</span></span>
<span id="cb42-8"><a href="#cb42-8"></a></span>
<span id="cb42-9"><a href="#cb42-9"></a><span class="fu"># Setup</span></span>
<span id="cb42-10"><a href="#cb42-10"></a></span>
<span id="cb42-11"><a href="#cb42-11"></a>Loads required libraries</span>
<span id="cb42-12"><a href="#cb42-12"></a></span>
<span id="cb42-15"><a href="#cb42-15"></a><span class="in">```{r}</span></span>
<span id="cb42-16"><a href="#cb42-16"></a><span class="co">#| output: false</span></span>
<span id="cb42-17"><a href="#cb42-17"></a><span class="fu">options</span>(<span class="at">repos =</span> <span class="fu">c</span>(<span class="at">CRAN =</span> <span class="st">"https://packagemanager.posit.co/cran/__linux__/jammy/latest"</span>))</span>
<span id="cb42-18"><a href="#cb42-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb42-19"><a href="#cb42-19"></a><span class="fu">library</span>(here)</span>
<span id="cb42-20"><a href="#cb42-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb42-21"><a href="#cb42-21"></a><span class="fu">library</span>(DECIPHER)</span>
<span id="cb42-22"><a href="#cb42-22"></a><span class="in">```</span></span>
<span id="cb42-23"><a href="#cb42-23"></a></span>
<span id="cb42-24"><a href="#cb42-24"></a><span class="fu"># Read and format</span></span>
<span id="cb42-25"><a href="#cb42-25"></a></span>
<span id="cb42-26"><a href="#cb42-26"></a>Read 16S rRNA fasta sequences from the prior step.</span>
<span id="cb42-27"><a href="#cb42-27"></a></span>
<span id="cb42-30"><a href="#cb42-30"></a><span class="in">```{r}</span></span>
<span id="cb42-31"><a href="#cb42-31"></a>syncom16s_fna <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"16S_rRNA_SynCom.fna"</span>)</span>
<span id="cb42-32"><a href="#cb42-32"></a><span class="in">```</span></span>
<span id="cb42-33"><a href="#cb42-33"></a></span>
<span id="cb42-34"><a href="#cb42-34"></a>Convert to a DNA stringset object</span>
<span id="cb42-35"><a href="#cb42-35"></a></span>
<span id="cb42-38"><a href="#cb42-38"></a><span class="in">```{r}</span></span>
<span id="cb42-39"><a href="#cb42-39"></a>syncom16s_seqs <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(syncom16s_fna)</span>
<span id="cb42-40"><a href="#cb42-40"></a><span class="in">```</span></span>
<span id="cb42-41"><a href="#cb42-41"></a></span>
<span id="cb42-42"><a href="#cb42-42"></a><span class="fu"># Redundancy</span></span>
<span id="cb42-43"><a href="#cb42-43"></a></span>
<span id="cb42-44"><a href="#cb42-44"></a>All sequences are unique which is expected</span>
<span id="cb42-45"><a href="#cb42-45"></a></span>
<span id="cb42-48"><a href="#cb42-48"></a><span class="in">```{r}</span></span>
<span id="cb42-49"><a href="#cb42-49"></a><span class="fu">duplicated</span>(syncom16s_seqs)</span>
<span id="cb42-50"><a href="#cb42-50"></a><span class="in">```</span></span>
<span id="cb42-51"><a href="#cb42-51"></a></span>
<span id="cb42-52"><a href="#cb42-52"></a><span class="fu"># Cluster</span></span>
<span id="cb42-53"><a href="#cb42-53"></a></span>
<span id="cb42-56"><a href="#cb42-56"></a><span class="in">```{r}</span></span>
<span id="cb42-57"><a href="#cb42-57"></a><span class="fu">set.seed</span>(<span class="dv">36742</span>)</span>
<span id="cb42-58"><a href="#cb42-58"></a>syncom16s_clusters <span class="ot">&lt;-</span> <span class="fu">Clusterize</span>(syncom16s_seqs,</span>
<span id="cb42-59"><a href="#cb42-59"></a>                       <span class="at">cutoff=</span><span class="fl">0.01</span>,     <span class="co"># at ~ species level (ie 99% identity)</span></span>
<span id="cb42-60"><a href="#cb42-60"></a>                       <span class="at">minCoverage=</span><span class="fl">0.5</span>, <span class="co"># &gt; 50% coverage</span></span>
<span id="cb42-61"><a href="#cb42-61"></a>                       <span class="at">processors=</span><span class="cn">NULL</span>) <span class="co"># use all CPUs</span></span>
<span id="cb42-62"><a href="#cb42-62"></a><span class="co"># reset random seed</span></span>
<span id="cb42-63"><a href="#cb42-63"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span>
<span id="cb42-64"><a href="#cb42-64"></a><span class="in">```</span></span>
<span id="cb42-65"><a href="#cb42-65"></a></span>
<span id="cb42-66"><a href="#cb42-66"></a><span class="fu"># Classify (IDTAXA)</span></span>
<span id="cb42-67"><a href="#cb42-67"></a></span>
<span id="cb42-68"><a href="#cb42-68"></a>Using</span>
<span id="cb42-69"><a href="#cb42-69"></a><span class="co">[</span><span class="ot">IDTAXA</span><span class="co">](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0521-5)</span></span>
<span id="cb42-70"><a href="#cb42-70"></a>following the [approach shown</span>
<span id="cb42-71"><a href="#cb42-71"></a>here](https://www2.decipher.codes/ClassifyOrganisms.html).</span>
<span id="cb42-72"><a href="#cb42-72"></a></span>
<span id="cb42-75"><a href="#cb42-75"></a><span class="in">```{r}</span></span>
<span id="cb42-76"><a href="#cb42-76"></a><span class="co">#| eval: false</span></span>
<span id="cb42-77"><a href="#cb42-77"></a><span class="co">#| include: true</span></span>
<span id="cb42-78"><a href="#cb42-78"></a><span class="fu">set.seed</span>(<span class="dv">234676</span>)</span>
<span id="cb42-79"><a href="#cb42-79"></a></span>
<span id="cb42-80"><a href="#cb42-80"></a><span class="co"># load the training data set</span></span>
<span id="cb42-81"><a href="#cb42-81"></a><span class="fu">load</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"idtaxa"</span>, <span class="st">"GTDB_r226-mod_April2025.RData"</span>))</span>
<span id="cb42-82"><a href="#cb42-82"></a></span>
<span id="cb42-83"><a href="#cb42-83"></a><span class="co"># classify the sequences</span></span>
<span id="cb42-84"><a href="#cb42-84"></a>syncom16s_ids <span class="ot">&lt;-</span> <span class="fu">IdTaxa</span>(syncom16s_seqs,</span>
<span id="cb42-85"><a href="#cb42-85"></a>   trainingSet,</span>
<span id="cb42-86"><a href="#cb42-86"></a>   <span class="at">strand=</span><span class="st">"both"</span>, <span class="co"># or "top" if same as trainingSet</span></span>
<span id="cb42-87"><a href="#cb42-87"></a>   <span class="at">threshold=</span><span class="dv">60</span>, <span class="co"># 60 (cautious) or 50 (sensible)</span></span>
<span id="cb42-88"><a href="#cb42-88"></a>   <span class="at">processors=</span><span class="cn">NULL</span>) <span class="co"># use all available processors</span></span>
<span id="cb42-89"><a href="#cb42-89"></a><span class="co"># reset random seed</span></span>
<span id="cb42-90"><a href="#cb42-90"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span>
<span id="cb42-91"><a href="#cb42-91"></a><span class="in">```</span></span>
<span id="cb42-92"><a href="#cb42-92"></a></span>
<span id="cb42-95"><a href="#cb42-95"></a><span class="in">```{r}</span></span>
<span id="cb42-96"><a href="#cb42-96"></a><span class="co">#| eval: false</span></span>
<span id="cb42-97"><a href="#cb42-97"></a><span class="co">#| include: false</span></span>
<span id="cb42-98"><a href="#cb42-98"></a><span class="fu">write_rds</span>(syncom16s_ids, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_ids.rds"</span>))</span>
<span id="cb42-99"><a href="#cb42-99"></a><span class="in">```</span></span>
<span id="cb42-100"><a href="#cb42-100"></a></span>
<span id="cb42-103"><a href="#cb42-103"></a><span class="in">```{r}</span></span>
<span id="cb42-104"><a href="#cb42-104"></a><span class="co">#| eval: true</span></span>
<span id="cb42-105"><a href="#cb42-105"></a><span class="co">#| include: false</span></span>
<span id="cb42-106"><a href="#cb42-106"></a>syncom16s_ids <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_ids.rds"</span>))</span>
<span id="cb42-107"><a href="#cb42-107"></a><span class="in">```</span></span>
<span id="cb42-108"><a href="#cb42-108"></a></span>
<span id="cb42-111"><a href="#cb42-111"></a><span class="in">```{r}</span></span>
<span id="cb42-112"><a href="#cb42-112"></a>syncom16s_idtx <span class="ot">&lt;-</span> <span class="fu">imap</span>(</span>
<span id="cb42-113"><a href="#cb42-113"></a>  syncom16s_ids,</span>
<span id="cb42-114"><a href="#cb42-114"></a>  \(x, idx) <span class="fu">tibble</span>(<span class="at">a =</span> x<span class="sc">$</span>rank, <span class="at">b =</span> <span class="fu">paste</span>(x<span class="sc">$</span>taxon,<span class="st">" ("</span>,<span class="fu">round</span>(x<span class="sc">$</span>confidence, <span class="at">digits =</span> <span class="dv">1</span>),<span class="st">"%)"</span>, <span class="at">sep =</span> <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-115"><a href="#cb42-115"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> a, <span class="at">values_from =</span> b) <span class="sc">%&gt;%</span></span>
<span id="cb42-116"><a href="#cb42-116"></a>    <span class="fu">mutate</span>(<span class="at">id =</span> idx)</span>
<span id="cb42-117"><a href="#cb42-117"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb42-118"><a href="#cb42-118"></a>  <span class="fu">list_rbind</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb42-119"><a href="#cb42-119"></a>  <span class="fu">relocate</span>(id)</span>
<span id="cb42-120"><a href="#cb42-120"></a><span class="in">```</span></span>
<span id="cb42-121"><a href="#cb42-121"></a></span>
<span id="cb42-122"><a href="#cb42-122"></a><span class="fu"># Classify (Blastn)</span></span>
<span id="cb42-123"><a href="#cb42-123"></a></span>
<span id="cb42-124"><a href="#cb42-124"></a>These steps need to be run outside of R/Rstudio</span>
<span id="cb42-125"><a href="#cb42-125"></a></span>
<span id="cb42-126"><a href="#cb42-126"></a>First we need to</span>
<span id="cb42-127"><a href="#cb42-127"></a><span class="co">[</span><span class="ot">download</span><span class="co">](https://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST/)</span> and</span>
<span id="cb42-128"><a href="#cb42-128"></a><span class="co">[</span><span class="ot">install</span><span class="co">](https://www.ncbi.nlm.nih.gov/books/NBK52640/)</span> NCBI blast+</span>
<span id="cb42-129"><a href="#cb42-129"></a>suite (I am using v2.16.0) and</span>
<span id="cb42-130"><a href="#cb42-130"></a><span class="co">[</span><span class="ot">download</span><span class="co">](https://ftp.ncbi.nlm.nih.gov/blast/db/)</span> the 16S_ribosomal_RNA</span>
<span id="cb42-131"><a href="#cb42-131"></a>database.</span>
<span id="cb42-132"><a href="#cb42-132"></a></span>
<span id="cb42-133"><a href="#cb42-133"></a><span class="in">```</span></span>
<span id="cb42-134"><a href="#cb42-134"></a><span class="in">blastn -num_threads 12 -db 16S_ribosomal_RNA -query 16S_rRNA_SynCom.fna -dust no -max_target_seqs 5 -out 16S_rRNA_SynCom_v_blast16SDB.tsv -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore staxid"</span></span>
<span id="cb42-135"><a href="#cb42-135"></a><span class="in">```</span></span>
<span id="cb42-136"><a href="#cb42-136"></a></span>
<span id="cb42-137"><a href="#cb42-137"></a>The last column contains the taxonomy ID from the NCBI 16S Ribosomal</span>
<span id="cb42-138"><a href="#cb42-138"></a>database. We will just take the taxonomic lineage of the best hit</span>
<span id="cb42-139"><a href="#cb42-139"></a></span>
<span id="cb42-140"><a href="#cb42-140"></a><span class="in">```</span></span>
<span id="cb42-141"><a href="#cb42-141"></a><span class="in">csvtk tab2csv 16S_rRNA_SynCom_v_blast16SDB.tsv | csvtk cut -f1,13 | csvtk spread -k1 -v2 | sed 's/^,//g' | csvtk transpose | cut -d ";" -f1 | csvtk cut -f2,1 | csvtk csv2tab &gt; a</span></span>
<span id="cb42-142"><a href="#cb42-142"></a><span class="in">csvtk tab2csv 16S_rRNA_SynCom_v_blast16SDB.tsv | csvtk cut -f1,13 | csvtk spread -k1 -v2 | sed 's/^,//g' | csvtk transpose | cut -d ";" -f1 | csvtk cut -f2 | taxonkit lineage -n -L &gt; b</span></span>
<span id="cb42-143"><a href="#cb42-143"></a><span class="in"># combine into one file</span></span>
<span id="cb42-144"><a href="#cb42-144"></a><span class="in">paste a b &gt; blastn_lineage_best_hit.tsv</span></span>
<span id="cb42-145"><a href="#cb42-145"></a><span class="in">rm a b</span></span>
<span id="cb42-146"><a href="#cb42-146"></a><span class="in">```</span></span>
<span id="cb42-147"><a href="#cb42-147"></a></span>
<span id="cb42-148"><a href="#cb42-148"></a>Now read this file back into R</span>
<span id="cb42-149"><a href="#cb42-149"></a></span>
<span id="cb42-152"><a href="#cb42-152"></a><span class="in">```{r}</span></span>
<span id="cb42-153"><a href="#cb42-153"></a>syncom16s_blastn <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"blastn_lineage_best_hit.tsv"</span>),</span>
<span id="cb42-154"><a href="#cb42-154"></a>                <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-155"><a href="#cb42-155"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">id =</span> X2, <span class="at">ncbi_taxid =</span> X1, <span class="at">ncbi_best_species =</span> X4)</span>
<span id="cb42-156"><a href="#cb42-156"></a><span class="in">```</span></span>
<span id="cb42-157"><a href="#cb42-157"></a></span>
<span id="cb42-158"><a href="#cb42-158"></a><span class="fu"># Combine clustering and classification</span></span>
<span id="cb42-159"><a href="#cb42-159"></a></span>
<span id="cb42-162"><a href="#cb42-162"></a><span class="in">```{r}</span></span>
<span id="cb42-163"><a href="#cb42-163"></a>syncom16s_idtx_lineage_info <span class="ot">&lt;-</span> syncom16s_idtx <span class="sc">%&gt;%</span> </span>
<span id="cb42-164"><a href="#cb42-164"></a>  <span class="fu">left_join</span>(<span class="fu">rownames_to_column</span>(syncom16s_clusters, <span class="at">var =</span> <span class="st">"id"</span>), <span class="at">by =</span> <span class="fu">join_by</span>(id)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-165"><a href="#cb42-165"></a>  <span class="fu">left_join</span>(syncom16s_blastn, <span class="at">by =</span> <span class="fu">join_by</span>(id)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-166"><a href="#cb42-166"></a>  <span class="fu">relocate</span>(ncbi_best_species, <span class="at">.after=</span><span class="st">"genus"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-167"><a href="#cb42-167"></a>  <span class="fu">relocate</span>(cluster, <span class="at">.after=</span><span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-168"><a href="#cb42-168"></a>  <span class="fu">arrange</span>(cluster)</span>
<span id="cb42-169"><a href="#cb42-169"></a><span class="in">```</span></span>
<span id="cb42-170"><a href="#cb42-170"></a></span>
<span id="cb42-171"><a href="#cb42-171"></a><span class="fu"># 16S rRNA sequence alignment</span></span>
<span id="cb42-172"><a href="#cb42-172"></a></span>
<span id="cb42-173"><a href="#cb42-173"></a>We start with DNA sequences and convert to RNA so that DECIPHER can leverage structural information from the RNA</span>
<span id="cb42-174"><a href="#cb42-174"></a></span>
<span id="cb42-177"><a href="#cb42-177"></a><span class="in">```{r}</span></span>
<span id="cb42-178"><a href="#cb42-178"></a>syncom16s_seqs_rna <span class="ot">&lt;-</span> <span class="fu">RNAStringSet</span>(syncom16s_seqs)</span>
<span id="cb42-179"><a href="#cb42-179"></a><span class="in">```</span></span>
<span id="cb42-180"><a href="#cb42-180"></a></span>
<span id="cb42-181"><a href="#cb42-181"></a>Align the sequences</span>
<span id="cb42-182"><a href="#cb42-182"></a></span>
<span id="cb42-183"><a href="#cb42-183"></a>From [DECIPHER documentation on sequence</span>
<span id="cb42-184"><a href="#cb42-184"></a>alignment](https://bioconductor.org/packages/devel/bioc/vignettes/DECIPHER/inst/doc/ArtOfAlignmentInR.pdf)</span>
<span id="cb42-185"><a href="#cb42-185"></a></span>
<span id="cb42-186"><a href="#cb42-186"></a><span class="at">&gt; 5.2 Example: Non-coding RNA sequences Much like proteins, non-coding</span></span>
<span id="cb42-187"><a href="#cb42-187"></a><span class="at">&gt; RNAs often have a conserved secondary structure that can be used to</span></span>
<span id="cb42-188"><a href="#cb42-188"></a><span class="at">&gt; improve their alignment. The PredictDBN function will predict base</span></span>
<span id="cb42-189"><a href="#cb42-189"></a><span class="at">&gt; pairings from a sequence alignment by calculating the mutual</span></span>
<span id="cb42-190"><a href="#cb42-190"></a><span class="at">&gt; information between pairs of positions. If RNA sequences are given as</span></span>
<span id="cb42-191"><a href="#cb42-191"></a><span class="at">&gt; input, AlignSeqs will automatically use the output of PredictDBN to</span></span>
<span id="cb42-192"><a href="#cb42-192"></a><span class="at">&gt; iteratively improve the alignment. Providing an RNAStringSet also</span></span>
<span id="cb42-193"><a href="#cb42-193"></a><span class="at">&gt; causes single-base and double-base substitution matrices to be used,</span></span>
<span id="cb42-194"><a href="#cb42-194"></a><span class="at">&gt; and is preferable to providing a DNAStringSet when the sequences are</span></span>
<span id="cb42-195"><a href="#cb42-195"></a><span class="at">&gt; non-coding RNA. The type of the input sequences can easily be</span></span>
<span id="cb42-196"><a href="#cb42-196"></a><span class="at">&gt; converted to RNA, as shown below.</span></span>
<span id="cb42-197"><a href="#cb42-197"></a></span>
<span id="cb42-200"><a href="#cb42-200"></a><span class="in">```{r}</span></span>
<span id="cb42-201"><a href="#cb42-201"></a><span class="co">#| eval: false</span></span>
<span id="cb42-202"><a href="#cb42-202"></a><span class="co">#| include: true</span></span>
<span id="cb42-203"><a href="#cb42-203"></a><span class="fu">set.seed</span>(<span class="dv">2342</span>)</span>
<span id="cb42-204"><a href="#cb42-204"></a>syncom16s_seqs_rna_aligned <span class="ot">&lt;-</span> <span class="fu">AlignSeqs</span>(syncom16s_seqs_rna)</span>
<span id="cb42-205"><a href="#cb42-205"></a><span class="co"># reset random seed</span></span>
<span id="cb42-206"><a href="#cb42-206"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span>
<span id="cb42-207"><a href="#cb42-207"></a><span class="in">```</span></span>
<span id="cb42-208"><a href="#cb42-208"></a></span>
<span id="cb42-211"><a href="#cb42-211"></a><span class="in">```{r}</span></span>
<span id="cb42-212"><a href="#cb42-212"></a><span class="co">#| eval: false</span></span>
<span id="cb42-213"><a href="#cb42-213"></a><span class="co">#| include: false</span></span>
<span id="cb42-214"><a href="#cb42-214"></a><span class="fu">write_rds</span>(syncom16s_seqs_rna_aligned, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_rna_aligned.rds"</span>))</span>
<span id="cb42-215"><a href="#cb42-215"></a><span class="in">```</span></span>
<span id="cb42-216"><a href="#cb42-216"></a></span>
<span id="cb42-219"><a href="#cb42-219"></a><span class="in">```{r}</span></span>
<span id="cb42-220"><a href="#cb42-220"></a><span class="co">#| eval: true</span></span>
<span id="cb42-221"><a href="#cb42-221"></a><span class="co">#| include: false</span></span>
<span id="cb42-222"><a href="#cb42-222"></a>syncom16s_seqs_rna_aligned <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_rna_aligned.rds"</span>))</span>
<span id="cb42-223"><a href="#cb42-223"></a><span class="in">```</span></span>
<span id="cb42-224"><a href="#cb42-224"></a></span>
<span id="cb42-227"><a href="#cb42-227"></a><span class="in">```{r}</span></span>
<span id="cb42-228"><a href="#cb42-228"></a><span class="co"># Write as aligned fasta DNA file </span></span>
<span id="cb42-229"><a href="#cb42-229"></a><span class="fu">writeXStringSet</span>(<span class="fu">DNAStringSet</span>(syncom16s_seqs_rna_aligned),</span>
<span id="cb42-230"><a href="#cb42-230"></a>                here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"16S_rRNA_SynCom_aligned.fna"</span>),</span>
<span id="cb42-231"><a href="#cb42-231"></a>                <span class="at">format=</span><span class="st">'fasta'</span>)</span>
<span id="cb42-232"><a href="#cb42-232"></a><span class="in">```</span></span>
<span id="cb42-233"><a href="#cb42-233"></a></span>
<span id="cb42-234"><a href="#cb42-234"></a><span class="fu"># Phylogenetic tree building</span></span>
<span id="cb42-235"><a href="#cb42-235"></a></span>
<span id="cb42-236"><a href="#cb42-236"></a>Following maximum likelihood tutorial [DECIPHER documentation on</span>
<span id="cb42-237"><a href="#cb42-237"></a>phylogenetic</span>
<span id="cb42-238"><a href="#cb42-238"></a>trees](https://bioconductor.org/packages/devel/bioc/vignettes/DECIPHER/inst/doc/GrowingTrees.pdf)</span>
<span id="cb42-239"><a href="#cb42-239"></a></span>
<span id="cb42-240"><a href="#cb42-240"></a>This step constructs a phylogenetic tree using the general time reversable (GTR) model with a discrete Gamma model <span class="co">[</span><span class="ot">(Yang, 1994)</span><span class="co">](https://link.springer.com/article/10.1007/BF00160154)</span> with the default 4 rate categories. Branch support is depicted using aBayes probabilities, which is a Bayesian-like transformation of an approximate Likelihood Ratio Test <span class="co">[</span><span class="ot">(Anisimova et al., 2011)</span><span class="co">](https://academic.oup.com/sysbio/article/60/5/685/1644562)</span>. </span>
<span id="cb42-241"><a href="#cb42-241"></a></span>
<span id="cb42-244"><a href="#cb42-244"></a><span class="in">```{r}</span></span>
<span id="cb42-245"><a href="#cb42-245"></a><span class="co">#| eval: false</span></span>
<span id="cb42-246"><a href="#cb42-246"></a><span class="co">#| include: true</span></span>
<span id="cb42-247"><a href="#cb42-247"></a><span class="fu">set.seed</span>(<span class="dv">26345</span>)</span>
<span id="cb42-248"><a href="#cb42-248"></a>syncom16s_seqs_mltree <span class="ot">&lt;-</span> <span class="fu">Treeline</span>(</span>
<span id="cb42-249"><a href="#cb42-249"></a>  syncom16s_seqs_rna_aligned,</span>
<span id="cb42-250"><a href="#cb42-250"></a>  <span class="at">method =</span> <span class="st">"ML"</span>,</span>
<span id="cb42-251"><a href="#cb42-251"></a>  <span class="at">model =</span> <span class="st">"GTR+G4"</span>,</span>
<span id="cb42-252"><a href="#cb42-252"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb42-253"><a href="#cb42-253"></a>  <span class="at">processors =</span> <span class="dv">12</span></span>
<span id="cb42-254"><a href="#cb42-254"></a>)</span>
<span id="cb42-255"><a href="#cb42-255"></a><span class="co"># reset random seed</span></span>
<span id="cb42-256"><a href="#cb42-256"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span>
<span id="cb42-257"><a href="#cb42-257"></a><span class="in">```</span></span>
<span id="cb42-258"><a href="#cb42-258"></a></span>
<span id="cb42-261"><a href="#cb42-261"></a><span class="in">```{r}</span></span>
<span id="cb42-262"><a href="#cb42-262"></a><span class="co">#| eval: false</span></span>
<span id="cb42-263"><a href="#cb42-263"></a><span class="co">#| include: false</span></span>
<span id="cb42-264"><a href="#cb42-264"></a><span class="fu">write_rds</span>(syncom16s_seqs_mltree, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.rds"</span>))</span>
<span id="cb42-265"><a href="#cb42-265"></a><span class="in">```</span></span>
<span id="cb42-266"><a href="#cb42-266"></a></span>
<span id="cb42-269"><a href="#cb42-269"></a><span class="in">```{r}</span></span>
<span id="cb42-270"><a href="#cb42-270"></a><span class="co">#| eval: true</span></span>
<span id="cb42-271"><a href="#cb42-271"></a><span class="co">#| include: false</span></span>
<span id="cb42-272"><a href="#cb42-272"></a>syncom16s_seqs_mltree <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.rds"</span>))</span>
<span id="cb42-273"><a href="#cb42-273"></a><span class="in">```</span></span>
<span id="cb42-274"><a href="#cb42-274"></a></span>
<span id="cb42-277"><a href="#cb42-277"></a><span class="in">```{r}</span></span>
<span id="cb42-278"><a href="#cb42-278"></a><span class="co"># this function adds an 'edgetext' attribute that is recognized by WriteDendrogram so that</span></span>
<span id="cb42-279"><a href="#cb42-279"></a><span class="co"># branch support values (aBayes) are written properly</span></span>
<span id="cb42-280"><a href="#cb42-280"></a>add_boots <span class="ot">&lt;-</span> <span class="cf">function</span>(node){</span>
<span id="cb42-281"><a href="#cb42-281"></a>  s <span class="ot">&lt;-</span> <span class="fu">attr</span>(node, <span class="st">"probability"</span>)</span>
<span id="cb42-282"><a href="#cb42-282"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(s))</span>
<span id="cb42-283"><a href="#cb42-283"></a>    <span class="fu">attr</span>(node, <span class="st">"edgetext"</span>) <span class="ot">&lt;-</span> <span class="fu">formatC</span>(<span class="fu">as.numeric</span>(s), <span class="at">digits=</span><span class="dv">2</span>, <span class="at">format=</span><span class="st">"f"</span>)</span>
<span id="cb42-284"><a href="#cb42-284"></a>  <span class="fu">return</span>(node)</span>
<span id="cb42-285"><a href="#cb42-285"></a>}</span>
<span id="cb42-286"><a href="#cb42-286"></a></span>
<span id="cb42-287"><a href="#cb42-287"></a>syncom16s_seqs_mltree <span class="ot">&lt;-</span> <span class="fu">dendrapply</span>(syncom16s_seqs_mltree, add_boots)</span>
<span id="cb42-288"><a href="#cb42-288"></a><span class="in">```</span></span>
<span id="cb42-289"><a href="#cb42-289"></a></span>
<span id="cb42-290"><a href="#cb42-290"></a>Save ML tree as newick file</span>
<span id="cb42-291"><a href="#cb42-291"></a></span>
<span id="cb42-294"><a href="#cb42-294"></a><span class="in">```{r}</span></span>
<span id="cb42-295"><a href="#cb42-295"></a><span class="fu">WriteDendrogram</span>(syncom16s_seqs_mltree, </span>
<span id="cb42-296"><a href="#cb42-296"></a>                <span class="at">file=</span>here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.nwk"</span>),</span>
<span id="cb42-297"><a href="#cb42-297"></a>                <span class="co"># quote ="" ensure that bootstraps are not written quoted as strings</span></span>
<span id="cb42-298"><a href="#cb42-298"></a>                <span class="co"># so that ggtree doesn't fail to parse them</span></span>
<span id="cb42-299"><a href="#cb42-299"></a>                <span class="at">quote=</span><span class="st">""</span>)</span>
<span id="cb42-300"><a href="#cb42-300"></a><span class="in">```</span></span>
<span id="cb42-301"><a href="#cb42-301"></a></span>
<span id="cb42-302"><a href="#cb42-302"></a>Tree building results</span>
<span id="cb42-303"><a href="#cb42-303"></a></span>
<span id="cb42-304"><a href="#cb42-304"></a><span class="in">```         </span></span>
<span id="cb42-305"><a href="#cb42-305"></a><span class="in">Fitting initial tree to model:</span></span>
<span id="cb42-306"><a href="#cb42-306"></a><span class="in">GTR+G4 -ln(L) = 15520, AICc = 31633, BIC = 32781</span></span>
<span id="cb42-307"><a href="#cb42-307"></a></span>
<span id="cb42-308"><a href="#cb42-308"></a><span class="in">Optimizing up to 400 candidate trees:</span></span>
<span id="cb42-309"><a href="#cb42-309"></a><span class="in">Tree #136. -ln(L) = 15395.993 (0.000%), 9 Climbs, 0 Grafts of 7                                                                                                       </span></span>
<span id="cb42-310"><a href="#cb42-310"></a></span>
<span id="cb42-311"><a href="#cb42-311"></a><span class="in">Finalizing the best tree (#130):</span></span>
<span id="cb42-312"><a href="#cb42-312"></a><span class="in">-ln(L) = 15395.993 (0.000%), 1 Climb                                                                                                                                  </span></span>
<span id="cb42-313"><a href="#cb42-313"></a></span>
<span id="cb42-314"><a href="#cb42-314"></a><span class="in">Model parameters:</span></span>
<span id="cb42-315"><a href="#cb42-315"></a><span class="in">Frequency(A) = 0.244</span></span>
<span id="cb42-316"><a href="#cb42-316"></a><span class="in">Frequency(C) = 0.223</span></span>
<span id="cb42-317"><a href="#cb42-317"></a><span class="in">Frequency(G) = 0.312</span></span>
<span id="cb42-318"><a href="#cb42-318"></a><span class="in">Frequency(T) = 0.221</span></span>
<span id="cb42-319"><a href="#cb42-319"></a><span class="in">Rate A &lt;-&gt; C = 0.816</span></span>
<span id="cb42-320"><a href="#cb42-320"></a><span class="in">Rate A &lt;-&gt; G = 1.978</span></span>
<span id="cb42-321"><a href="#cb42-321"></a><span class="in">Rate A &lt;-&gt; T = 1.262</span></span>
<span id="cb42-322"><a href="#cb42-322"></a><span class="in">Rate C &lt;-&gt; G = 0.677</span></span>
<span id="cb42-323"><a href="#cb42-323"></a><span class="in">Rate C &lt;-&gt; T = 3.029</span></span>
<span id="cb42-324"><a href="#cb42-324"></a><span class="in">Rate G &lt;-&gt; T = 1.000</span></span>
<span id="cb42-325"><a href="#cb42-325"></a><span class="in">Alpha = 0.302</span></span>
<span id="cb42-326"><a href="#cb42-326"></a></span>
<span id="cb42-327"><a href="#cb42-327"></a><span class="in">Time difference of 1090.3 secs</span></span>
<span id="cb42-328"><a href="#cb42-328"></a><span class="in">```</span></span>
<span id="cb42-329"><a href="#cb42-329"></a></span>
<span id="cb42-330"><a href="#cb42-330"></a><span class="fu"># Phylogenetic tree plotting</span></span>
<span id="cb42-331"><a href="#cb42-331"></a></span>
<span id="cb42-334"><a href="#cb42-334"></a><span class="in">```{r}</span></span>
<span id="cb42-335"><a href="#cb42-335"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb42-336"><a href="#cb42-336"></a><span class="fu">library</span>(treeio)</span>
<span id="cb42-337"><a href="#cb42-337"></a>syncom16s_seqs_mltree_nwk <span class="ot">&lt;-</span> treeio<span class="sc">::</span><span class="fu">read.newick</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"sanger_seq"</span>, <span class="st">"syncom16s_seqs_mltree.nwk"</span>),</span>
<span id="cb42-338"><a href="#cb42-338"></a>                            <span class="at">node.label=</span><span class="st">"support"</span>)</span>
<span id="cb42-339"><a href="#cb42-339"></a><span class="in">```</span></span>
<span id="cb42-340"><a href="#cb42-340"></a></span>
<span id="cb42-341"><a href="#cb42-341"></a>Goal is to produce a tree that has for tip labels the Genus assigned by IDTAXA + the strain identifier, with different internal nodes labeled by coarser taxonomic grouping. The clusters assigned by <span class="in">`clusterizer`</span> will be represented by different colors at the tips.</span>
<span id="cb42-342"><a href="#cb42-342"></a></span>
<span id="cb42-343"><a href="#cb42-343"></a>First lets look at the tree with just the identifiers on the leaves</span>
<span id="cb42-344"><a href="#cb42-344"></a></span>
<span id="cb42-345"><a href="#cb42-345"></a>::: {#fig-01}</span>
<span id="cb42-348"><a href="#cb42-348"></a><span class="in">```{r}</span></span>
<span id="cb42-349"><a href="#cb42-349"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb42-350"><a href="#cb42-350"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb42-351"><a href="#cb42-351"></a><span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>) <span class="sc">+</span> </span>
<span id="cb42-352"><a href="#cb42-352"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb42-353"><a href="#cb42-353"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb42-354"><a href="#cb42-354"></a>  <span class="fu">geom_treescale</span>(<span class="at">offset=</span><span class="dv">1</span>)</span>
<span id="cb42-355"><a href="#cb42-355"></a><span class="in">```</span></span>
<span id="cb42-356"><a href="#cb42-356"></a>Maximum likelihood phylogenetic tree inferred from an alignment of the 16S rRNA gene amplified by Sanger sequencing from the duckweed SynCom. Scale bar at bottom represents 0.1 substitutions per site.</span>
<span id="cb42-357"><a href="#cb42-357"></a>:::</span>
<span id="cb42-358"><a href="#cb42-358"></a></span>
<span id="cb42-359"><a href="#cb42-359"></a>We can also look at the internal node numbering by rendering the tree as a dataframe</span>
<span id="cb42-360"><a href="#cb42-360"></a></span>
<span id="cb42-363"><a href="#cb42-363"></a><span class="in">```{r}</span></span>
<span id="cb42-364"><a href="#cb42-364"></a><span class="co"># the second tibble call is so quarto renders it properly</span></span>
<span id="cb42-365"><a href="#cb42-365"></a>syncom16s_seqs_mltree_nwk <span class="sc">%&gt;%</span> treeio<span class="sc">::</span><span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> <span class="fu">tibble</span>()</span>
<span id="cb42-366"><a href="#cb42-366"></a><span class="in">```</span></span>
<span id="cb42-367"><a href="#cb42-367"></a>First we need to get our naming scheme set up</span>
<span id="cb42-368"><a href="#cb42-368"></a></span>
<span id="cb42-371"><a href="#cb42-371"></a><span class="in">```{r}</span></span>
<span id="cb42-372"><a href="#cb42-372"></a>newtipnames <span class="ot">&lt;-</span> syncom16s_idtx_lineage_info <span class="sc">%&gt;%</span> </span>
<span id="cb42-373"><a href="#cb42-373"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(genus, <span class="st">"(^[:alpha:]*).*"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-374"><a href="#cb42-374"></a>  <span class="fu">mutate</span>(<span class="at">genus =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(genus) <span class="sc">|</span> genus <span class="sc">==</span> <span class="st">"unclassified"</span> <span class="sc">~</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(ncbi_best_species, <span class="st">"^[:alpha:]*"</span>),</span>
<span id="cb42-375"><a href="#cb42-375"></a>                           <span class="cn">TRUE</span> <span class="sc">~</span> genus)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-376"><a href="#cb42-376"></a>  <span class="fu">mutate</span>(<span class="at">newname =</span> <span class="fu">paste</span>(genus, id)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-377"><a href="#cb42-377"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> <span class="fu">paste0</span>(<span class="st">"c"</span>, cluster)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-378"><a href="#cb42-378"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(id, newname, cluster, genus, order, family)</span>
<span id="cb42-379"><a href="#cb42-379"></a><span class="in">```</span></span>
<span id="cb42-380"><a href="#cb42-380"></a></span>
<span id="cb42-381"><a href="#cb42-381"></a>First we need to generate a color palette that lets us distinguish the different colors</span>
<span id="cb42-382"><a href="#cb42-382"></a></span>
<span id="cb42-385"><a href="#cb42-385"></a><span class="in">```{r}</span></span>
<span id="cb42-386"><a href="#cb42-386"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb42-387"><a href="#cb42-387"></a><span class="fu">set.seed</span>(<span class="dv">341234</span>)</span>
<span id="cb42-388"><a href="#cb42-388"></a>mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">max</span>(syncom16s_idtx_lineage_info<span class="sc">$</span>cluster), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb42-389"><a href="#cb42-389"></a><span class="fu">rm</span>(.Random.seed, <span class="at">envir=</span><span class="fu">globalenv</span>())</span>
<span id="cb42-390"><a href="#cb42-390"></a><span class="in">```</span></span>
<span id="cb42-391"><a href="#cb42-391"></a></span>
<span id="cb42-392"><a href="#cb42-392"></a>Now we need to identify the internal nodes that we want to highlight. We can do this using the <span class="in">`MRCA`</span> function from ggtree</span>
<span id="cb42-393"><a href="#cb42-393"></a></span>
<span id="cb42-396"><a href="#cb42-396"></a><span class="in">```{r}</span></span>
<span id="cb42-397"><a href="#cb42-397"></a><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>)</span>
<span id="cb42-398"><a href="#cb42-398"></a><span class="in">```</span></span>
<span id="cb42-399"><a href="#cb42-399"></a></span>
<span id="cb42-400"><a href="#cb42-400"></a>Code to plot the tree</span>
<span id="cb42-401"><a href="#cb42-401"></a></span>
<span id="cb42-404"><a href="#cb42-404"></a><span class="in">```{r}</span></span>
<span id="cb42-405"><a href="#cb42-405"></a>offset <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb42-406"><a href="#cb42-406"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(syncom16s_seqs_mltree_nwk, <span class="at">layout=</span><span class="st">"rectangular"</span>)</span>
<span id="cb42-407"><a href="#cb42-407"></a></span>
<span id="cb42-408"><a href="#cb42-408"></a>p2 <span class="ot">&lt;-</span> p1 <span class="sc">%&lt;+%</span> newtipnames <span class="sc">+</span> </span>
<span id="cb42-409"><a href="#cb42-409"></a>  <span class="fu">geom_tiplab</span>(<span class="fu">aes</span>(<span class="at">label =</span> newname), <span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.10</span>) <span class="sc">+</span></span>
<span id="cb42-410"><a href="#cb42-410"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> cluster), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb42-411"><a href="#cb42-411"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal, <span class="at">guide=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb42-412"><a href="#cb42-412"></a>  <span class="co"># Pseudomonadaceae </span></span>
<span id="cb42-413"><a href="#cb42-413"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL22"</span>, <span class="st">"TL23"</span>), </span>
<span id="cb42-414"><a href="#cb42-414"></a>                <span class="at">label=</span><span class="st">"(γ) Pseudomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-415"><a href="#cb42-415"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-416"><a href="#cb42-416"></a>  <span class="co"># Aeromonadaceae</span></span>
<span id="cb42-417"><a href="#cb42-417"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA41"</span>, <span class="st">"HA04"</span>), </span>
<span id="cb42-418"><a href="#cb42-418"></a>                <span class="at">label=</span><span class="st">"(γ) Aeromonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-419"><a href="#cb42-419"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-420"><a href="#cb42-420"></a>  <span class="co"># Enterobacteriaceae</span></span>
<span id="cb42-421"><a href="#cb42-421"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA14"</span>, <span class="st">"HA44"</span>), </span>
<span id="cb42-422"><a href="#cb42-422"></a>                <span class="at">label=</span><span class="st">"(γ) Enterobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-423"><a href="#cb42-423"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-424"><a href="#cb42-424"></a>  <span class="co"># Moraxellaceae</span></span>
<span id="cb42-425"><a href="#cb42-425"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL64"</span>, <span class="st">"HA82"</span>), </span>
<span id="cb42-426"><a href="#cb42-426"></a>                <span class="at">label=</span><span class="st">"(γ) Moraxellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-427"><a href="#cb42-427"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-428"><a href="#cb42-428"></a>  <span class="co"># Chromobacteriaceae</span></span>
<span id="cb42-429"><a href="#cb42-429"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA47"</span>, <span class="st">"HA65"</span>), </span>
<span id="cb42-430"><a href="#cb42-430"></a>                <span class="at">label=</span><span class="st">"(γ) Chromobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-431"><a href="#cb42-431"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-432"><a href="#cb42-432"></a>  <span class="co"># Burkholderiaceae</span></span>
<span id="cb42-433"><a href="#cb42-433"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA52"</span>, <span class="st">"HA18"</span>), </span>
<span id="cb42-434"><a href="#cb42-434"></a>                <span class="at">label=</span><span class="st">"(γ) Burkholderiaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-435"><a href="#cb42-435"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-436"><a href="#cb42-436"></a>  <span class="co"># Sphingomonadaceae</span></span>
<span id="cb42-437"><a href="#cb42-437"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="dv">23</span>, </span>
<span id="cb42-438"><a href="#cb42-438"></a>                <span class="at">label=</span><span class="st">"(α) Sphingomonadaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-439"><a href="#cb42-439"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-440"><a href="#cb42-440"></a>  <span class="co"># Streptococcaceae</span></span>
<span id="cb42-441"><a href="#cb42-441"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"TL12"</span>, <span class="st">"TL39"</span>), </span>
<span id="cb42-442"><a href="#cb42-442"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Streptococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-443"><a href="#cb42-443"></a>                <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-444"><a href="#cb42-444"></a>  <span class="co"># Exiguobacteriaceae</span></span>
<span id="cb42-445"><a href="#cb42-445"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA63"</span>, <span class="st">"HA25"</span>), </span>
<span id="cb42-446"><a href="#cb42-446"></a>                <span class="at">label=</span><span class="st">"(Bacilli) Exiguobacteriaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-447"><a href="#cb42-447"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset) <span class="sc">+</span></span>
<span id="cb42-448"><a href="#cb42-448"></a>  <span class="co"># Deinococcaceae</span></span>
<span id="cb42-449"><a href="#cb42-449"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA42"</span>, <span class="st">"HA29"</span>), </span>
<span id="cb42-450"><a href="#cb42-450"></a>                <span class="at">label=</span><span class="st">"(Deinococci) Deinococcaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-451"><a href="#cb42-451"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.2</span>) <span class="sc">+</span></span>
<span id="cb42-452"><a href="#cb42-452"></a>  <span class="co"># Weeksellaceae</span></span>
<span id="cb42-453"><a href="#cb42-453"></a>  <span class="fu">geom_cladelab</span>(<span class="at">node=</span><span class="fu">MRCA</span>(syncom16s_seqs_mltree_nwk, <span class="st">"HA56"</span>, <span class="st">"HA57"</span>), </span>
<span id="cb42-454"><a href="#cb42-454"></a>                <span class="at">label=</span><span class="st">"(Bacteroidia) Weeksellaceae"</span>, <span class="at">align=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-455"><a href="#cb42-455"></a>                  <span class="at">geom=</span><span class="st">'text'</span>, <span class="at">offset=</span>offset<span class="fl">+0.4</span>) <span class="sc">+</span></span>
<span id="cb42-456"><a href="#cb42-456"></a>  <span class="fu">xlim</span>(<span class="cn">NA</span>, <span class="dv">2</span>)</span>
<span id="cb42-457"><a href="#cb42-457"></a><span class="in">```</span></span>
<span id="cb42-458"><a href="#cb42-458"></a></span>
<span id="cb42-459"><a href="#cb42-459"></a>::: {#fig-02}</span>
<span id="cb42-462"><a href="#cb42-462"></a><span class="in">```{r}</span></span>
<span id="cb42-463"><a href="#cb42-463"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb42-464"><a href="#cb42-464"></a><span class="co">#| fig-height: 15</span></span>
<span id="cb42-465"><a href="#cb42-465"></a><span class="co">#| eval: true</span></span>
<span id="cb42-466"><a href="#cb42-466"></a><span class="co">#| echo: false</span></span>
<span id="cb42-467"><a href="#cb42-467"></a>p2</span>
<span id="cb42-468"><a href="#cb42-468"></a><span class="in">```</span></span>
<span id="cb42-469"><a href="#cb42-469"></a>As in @fig-01, but isolates have been labeled with their inferred genus (text) as well as their broader taxonomic class (right, vertical lines). Colors represent sequences clusters based on a 99% identity threshold (i.e., nodes of the same color have 16S rRNA gene sequences that are &gt; 99% similar)</span>
<span id="cb42-470"><a href="#cb42-470"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>